---
title: Replication of literature GxEs related to SFA and development of genetic response scores
output: pdf_document
---

```{r setup, include=F}
knitr::opts_chunk$set(echo=F, cache.path="../cache/replication_to_grs/",
                      fig.path="../output/figures/replication_to_grs/")
suppressMessages(silent <- lapply(
  c("knitr", "tidyverse", "GWASTools", "SNPRelate", "kableExtra", "foreach",
    "broom"), 
  library, character.only=T))
```

```{r funcs}
bim_to_SnADF <- function(bfile) {
  bim_df <- read_tsv(paste0(bfile, ".bim"), 
                     col_names=c("chromosome", "rsID", "cm", 
                                 "position", "A1", "A2")) %>%
    mutate(snpID=1:nrow(.)) %>%
    mutate_at(vars(chromosome, position), as.integer)
  SnpAnnotationDataFrame(data.frame(bim_df, stringsAsFactors=F))
}

fam_to_ScADF <- function(bfile, phenos) {
  fam_df <- read_delim(paste0(bfile, ".fam"), 
                       delim=" ", col_names=c("FID", "IID", "father", "mother", 
                                               "sex", "pheno")) %>%
    select(IID) %>%
    left_join(phenos, by=c("IID")) %>%
    mutate(scanID=IID)
  ScanAnnotationDataFrame(data.frame(fam_df, stringsAsFactors=F))
}

make_gds <- function(bfile, gds_name, summary=F) {
  snpgdsBED2GDS(paste0(bfile, ".bed"),
                paste0(bfile, ".fam"),
                paste0(bfile, ".bim"),
                gds_name,
                cvt.snpid="int",
                verbose=F)
  if (summary) snpgdsSummary(gds_name)
}

run_test <- function(genoData, outcome, covars, ivar, robust, start, end) {
  assocRegression(
    genoData,
    outcome=outcome,
    model.type="linear",
    gene.action="additive",
    covar=covars,
    ivar=ivar,
    robust=robust,
    snpStart=start, 
    snpEnd=end,
    verbose=F)
}

make_qqplot <- function(p_vec, plotTitle="Title") {
  p_vec <- p_vec[!is.na(p_vec)]
  qqplot(-log10(1:length(p_vec) / length(p_vec)), -log10(p_vec), pch=".", cex=5, 
         main=plotTitle, xlab="Expected (-logP)", ylab="Observed (-logP)")
  abline(0, 1, col="red")
}

gControl <- function(p_vals) {
  # See van Iterson 2017 methods and/or Lehne 2015 code for details on genomic control for EWAS
  # Below is modeled after Lehne 2015
  lambda <- median(qchisq(p_vals, df=1, lower.tail=F), 
                   na.rm=T) / qchisq(0.5, df=1)
  round(lambda, 2)
}
```

```{r cardiogxe, message=F}
cardio_gxe <- read_tsv("../data/raw/literature/cardio_gxe_catalog.txt", skip=2)
all_cardioGxE_snps <- unique(unlist(strsplit(cardio_gxe$SNP, split=", | or ")))
write_lines(all_cardioGxE_snps, "../data/processed/all_CardioGxE_snps.txt")

sfa_terms <- c("SFA", "saturated fat")
fat_terms <- c("fat, total", "high-fat", "low-fat")

sfa_gxe <- cardio_gxe %>% 
  filter(`Interaction significance` == "significant interaction",
         grepl(paste(sfa_terms, collapse="|"), `Environmental factor`)
         # grepl(paste(fat_terms, collapse="|"), `Environmental factor`)
         ) %>%
  select(SNP, Gene, `Risk allele`,
         `Environmental factor`, `Condition of environmental factor`,
         Phenotype)

fat_gxe <- cardio_gxe %>% 
  filter(`Interaction significance` == "significant interaction",
         grepl(paste(fat_terms, collapse="|"), `Environmental factor`)
         ) %>%
  select(SNP, Gene, `Risk allele`,
         `Environmental factor`, `Condition of environmental factor`,
         Phenotype)

rfs <- c("logBMI", "logSBP", "ldl", "logHDL", "logTG", "logGLU")

rf_terms <- list(
  logBMI=c("delta body weight", "body weight", "weight", "BMI", "delta BMI"),
  logSBP=c("blood pressure"),
  ldl=c("LDL", "cholesterol"),
  logHDL=c("HDL"),
  logTG=c("triglyceride", "tg"),
  logGLU=c("glu")
)
```

```{r load-whi, message=F, warning=F}
whi_bfile <- "../data/processed/whi/whi_hypothesis"
whi_gds_name <- paste0(whi_bfile, ".gds")
make_gds(whi_bfile, whi_gds_name)
whi_gds <- GdsGenotypeReader(openfn.gds(whi_gds_name, allow.fork=T))

whi_snpAnnot <- bim_to_SnADF(whi_bfile)
whi_rs_to_id <- with(getAnnotation(whi_snpAnnot), setNames(snpID, rsID))


whi_phenos <- read_delim("../data/processed/gen4/whi_white_gwas_phenos.txt",
                         delim=" ") %>%
  mutate(sex="F",
         sfa_pct=sfa / tot_cal * 100,
         mufa_pct = mufa / tot_cal * 100,
         pufa_pct = pufa / tot_cal * 100,
         fat_pct=fat / tot_cal * 100,
         fat2carb = fat / cho,
         sfa_binary = sfa > quantile(sfa, 0.5),
         sfa_pct_binary = sfa_pct > quantile(sfa_pct, 0.5),
         sfa_tert = findInterval(sfa, quantile(sfa, c(0.33, 0.66))),
         sfa_tert_nomid = ifelse(sfa_tert == 1, NA, sfa_tert),
         sfa_pct_tert = findInterval(sfa_pct, quantile(sfa_pct, c(0.33, 0.66))),
         sfa_pct_tert_nomid = ifelse(sfa_pct_tert == 1, NA, sfa_pct_tert),
         logsfa=log(sfa),
         logsfa_pct=log(sfa_pct),
         fat_binary = fat > quantile(fat, 0.5),
         fat_pct_binary = fat_pct > quantile(fat_pct, 0.5),
         fat_tert = findInterval(fat, quantile(fat, c(0.33, 0.66))),
         fat_tert_nomid = ifelse(fat_tert == 1, NA, fat_tert),
         fat_pct_tert = findInterval(fat_pct, quantile(fat_pct, c(0.33, 0.66))),
         fat_pct_tert_nomid = ifelse(fat_pct_tert == 1, NA, fat_pct_tert),
         logfat=log(fat),
         logfat_pct=log(fat_pct),
         palm_pct = palm / tot_cal * 100,
         palm_binary = palm > quantile(palm, 0.5))

rf_pca <- whi_phenos %>%
  select(rfs) %>%
  na.omit() %>%
  prcomp(scale.=T)
whi_phenos$rfPC1 <- predict(rf_pca, select(whi_phenos, rfs))[, "PC1"]

# mutate(myhei=-scale(palm) + scale(pufa) + scale(n3))
whi_scanAnnot <- fam_to_ScADF(whi_bfile, whi_phenos)
whi_genoData <- GenotypeData(whi_gds, snpAnnot=whi_snpAnnot, 
                             scanAnnot=whi_scanAnnot)
```

```{r snp-sets}
sfa_rf_snp_list <- lapply(rf_terms, function(terms) {
  raw <- sfa_gxe$SNP[grepl(paste(terms, collapse="|"), sfa_gxe$Phenotype)]
  unique(raw[raw %in% names(whi_rs_to_id)])
})

all_sfa_snps <- unlist(strsplit(sfa_gxe$SNP, split=","))
all_sfa_snps <- all_sfa_snps[grepl("^rs", all_sfa_snps)]
all_sfa_snps <- unique(all_sfa_snps[all_sfa_snps %in% names(whi_rs_to_id)])

fat_rf_snp_list <- lapply(rf_terms, function(terms) {
  raw <- fat_gxe$SNP[grepl(paste(terms, collapse="|"), fat_gxe$Phenotype)]
  unique(raw[raw %in% names(whi_rs_to_id)])
})

all_fat_snps <- unlist(strsplit(fat_gxe$SNP, split=","))
all_fat_snps <- all_fat_snps[grepl("^rs", all_fat_snps)]
all_fat_snps <- unique(all_fat_snps[all_fat_snps %in% names(whi_rs_to_id)])
```


# Power calculations

Using Quanto tool, with the following assumptions:

* Variance explained by genotype alone: 15%
* Binary environment w/ 50% prevalence, explaining 5% of variance
* Additive model
(Note: no effect of MAF here given that variances explained are specified.)

```{r power-calcs}
quanto_res <- data.frame(var_expl_GE=c(0.05, 0.1, 0.5, 1),
                         nominal=c(12554, 6275, 1252, 624),
                         suggestive=c(44234, 22110, 4411, 2199),
                         genome_wide=c(63342, 31661, 6316, 3148))

quanto_res %>%
  setNames(c("GxE variance explained (%)", "N (nominal)", "N (suggestive)", 
             "N (genome-wide)")) %>%
  kable(booktabs=T, longtable=T,
        caption="Sample size necessary to achieve power of 0.8")
```

# CardioGxE replication in WHI

1) Set of p-values for every specific SFA -> RF combo --> QQ & K-S test for difference from null p-value distribution --> table listing %s of these groups that are nominally replicated

2) Repeat (1) but for every SFA-related SNP in each RF --> table showing results from (1) and (2)

```{r initial-p-values, message=F, cache=1}
run_sfa_binary_test <- function(snpset, outcome, robust=T) {
  res <- foreach(snp=whi_rs_to_id[snpset],
                 .combine=rbind) %do%
    run_test(whi_genoData, outcome, c("sfa_binary", "tot_cal", "age",
                                      "PC1", "PC2", "PC3", "PC4", "PC5"), 
             "sfa_binary", robust=robust, start=snp, end=snp)
  res
}

run_sfa_pct_binary_test <- function(snpset, outcome, robust=T) {
  res <- foreach(snp=whi_rs_to_id[snpset],
                 .combine=rbind) %do%
    run_test(whi_genoData, outcome, c("sfa_pct_binary", "tot_cal", "age",
                                      "PC1", "PC2", "PC3", "PC4", "PC5"), 
             "sfa_pct_binary", robust=robust, start=snp, end=snp)
  res
}

run_sfa_carbex_test <- function(snpset, outcome, robust=T) {
  res <- foreach(snp=whi_rs_to_id[snpset],
                 .combine=rbind) %do%
    run_test(whi_genoData, outcome, c("sfa", "mufa", "pufa", "pro", 
                                      "alc", "tot_cal", "age",
                                      "PC1", "PC2", "PC3", "PC4", "PC5"), 
             "sfa", robust=robust, start=snp, end=snp)
  res
}

run_sfa_pct_carbex_test <- function(snpset, outcome, robust=T) {
  res <- foreach(snp=whi_rs_to_id[snpset],
                 .combine=rbind) %do%
    run_test(whi_genoData, outcome, c("sfa_pct", "mufa", "pufa", "pro", 
                                      "alc", "tot_cal", "age",
                                      "PC1", "PC2", "PC3", "PC4", "PC5"), 
             "sfa_pct", robust=robust, start=snp, end=snp)
  res
}

run_sfa_pufaex_test <- function(snpset, outcome, robust=T) {
  res <- foreach(snp=whi_rs_to_id[snpset],
                 .combine=rbind) %do%
    run_test(whi_genoData, outcome, c("sfa", "mufa", "cho", "pro", 
                                      "alc", "tot_cal", "age",
                                      "PC1", "PC2", "PC3", "PC4", "PC5"), 
             "sfa", robust=robust, start=snp, end=snp)
  res
}

sfa_specific_res <- map(setNames(rfs, rfs), function(rf) {
  if (length(sfa_rf_snp_list[[rf]]) == 0) return(data.frame(snpID=NA))
  run_sfa_binary_test(sfa_rf_snp_list[[rf]], rf)
})

sfa_allsnps_res <- map(c(setNames(rfs, rfs), rfPC1="rfPC1"), function(rf) {
  run_sfa_binary_test(all_sfa_snps, rf)
})

sfa_pct_allsnps_res <- map(setNames(rfs, rfs), function(rf) {
  run_sfa_pct_binary_test(all_sfa_snps, rf)
})

sfa_carbex_allsnps_res <- map(setNames(rfs, rfs), function(rf) {
  run_sfa_carbex_test(all_sfa_snps, rf)
})

sfa_pct_carbex_allsnps_res <- map(setNames(rfs, rfs), function(rf) {
  run_sfa_pct_carbex_test(all_sfa_snps, rf)
})

sfa_pufaex_allsnps_res <- map(setNames(rfs, rfs), function(rf) {
  run_sfa_pufaex_test(all_sfa_snps, rf)
})

run_pro_binary_test <- function(snpset, outcome, robust=T) {
  res <- foreach(snp=whi_rs_to_id[snpset],
                 .combine=rbind) %do%
    run_test(whi_genoData, outcome, c("pro", "tot_cal", "age",
                                      "PC1", "PC2", "PC3", "PC4", "PC5"), 
             "pro", robust=robust, start=snp, end=snp)
  res
}

pro_allsnps_res <- map(setNames(rfs, rfs), function(rf) {
  run_pro_binary_test(all_sfa_snps, rf)
})
```

```{r show-p-distribution}
par(mfrow=c(2, 2))

# silent <- lapply(rfs, function(rf) {
#   make_qqplot(
#     sfa_allsnps_res[[rf]]$GxE.pval,
#     paste0(rf, ": K-S p-value = ",
#            signif(ks.test(sfa_allsnps_res[[rf]]$GxE.pval, punif)$p.value, 3)))
# })

# silent <- lapply(rfs, function(rf) {
#   make_qqplot(
#     pro_allsnps_res[[rf]]$GxE.pval,
#     paste0(rf, ": K-S p-value = ",
#            signif(ks.test(pro_allsnps_res[[rf]]$GxE.pval, punif)$p.value, 3)))
# })

ks_pvals <- lapply(c(setNames(rfs, rfs), rfPC1="rfPC1"), function(rf) {
  round(ks.test(sfa_allsnps_res[[rf]]$GxE.pval, punif)$p.value, 3)
})

sfa_specific_res_tbl <- sfa_specific_res %>%
  bind_rows(.id="rf") %>%
  group_by(rf) %>%
  summarise(num_replicated=sum(GxE.pval < 0.05, na.rm=T)) %>%
  mutate(num_rf_assoc=unlist(lapply(sfa_rf_snp_list[rf], length)),
         frac_replicated_spec=paste(num_replicated, "/", num_rf_assoc)) %>%
  select(rf, frac_replicated_spec)

sfa_allsnps_res %>%
  bind_rows(.id="rf") %>%
  group_by(rf) %>%
  summarise(num_replicated=sum(GxE.pval < 0.05, na.rm=T)) %>%
  mutate(perc_replicated_all=round(num_replicated / length(all_sfa_snps) * 100),
         frac_replicated_all=paste0(num_replicated, " / ", 
                                    length(all_sfa_snps), 
                                    " (", perc_replicated_all, "%)"),
         ks_pval=unlist(ks_pvals[rf])) %>%
  inner_join(sfa_specific_res_tbl, by="rf") %>%
  arrange(desc(perc_replicated_all)) %>%
  select(rf, frac_replicated_spec, frac_replicated_all, ks_pval) %>%
  setNames(c("Risk factor", "Replicated (specific)", "Replicated (all SFA SNPs)",
             "K-S p-value")) %>%
  kable(booktabs=T, longtable=T, caption="Replication of CardioGxE SNPs in WHI")

ks_pvals_pct <- lapply(setNames(rfs, rfs), function(rf) {
  round(ks.test(sfa_pct_allsnps_res[[rf]]$GxE.pval, punif)$p.value, 3)
})

sfa_pct_allsnps_res %>%
  bind_rows(.id="rf") %>%
  group_by(rf) %>%
  summarise(num_replicated=sum(GxE.pval < 0.05, na.rm=T)) %>%
  mutate(perc_replicated_all=round(num_replicated / length(all_sfa_snps) * 100),
         frac_replicated_all=paste0(num_replicated, " / ", 
                                    length(all_sfa_snps), 
                                    " (", perc_replicated_all, "%)"),
         ks_pval=unlist(ks_pvals_pct[rf])) %>%
  inner_join(sfa_specific_res_tbl, by="rf") %>%
  arrange(desc(perc_replicated_all)) %>%
  select(rf, frac_replicated_spec, frac_replicated_all, ks_pval) %>%
  setNames(c("Risk factor", "Replicated (specific)", "Replicated (all SFA SNPs)",
             "K-S p-value")) %>%
  kable(booktabs=T, longtable=T, caption="Replication of CardioGxE SNPs in WHI w/ percentages")

ks_pvals_carbex <- lapply(setNames(rfs, rfs), function(rf) {
  round(ks.test(sfa_carbex_allsnps_res[[rf]]$GxE.pval, punif)$p.value, 3)
})

sfa_carbex_allsnps_res %>%
  bind_rows(.id="rf") %>%
  group_by(rf) %>%
  summarise(num_replicated=sum(GxE.pval < 0.05, na.rm=T)) %>%
  mutate(perc_replicated_all=round(num_replicated / length(all_sfa_snps) * 100),
         frac_replicated_all=paste0(num_replicated, " / ", 
                                    length(all_sfa_snps), 
                                    " (", perc_replicated_all, "%)"),
         ks_pval=unlist(ks_pvals_carbex[rf])) %>%
  inner_join(sfa_specific_res_tbl, by="rf") %>%
  arrange(desc(perc_replicated_all)) %>%
  select(rf, frac_replicated_spec, frac_replicated_all, ks_pval) %>%
  setNames(c("Risk factor", "Replicated (specific)", "Replicated (all SFA SNPs)",
             "K-S p-value")) %>%
  kable(booktabs=T, longtable=T, caption="Replication of CardioGxE SNPs in WHI w/ carbex")

# ks_pvals_pct_carbex <- lapply(setNames(rfs, rfs), function(rf) {
#   round(ks.test(sfa_pct_carbex_allsnps_res[[rf]]$GxE.pval, punif)$p.value, 3)
# })
# 
# sfa_pct_carbex_allsnps_res %>%
#   bind_rows(.id="rf") %>%
#   group_by(rf) %>%
#   summarise(num_replicated=sum(GxE.pval < 0.05, na.rm=T)) %>%
#   mutate(perc_replicated_all=round(num_replicated / length(all_sfa_snps) * 100),
#          frac_replicated_all=paste0(num_replicated, " / ", 
#                                     length(all_sfa_snps), 
#                                     " (", perc_replicated_all, "%)"),
#          ks_pval=unlist(ks_pvals_pct_carbex[rf])) %>%
#   inner_join(sfa_specific_res_tbl, by="rf") %>%
#   arrange(desc(perc_replicated_all)) %>%
#   select(rf, frac_replicated_spec, frac_replicated_all, ks_pval) %>%
#   setNames(c("Risk factor", "Replicated (specific)", "Replicated (all SFA SNPs)",
#              "K-S p-value")) %>%
#   kable(booktabs=T, longtable=T, caption="Replication of CardioGxE SNPs in WHI w/ pct carbex")

# ks_pvals_pufaex <- lapply(setNames(rfs, rfs), function(rf) {
#   round(ks.test(sfa_pufaex_allsnps_res[[rf]]$GxE.pval, punif)$p.value, 3)
# })
# 
# sfa_pufaex_allsnps_res %>%
#   bind_rows(.id="rf") %>%
#   group_by(rf) %>%
#   summarise(num_replicated=sum(GxE.pval < 0.05, na.rm=T)) %>%
#   mutate(perc_replicated_all=round(num_replicated / length(all_sfa_snps) * 100),
#          frac_replicated_all=paste0(num_replicated, " / ", 
#                                     length(all_sfa_snps), 
#                                     " (", perc_replicated_all, "%)"),
#          ks_pval=unlist(ks_pvals_pufaex[rf])) %>%
#   inner_join(sfa_specific_res_tbl, by="rf") %>%
#   arrange(desc(perc_replicated_all)) %>%
#   select(rf, frac_replicated_spec, frac_replicated_all, ks_pval) %>%
#   setNames(c("Risk factor", "Replicated (specific)", "Replicated (all SFA SNPs)",
#              "K-S p-value")) %>%
#   kable(booktabs=T, longtable=T, caption="Replication of CardioGxE SNPs in WHI w/ pufaex")
```

```{r initial-p-values-fat, message=F, cache=1}
run_fat_binary_test <- function(snpset, outcome, robust=T) {
  res <- foreach(snp=whi_rs_to_id[snpset],
                 .combine=rbind) %do%
    run_test(whi_genoData, outcome, c("fat_binary", "tot_cal", "age",
                                      "PC1", "PC2", "PC3", "PC4", "PC5"), 
             "fat_binary", robust=robust, start=snp, end=snp)
  res
}

run_fat_pct_binary_test <- function(snpset, outcome, robust=T) {
  res <- foreach(snp=whi_rs_to_id[snpset],
                 .combine=rbind) %do%
    run_test(whi_genoData, outcome, c("fat_pct_binary", "tot_cal", "age",
                                      "PC1", "PC2", "PC3", "PC4", "PC5"), 
             "fat_pct_binary", robust=robust, start=snp, end=snp)
  res
}

run_fat_carbex_test <- function(snpset, outcome, robust=T) {
  res <- foreach(snp=whi_rs_to_id[snpset],
                 .combine=rbind) %do%
    run_test(whi_genoData, outcome, c("fat", "mufa", "pufa", "pro", 
                                      "alc", "tot_cal", "age",
                                      "PC1", "PC2", "PC3", "PC4", "PC5"), 
             "fat", robust=robust, start=snp, end=snp)
  res
}

run_fat_pct_carbex_test <- function(snpset, outcome, robust=T) {
  res <- foreach(snp=whi_rs_to_id[snpset],
                 .combine=rbind) %do%
    run_test(whi_genoData, outcome, c("fat_pct", "mufa", "pufa", "pro", 
                                      "alc", "tot_cal", "age",
                                      "PC1", "PC2", "PC3", "PC4", "PC5"), 
             "fat_pct", robust=robust, start=snp, end=snp)
  res
}

run_fat_pufaex_test <- function(snpset, outcome, robust=T) {
  res <- foreach(snp=whi_rs_to_id[snpset],
                 .combine=rbind) %do%
    run_test(whi_genoData, outcome, c("fat", "mufa", "cho", "pro", 
                                      "alc", "tot_cal", "age",
                                      "PC1", "PC2", "PC3", "PC4", "PC5"), 
             "fat", robust=robust, start=snp, end=snp)
  res
}

fat_specific_res <- map(setNames(rfs, rfs), function(rf) {
  if (length(fat_rf_snp_list[[rf]]) == 0) return(data.frame(snpID=NA))
  run_fat_binary_test(fat_rf_snp_list[[rf]], rf)
})

fat_allsnps_res <- map(c(setNames(rfs, rfs), rfPC1="rfPC1"), function(rf) {
  run_fat_binary_test(all_fat_snps, rf)
})

fat_pct_allsnps_res <- map(setNames(rfs, rfs), function(rf) {
  run_fat_pct_binary_test(all_fat_snps, rf)
})

fat_carbex_allsnps_res <- map(setNames(rfs, rfs), function(rf) {
  run_fat_carbex_test(all_fat_snps, rf)
})

fat_pct_carbex_allsnps_res <- map(setNames(rfs, rfs), function(rf) {
  run_fat_pct_carbex_test(all_fat_snps, rf)
})

fat_pufaex_allsnps_res <- map(setNames(rfs, rfs), function(rf) {
  run_fat_pufaex_test(all_fat_snps, rf)
})

run_pro_binary_test <- function(snpset, outcome, robust=T) {
  res <- foreach(snp=whi_rs_to_id[snpset],
                 .combine=rbind) %do%
    run_test(whi_genoData, outcome, c("pro", "tot_cal", "age",
                                      "PC1", "PC2", "PC3", "PC4", "PC5"), 
             "pro", robust=robust, start=snp, end=snp)
  res
}

pro_allsnps_res <- map(setNames(rfs, rfs), function(rf) {
  run_pro_binary_test(all_fat_snps, rf)
})
```

```{r show-p-distribution-fat}
par(mfrow=c(2, 2))

ks_pvals <- lapply(c(setNames(rfs, rfs), rfPC1="rfPC1"), function(rf) {
  round(ks.test(fat_allsnps_res[[rf]]$GxE.pval, punif)$p.value, 3)
})

fat_specific_res_tbl <- fat_specific_res %>%
  bind_rows(.id="rf") %>%
  group_by(rf) %>%
  summarise(num_replicated=sum(GxE.pval < 0.05, na.rm=T)) %>%
  mutate(num_rf_assoc=unlist(lapply(fat_rf_snp_list[rf], length)),
         frac_replicated_spec=paste(num_replicated, "/", num_rf_assoc)) %>%
  select(rf, frac_replicated_spec)

fat_allsnps_res %>%
  bind_rows(.id="rf") %>%
  group_by(rf) %>%
  summarise(num_replicated=sum(GxE.pval < 0.05, na.rm=T)) %>%
  mutate(perc_replicated_all=round(num_replicated / length(all_fat_snps) * 100),
         frac_replicated_all=paste0(num_replicated, " / ", 
                                    length(all_fat_snps), 
                                    " (", perc_replicated_all, "%)"),
         ks_pval=unlist(ks_pvals[rf])) %>%
  inner_join(fat_specific_res_tbl, by="rf") %>%
  arrange(desc(perc_replicated_all)) %>%
  select(rf, frac_replicated_spec, frac_replicated_all, ks_pval) %>%
  setNames(c("Risk factor", "Replicated (specific)", "Replicated (all fat SNPs)",
             "K-S p-value")) %>%
  kable(booktabs=T, longtable=T, caption="Replication of CardioGxE SNPs in WHI")

ks_pvals_pct <- lapply(setNames(rfs, rfs), function(rf) {
  round(ks.test(fat_pct_allsnps_res[[rf]]$GxE.pval, punif)$p.value, 3)
})

fat_pct_allsnps_res %>%
  bind_rows(.id="rf") %>%
  group_by(rf) %>%
  summarise(num_replicated=sum(GxE.pval < 0.05, na.rm=T)) %>%
  mutate(perc_replicated_all=round(num_replicated / length(all_fat_snps) * 100),
         frac_replicated_all=paste0(num_replicated, " / ", 
                                    length(all_fat_snps), 
                                    " (", perc_replicated_all, "%)"),
         ks_pval=unlist(ks_pvals_pct[rf])) %>%
  inner_join(fat_specific_res_tbl, by="rf") %>%
  arrange(desc(perc_replicated_all)) %>%
  select(rf, frac_replicated_spec, frac_replicated_all, ks_pval) %>%
  setNames(c("Risk factor", "Replicated (specific)", "Replicated (all fat SNPs)",
             "K-S p-value")) %>%
  kable(booktabs=T, longtable=T, caption="Replication of CardioGxE SNPs in WHI w/ percentages")

# ks_pvals_carbex <- lapply(setNames(rfs, rfs), function(rf) {
#   round(ks.test(fat_carbex_allsnps_res[[rf]]$GxE.pval, punif)$p.value, 3)
# })
# 
# fat_carbex_allsnps_res %>%
#   bind_rows(.id="rf") %>%
#   group_by(rf) %>%
#   summarise(num_replicated=sum(GxE.pval < 0.05, na.rm=T)) %>%
#   mutate(perc_replicated_all=round(num_replicated / length(all_fat_snps) * 100),
#          frac_replicated_all=paste0(num_replicated, " / ", 
#                                     length(all_fat_snps), 
#                                     " (", perc_replicated_all, "%)"),
#          ks_pval=unlist(ks_pvals_carbex[rf])) %>%
#   inner_join(fat_specific_res_tbl, by="rf") %>%
#   arrange(desc(perc_replicated_all)) %>%
#   select(rf, frac_replicated_spec, frac_replicated_all, ks_pval) %>%
#   setNames(c("Risk factor", "Replicated (specific)", "Replicated (all fat SNPs)",
#              "K-S p-value")) %>%
#   kable(booktabs=T, longtable=T, caption="Replication of CardioGxE SNPs in WHI w/ carbex")
# 
# ks_pvals_pct_carbex <- lapply(setNames(rfs, rfs), function(rf) {
#   round(ks.test(fat_pct_carbex_allsnps_res[[rf]]$GxE.pval, punif)$p.value, 3)
# })
# 
# fat_pct_carbex_allsnps_res %>%
#   bind_rows(.id="rf") %>%
#   group_by(rf) %>%
#   summarise(num_replicated=sum(GxE.pval < 0.05, na.rm=T)) %>%
#   mutate(perc_replicated_all=round(num_replicated / length(all_fat_snps) * 100),
#          frac_replicated_all=paste0(num_replicated, " / ", 
#                                     length(all_fat_snps), 
#                                     " (", perc_replicated_all, "%)"),
#          ks_pval=unlist(ks_pvals_pct_carbex[rf])) %>%
#   inner_join(fat_specific_res_tbl, by="rf") %>%
#   arrange(desc(perc_replicated_all)) %>%
#   select(rf, frac_replicated_spec, frac_replicated_all, ks_pval) %>%
#   setNames(c("Risk factor", "Replicated (specific)", "Replicated (all fat SNPs)",
#              "K-S p-value")) %>%
#   kable(booktabs=T, longtable=T, caption="Replication of CardioGxE SNPs in WHI w/ pct carbex")
```

3) Pick the best SNP set/RF combo from table above to look at phenotype wrangling (don't go crazy w/ options):

* SFA continuous
* SFA binary
* SFA tertile (no mid)
* SFA %
* SFA % binary

```{r compare-transformations, message=F, cache=1}
run_transform_test <- function(sfa_trans, snpset, outcome, robust=T) {
  covars <- c("tot_cal", "age", "PC1", "PC2", "PC3", "PC4", "PC5")
  if (grepl("carbEx", sfa_trans)) {
    sfa_trans <- gsub("_carbEx", "", sfa_trans)
    covars <- c(covars, c("pufa", "mufa", "pro", "alc"))
  } 
  res <- foreach(snp=whi_rs_to_id[snpset],
                 .combine=rbind) %do%
    run_test(whi_genoData, outcome, c(sfa_trans, covars), 
             sfa_trans, robust=robust, start=snp, end=snp)
  res
}

sfa_transforms <- c("sfa", "sfa_pct", "sfa_binary", "sfa_pct_binary",
                    "sfa_tert_nomid", "sfa_pct_tert_nomid")
sfa_transforms <- c(sfa_transforms, paste0(sfa_transforms, "_carbEx"))

sfa_transform_res <- map(setNames(sfa_transforms, sfa_transforms), function(st) {
  run_transform_test(st, all_sfa_snps, "logHDL")
})
```

```{r show-compare-transformations}
sfa_transform_res_summary <- sfa_transform_res %>%
  bind_rows(.id="sfa_transform") %>%
  group_by(sfa_transform) %>%
  summarise(mean_abs_t_stat = mean(abs(GxE.Stat)),
            best_p = min(GxE.pval),
            num_replicated = sum(GxE.pval < 0.05, na.rm=T)) %>%
  arrange(desc(mean_abs_t_stat))

kable(sfa_transform_res_summary, 
      caption="SFA transforms and adjustments for SFA -> logHDL GWIS")

# sfa_snp_res_order <- sfa_snp_res_summary %>%
#   group_by(sfa_transform) %>%
#   summarise(mean_mean = mean(mean_abs_t_stat)) %>%
#   arrange(desc(mean_mean))
# sfa_snp_res_summary %>%
#   mutate(sfa_transform = factor(sfa_transform,
#                                 levels=sfa_snp_res_order$sfa_transform)) %>%
#   ggplot(aes(x=rf, y=fct_rev(sfa_transform), fill=mean_abs_t_stat)) + 
#   geom_tile() +
#   labs(title="Comparison of SFA transformations using CardioGxE SNPs")
```

```{r compare-transformations-fat, message=F, cache=1}
run_transform_test <- function(fat_trans, snpset, outcome, robust=T) {
  covars <- c("tot_cal", "age", "PC1", "PC2", "PC3", "PC4", "PC5")
  if (grepl("carbEx", fat_trans)) {
    fat_trans <- gsub("_carbEx", "", fat_trans)
    covars <- c(covars, c("pufa", "mufa", "pro", "alc"))
  } 
  res <- foreach(snp=whi_rs_to_id[snpset],
                 .combine=rbind) %do%
    run_test(whi_genoData, outcome, c(fat_trans, covars), 
             fat_trans, robust=robust, start=snp, end=snp)
  res
}

fat_transforms <- c("fat", "fat_pct", "fat_binary", "fat_pct_binary",
                    "fat_tert_nomid", "fat_pct_tert_nomid")
fat_transforms <- c(fat_transforms, paste0(fat_transforms, "_carbEx"))

fat_transform_res <- map(setNames(fat_transforms, fat_transforms), function(st) {
  run_transform_test(st, all_fat_snps, "logHDL")
})
```

```{r show-compare-transformations-fat}
fat_transform_res_summary <- fat_transform_res %>%
  bind_rows(.id="fat_transform") %>%
  group_by(fat_transform) %>%
  summarise(mean_abs_t_stat = mean(abs(GxE.Stat), na.rm=T),
            best_p = min(GxE.pval, na.rm=T),
            num_replicated = sum(GxE.pval < 0.05, na.rm=T)) %>%
  arrange(desc(mean_abs_t_stat))

kable(fat_transform_res_summary, 
      caption="fat transforms and adjustments for fat -> logHDL GWIS")
```

4) Take that best combo of RF and SFA-transform and run a full GWIS --> figure characterizing the GWIS results (Manhattan, QQ, maybe enrichments, etc.)

```{r examine-gwis, eval=F}
# gwis_res <- read_tsv("../data/processed/gen4/sfa_pct_binary_logHDL_whi_white.res_annot")

gwis_res_list <- lapply(setNames(rfs, rfs), function(rf) {
  read_tsv(paste0("../data/processed/gen4/sfa_pct_binary_", rf, 
                  "_whi_white.res_annot"))
})

# Perform FUMA analysis
```

5) Create a set of risk scores using P&T w/ seed of p<0.05

i) Single best-performing interaction SNP
ii) All CardioGxE SFA SNPs
iii) All nominal BMI SNPs
iv) Genome-wide
v) CardioGxE SFA SNPs but with CardioGxE weights (no P&T)

```{r score-setup, eval=F}
cge_snps_score_setup <- transform_res$sfa_pct_binary %>%
  inner_join(getAnnotation(whi_snpAnnot), by="snpID") %>%
  mutate(BETA=GxE.Est,
         P=GxE.pval,
         BETA=ifelse(effect.allele == "A", BETA, -BETA)) %>%
  select(SNP=rsID, chr=chromosome, bp=position, A1, BETA, P)

# Single best-performing interaction SNP
cge_snps_score_setup %>%
  arrange(P) %>%
  dplyr::slice(1) %>%
  write_tsv("../data/processed/gen4/scores/single_snp_sumstats.txt")

# All CardioGxE SFA SNPs
cge_snps_score_setup %>%
  write_tsv("../data/processed/gen4/scores/full_cge_sumstats.txt")

# HDL nomianl SNPs
hdl_nominal_snps <- scan("../data/processed/whi_subsets/hdl_nominal_snps.txt",
                         what=character())
gwis_res %>%
  filter(SNP %in% hdl_nominal_snps) %>%
  select(SNP, chr, bp, A1, BETA, P) %>%
  write_tsv("../data/processed/gen4/scores/hdl_nominal_sumstats.txt")

# HDL suggestive SNPs
hdl_suggestive_snps <- scan("../data/processed/whi_subsets/hdl_suggestive_snps.txt",
                         what=character())
gwis_res %>%
  filter(SNP %in% hdl_suggestive_snps) %>%
  select(SNP, chr, bp, A1, BETA, P) %>%
  write_tsv("../data/processed/gen4/scores/hdl_suggestive_sumstats.txt")

# All SNPs
gwis_res %>%
  filter(P < 0.05) %>%
  select(SNP, chr, bp, A1, BETA, P) %>%
  write_tsv("../data/processed/gen4/scores/all_sumstats.txt")

# CardioGxE-based weights
sfa_weights <- sfa_gxe %>%
  inner_join(getAnnotation(whi_snpAnnot), by=c("SNP"="rsID")) %>% 
  distinct(SNP, .keep_all=T) %>%
  mutate(weight=case_when(`Risk allele` == A1 ~ 1,
                          `Risk allele` == A2 ~ -1,
                          TRUE ~ as.numeric(NA)),
         P=1e-5) %>%
  select(SNP, chr=chromosome, bp=position, A1, BETA=weight, P) %>%
  write_tsv("../data/processed/gen4/scores/sfa_lit_sumstats.txt")

# Direct HDL PRS-based weights
hdl_gwas_res <- read_tsv("../data/raw/literature/hdl_gwas_ma_res_glgc.txt") %>%
  mutate(chr=0,
         bp=0,
         A1=toupper(A1)) %>%
  select(SNP=rsid, chr, bp, A1, BETA=beta, P=`P-value`) %>%
  filter(P < 0.05) %>%
  write_tsv("../data/processed/gen4/scores/hdl_glgc_sumstats.txt")

  # select(SNP, `Risk allele`, A1, A2, weight) %>%
  # na.omit() %>%
  # select(SNP, REF=A2, WEIGHT=weight)
```

```{r main-effect-prs, eval=F}
ldl_gwas_res <- read_tsv("../data/raw/literature/ldl_gwas_ma_res_glgc.txt") %>%
  mutate(chr=0,
         bp=0,
         A1=toupper(A1)) %>%
  select(SNP=rsid, chr, bp, A1, BETA=beta, P=`P-value`) %>%
  filter(!duplicated(SNP),
         P < 0.05) %>%
  write_tsv("../data/processed/gen4/scores/sfa_pct_binary_ldl_onlyME_sumstats.txt")

hdl_gwas_res <- read_tsv("../data/raw/literature/hdl_gwas_ma_res_glgc.txt") %>%
  mutate(chr=0,
         bp=0,
         A1=toupper(A1)) %>%
  select(SNP=rsid, chr, bp, A1, BETA=beta, P=`P-value`) %>%
  filter(!duplicated(SNP),
         P < 0.05) %>%
  write_tsv("../data/processed/gen4/scores/sfa_pct_binary_logHDL_onlyME_sumstats.txt")

tg_gwas_res <- read_tsv("../data/raw/literature/tg_gwas_ma_res_glgc.txt") %>%
  mutate(chr=0,
         bp=0,
         A1=toupper(A1)) %>%
  select(SNP=rsid, chr, bp, A1, BETA=beta, P=`P-value`) %>%
  filter(!duplicated(SNP),
         P < 0.05) %>%
  write_tsv("../data/processed/gen4/scores/sfa_pct_binary_logTG_onlyME_sumstats.txt")

bmi_gwas_res <- read_delim("../data/raw/literature/Bmi.giant-ukbb.meta-analysis.combined.23May2018.HapMap2_only.txt", delim=" ") %>%
  select(SNP, chr=CHR, bp=POS, A1=Tested_Allele, BETA, P) %>%
  filter(P < 0.05) %>%
  write_tsv("../data/processed/gen4/scores/sfa_pct_binary_logBMI_onlyME_sumstats.txt")
  
# ## SBP SUMMARY STATS NEED UPDATING ##
# sbp_gwas_res <- read_csv("../data/raw/literature/2010-07-09663E-ICBP-summary-Nature.csv") %>%
#   mutate(BETA=1,
#          bp=0) %>%
#   select(SNP=rsid, chr=chr.hg18, bp, A1=Tested_Allele, BETA, P) %>%
#   filter(P < 0.05) %>%
#   write_tsv("../data/processed/gen4/scores/sfa_pct_binary_logSBP_onlyME_sumstats.txt")

glu_gwas_res <- read_tsv("../data/raw/literature/MAGIC_FastingGlucose.txt") %>%
  mutate(chr=0,
         bp=0) %>%
  select(SNP=snp, chr, bp, A1=effect_allele, BETA=effect, P=pvalue) %>%
  filter(P < 0.05) %>%
  write_tsv("../data/processed/gen4/scores/sfa_pct_binary_logGLU_onlyME_sumstats.txt")
```

```{r score-setup-sfa, eval=F}
output_sumstats <- function(rf) {
  non_log_rf <- tolower(gsub("log", "", rf))
  
  print(gControl(gwis_res_list[[rf]]$P))
  
  # Grab SFA_PCT results from GWASTools CardioGxE SNPs
  cge_snps_score_setup <- sfa_pct_allsnps_res[[rf]] %>%
    inner_join(getAnnotation(whi_snpAnnot), by="snpID") %>%
    mutate(BETA=GxE.Est,
           P=GxE.pval,
           BETA=ifelse(effect.allele == "A", BETA, -BETA)) %>%
    select(SNP=rsID, chr=chromosome, bp=position, A1, BETA, P)
  
  # Single best-performing interaction SNP
  cge_snps_score_setup %>%
    arrange(P) %>%
    dplyr::slice(1) %>%
    write_tsv(paste0("../data/processed/gen4/scores/sfa_pct_binary_",
                     rf, "_single_snp_sumstats.txt"))
  
  # All CardioGxE SFA SNPs
  cge_snps_score_setup %>%
    write_tsv(paste0("../data/processed/gen4/scores/sfa_pct_binary_",
                     rf, "_full_cge_sumstats.txt"))
  
  # Nominal SNPs
  nominal_snps <- scan(paste0("../data/processed/whi_subsets/",
                              non_log_rf, "_nominal_snps.txt"),
                       what=character())
  gwis_res_list[[rf]] %>%
    filter(P < 0.05) %>%
    filter(SNP %in% nominal_snps) %>%
    select(SNP, chr, bp, A1, BETA, P) %>%
    write_tsv(paste0("../data/processed/gen4/scores/sfa_pct_binary_",
                     rf, "_nominalME_sumstats.txt"))
  
  # Suggestive SNPs
  suggestive_snps <- scan(paste0("../data/processed/whi_subsets/",
                                 non_log_rf, "_suggestive_snps.txt"),
                          what=character())
  gwis_res_list[[rf]] %>%
    filter(P < 0.05) %>%
    filter(SNP %in% suggestive_snps) %>%
    select(SNP, chr, bp, A1, BETA, P) %>%
    write_tsv(paste0("../data/processed/gen4/scores/sfa_pct_binary_",
                     rf, "_suggestiveME_sumstats.txt"))
  
  # Genome-wide SNPs
  gw_snps <- scan(paste0("../data/processed/whi_subsets/",
                                 non_log_rf, "_gw_snps.txt"),
                          what=character())
  gwis_res_list[[rf]] %>%
    filter(P < 0.05) %>%
    filter(SNP %in% gw_snps) %>%
    select(SNP, chr, bp, A1, BETA, P) %>%
    write_tsv(paste0("../data/processed/gen4/scores/sfa_pct_binary_",
                     rf, "_gwME_sumstats.txt"))
  
  # All SNPs
  gwis_res_list[[rf]] %>%
    filter(P < 0.05) %>%
    select(SNP, chr, bp, A1, BETA, P) %>%
    write_tsv(paste0("../data/processed/gen4/scores/sfa_pct_binary_",
                     rf, "_all_sumstats.txt"))
}

# CardioGxE-based weights
sfa_weights <- sfa_gxe %>%
  inner_join(getAnnotation(whi_snpAnnot), by=c("SNP"="rsID")) %>% 
  distinct(SNP, .keep_all=T) %>%
  mutate(weight=case_when(`Risk allele` == A1 ~ 1,
                          `Risk allele` == A2 ~ -1,
                          TRUE ~ as.numeric(NA)),
         P=1e-5) %>%
  select(SNP, chr=chromosome, bp=position, A1, BETA=weight, P) %>%
  write_tsv("../data/processed/gen4/scores/sfa_lit_sumstats.txt")
```

```{r score-setup-hei, eval=F}
fat_pct_binary_gwis_res_list <- lapply(setNames(rfs, rfs), function(rf) {
  read_tsv(paste0("../data/processed/gen4/fat_pct_binary_", rf, 
                  "_whi_white.res_annot"))
})

fat_pct_binary_output_sumstats <- function(rf) {
  non_log_rf <- tolower(gsub("log", "", rf))
  
  print(gControl(fat_pct_binary_gwis_res_list[[rf]]$P))
  
  # All CardioGxE SNPs
  fat_pct_binary_gwis_res_list[[rf]] %>%
    filter(P < 0.05) %>%
    filter(SNP %in% all_cardioGxE_snps) %>%
    select(SNP, chr, bp, A1, BETA, P) %>%
    write_tsv(paste0("../data/processed/gen4/scores/fat_pct_binary_",
                     rf, "_full_cge_sumstats.txt"))
  
  # Nominal SNPs
  nominal_snps <- scan(paste0("../data/processed/whi_subsets/",
                              non_log_rf, "_nominal_snps.txt"),
                       what=character())
  fat_pct_binary_gwis_res_list[[rf]] %>%
    filter(P < 0.05) %>%
    filter(SNP %in% nominal_snps) %>%
    select(SNP, chr, bp, A1, BETA, P) %>%
    write_tsv(paste0("../data/processed/gen4/scores/fat_pct_binary_",
                     rf, "_nominalME_sumstats.txt"))
  
  # Suggestive SNPs
  suggestive_snps <- scan(paste0("../data/processed/whi_subsets/",
                                 non_log_rf, "_suggestive_snps.txt"),
                          what=character())
  fat_pct_binary_gwis_res_list[[rf]] %>%
    filter(P < 0.05) %>%
    filter(SNP %in% suggestive_snps) %>%
    select(SNP, chr, bp, A1, BETA, P) %>%
    write_tsv(paste0("../data/processed/gen4/scores/fat_pct_binary_",
                     rf, "_suggestiveME_sumstats.txt"))
  
  # Genome-wide SNPs
  gw_snps <- scan(paste0("../data/processed/whi_subsets/",
                                 non_log_rf, "_gw_snps.txt"),
                          what=character())
  fat_pct_binary_gwis_res_list[[rf]] %>%
    filter(P < 0.05) %>%
    filter(SNP %in% gw_snps) %>%
    select(SNP, chr, bp, A1, BETA, P) %>%
    write_tsv(paste0("../data/processed/gen4/scores/fat_pct_binary_",
                     rf, "_gwME_sumstats.txt"))
  
  # All SNPs
  fat_pct_binary_gwis_res_list[[rf]] %>%
    filter(P < 0.05) %>%
    select(SNP, chr, bp, A1, BETA, P) %>%
    write_tsv(paste0("../data/processed/gen4/scores/fat_pct_binary_",
                     rf, "_all_sumstats.txt"))
}

lapply(rfs, fat_pct_binary_output_sumstats)
```

```{r score-setup-fat, eval=F}
hei_gwis_res_list <- lapply(setNames(rfs, rfs)[-2], function(rf) {
  read_tsv(paste0("../data/processed/gen4/hei_", rf, 
                  "_whi_white.res_annot"))
})

hei_output_sumstats <- function(rf) {
  non_log_rf <- tolower(gsub("log", "", rf))
  
  print(gControl(hei_gwis_res_list[[rf]]$P))
  
  # All CardioGxE SNPs
  hei_gwis_res_list[[rf]] %>%
    filter(P < 0.05) %>%
    filter(SNP %in% all_cardioGxE_snps) %>%
    select(SNP, chr, bp, A1, BETA, P) %>%
    write_tsv(paste0("../data/processed/gen4/scores/hei_",
                     rf, "_full_cge_sumstats.txt"))
  
  # Nominal SNPs
  nominal_snps <- scan(paste0("../data/processed/whi_subsets/",
                              non_log_rf, "_nominal_snps.txt"),
                       what=character())
  hei_gwis_res_list[[rf]] %>%
    filter(P < 0.05) %>%
    filter(SNP %in% nominal_snps) %>%
    select(SNP, chr, bp, A1, BETA, P) %>%
    write_tsv(paste0("../data/processed/gen4/scores/hei_",
                     rf, "_nominalME_sumstats.txt"))
  
  # Suggestive SNPs
  suggestive_snps <- scan(paste0("../data/processed/whi_subsets/",
                                 non_log_rf, "_suggestive_snps.txt"),
                          what=character())
  hei_gwis_res_list[[rf]] %>%
    filter(P < 0.05) %>%
    filter(SNP %in% suggestive_snps) %>%
    select(SNP, chr, bp, A1, BETA, P) %>%
    write_tsv(paste0("../data/processed/gen4/scores/hei_",
                     rf, "_suggestiveME_sumstats.txt"))
  
  # Genome-wide SNPs
  gw_snps <- scan(paste0("../data/processed/whi_subsets/",
                                 non_log_rf, "_gw_snps.txt"),
                          what=character())
  hei_gwis_res_list[[rf]] %>%
    filter(P < 0.05) %>%
    filter(SNP %in% gw_snps) %>%
    select(SNP, chr, bp, A1, BETA, P) %>%
    write_tsv(paste0("../data/processed/gen4/scores/hei_",
                     rf, "_gwME_sumstats.txt"))
  
  # All SNPs
  hei_gwis_res_list[[rf]] %>%
    filter(P < 0.05) %>%
    select(SNP, chr, bp, A1, BETA, P) %>%
    write_tsv(paste0("../data/processed/gen4/scores/hei_",
                     rf, "_all_sumstats.txt"))
}

lapply(rfs, hei_output_sumstats)
```

```{r score-setup-prod, eval=F}
prod_rfs <- c("ldl", "hdl")

prod_gwis_res_list <- lapply(setNames(prod_rfs, prod_rfs), function(rf) {
  read_tsv(paste0("../data/processed/gen4/fat_", rf, "_prod",
                  "_whi_white.res_annot"))
})

prod_output_sumstats <- function(rf) {
  non_log_rf <- tolower(gsub("log", "", rf))
  
  print(gControl(prod_gwis_res_list[[rf]]$P))
  
  # All CardioGxE SNPs
  prod_gwis_res_list[[rf]] %>%
    filter(P < 0.05) %>%
    filter(SNP %in% all_cardioGxE_snps) %>%
    select(SNP, chr, bp, A1, BETA, P) %>%
    write_tsv(paste0("../data/processed/gen4/scores/fat_",
                     rf, "_prod_full_cge_sumstats.txt"))
  
  # Nominal SNPs
  nominal_snps <- scan(paste0("../data/processed/whi_subsets/",
                              non_log_rf, "_nominal_snps.txt"),
                       what=character())
  prod_gwis_res_list[[rf]] %>%
    filter(P < 0.05) %>%
    filter(SNP %in% nominal_snps) %>%
    select(SNP, chr, bp, A1, BETA, P) %>%
    write_tsv(paste0("../data/processed/gen4/scores/fat_",
                     rf, "_prod_nominalME_sumstats.txt"))
  
  # Suggestive SNPs
  suggestive_snps <- scan(paste0("../data/processed/whi_subsets/",
                                 non_log_rf, "_suggestive_snps.txt"),
                          what=character())
  prod_gwis_res_list[[rf]] %>%
    filter(P < 0.05) %>%
    filter(SNP %in% suggestive_snps) %>%
    select(SNP, chr, bp, A1, BETA, P) %>%
    write_tsv(paste0("../data/processed/gen4/scores/fat_",
                     rf, "_prod_suggestiveME_sumstats.txt"))
  
  # Genome-wide SNPs
  gw_snps <- scan(paste0("../data/processed/whi_subsets/",
                                 non_log_rf, "_gw_snps.txt"),
                          what=character())
  prod_gwis_res_list[[rf]] %>%
    filter(P < 0.05) %>%
    filter(SNP %in% gw_snps) %>%
    select(SNP, chr, bp, A1, BETA, P) %>%
    write_tsv(paste0("../data/processed/gen4/scores/fat_",
                     rf, "_prod_gwME_sumstats.txt"))
  
  # All SNPs
  prod_gwis_res_list[[rf]] %>%
    filter(P < 0.05) %>%
    select(SNP, chr, bp, A1, BETA, P) %>%
    write_tsv(paste0("../data/processed/gen4/scores/fat_",
                     rf, "_prod_all_sumstats.txt"))
}

lapply(prod_rfs, prod_output_sumstats)
```

6) Do these 5 scores work in WHI DM?

Secondary: do they also work for other RFs?

```{r test-scores, message=F}
dm_phenos <- read_delim("../data/processed/gen4/whi_white_DM_phenos.txt", 
                        delim=" ") %>%
  filter(dm_trial == T)

read_profile <- function(filename, df) {
  read_table(filename) %>%
    select(IID, SCORE) %>%
    right_join(dm_phenos, by=c("IID"="SampleID"))
}

scorefile_names <- list(
  single_snp="../data/processed/gen4/scores/single_snp.profile",
  full_cge="../data/processed/gen4/scores/full_cge.profile",
  hdl_nominal="../data/processed/gen4/scores/hdl_nominal.profile",
  hdl_suggestive="../data/processed/gen4/scores/hdl_suggestive.profile",
  all="../data/processed/gen4/scores/all.profile",
  sfa_lit="../data/processed/gen4/scores/sfa_lit.profile",
  hdl_glgc="../data/processed/gen4/scores/hdl_glgc.profile"
)

ordered_snpsets <- setNames(c("all", "nominalME", "suggestiveME", "gwME", "full_cge", 
                              "single_snp", "onlyME"),
                            c("Genome-wide", "Nominal main-effect", "Suggestive main-effect",
                               "Genome-wide sig. main-effect", "CardioGxE SNPs only",
                               "Single top SNP from GWIS (APOA2)", "Main-effect PRS"))

all_scores <- expand.grid(
  rf=rfs, 
  snpset=c("single_snp", "full_cge", "nominalME", "suggestiveME", "gwME", "all", "sfa_lit", "onlyME"), 
  stringsAsFactors=F)

all_scores$statistic <- map2_dbl(all_scores$rf, all_scores$snpset, function(r, s) {
  filename <- paste0("../data/processed/gen4/scores/sfa_pct_binary_", r, "_", s, ".profile")
  if(!file.exists(filename)) return (NA)
  df_with_scores <- read_profile(filename)
  delta_var <- paste0("delta_", tolower(gsub("log", "", r)))
  form <- as.formula(paste(delta_var, "~ SCORE"))
  lm_fit <- lm(form, filter(df_with_scores, dm_intervention == T)) %>%
    tidy() %>%
    filter(term == "SCORE")
  lm_fit$statistic
})

all_scores %>%
  mutate(sig=ifelse(abs(statistic) > 2, "*", "")) %>%
  mutate(snpset=factor(snpset, levels=ordered_snpsets,
                       labels=names(ordered_snpsets))) %>%
  ggplot(aes(x=snpset, y=rf, fill=statistic)) +
  geom_tile() +
  scale_fill_gradient2() +
  geom_text(aes(label=sig)) +
  labs(title="SFA-RF interaction-based scores (SFA as binary % of calories)") +
  theme(axis.text.x=element_text(angle=40, hjust=1))

hei_scores <- expand.grid(
  rf=rfs, 
  snpset=c("full_cge", "nominalME", "suggestiveME", "gwME", "all"), 
  stringsAsFactors=F)

hei_scores$statistic <- map2_dbl(hei_scores$rf, hei_scores$snpset, function(r, s) {
  filename <- paste0("../data/processed/gen4/scores/hei_", r, "_", s, ".profile")
  if(!file.exists(filename)) return (NA)
  df_with_scores <- read_profile(filename)
  delta_var <- paste0("delta_", tolower(gsub("log", "", r)))
  form <- as.formula(paste(delta_var, "~ SCORE"))
  lm_fit <- lm(form, filter(df_with_scores, dm_intervention == T)) %>%
    tidy() %>%
    filter(term == "SCORE")
  lm_fit$statistic
})

hei_scores %>%
  mutate(sig=ifelse(abs(statistic) > 2, "*", "")) %>%
  mutate(snpset=factor(snpset, levels=ordered_snpsets, 
                       labels=names(ordered_snpsets))) %>%
  ggplot(aes(x=snpset, y=rf, fill=statistic)) +
  geom_tile() +
  scale_fill_gradient2() +
  geom_text(aes(label=sig)) +
  labs(title="HEI-RF interaction-based scores (HEI = Healthy Eating Index)") +
  theme(axis.text.x=element_text(angle=40, hjust=1))

fat_pct_binary_scores <- expand.grid(
  rf=rfs, 
  snpset=c("full_cge", "nominalME", "suggestiveME", "gwME", "all"), 
  stringsAsFactors=F)

fat_pct_binary_scores$statistic <- map2_dbl(fat_pct_binary_scores$rf, 
                                            fat_pct_binary_scores$snpset, function(r, s) {
  filename <- paste0("../data/processed/gen4/scores/fat_pct_binary_", r, "_", s, ".profile")
  if(!file.exists(filename)) return (NA)
  df_with_scores <- read_profile(filename)
  delta_var <- paste0("delta_", tolower(gsub("log", "", r)))
  form <- as.formula(paste(delta_var, "~ SCORE + ", gsub("delta", "baseline", delta_var)))
  lm_fit <- lm(form, filter(df_with_scores, dm_intervention == T)) %>%
    tidy() %>%
    filter(term == "SCORE")
  lm_fit$statistic
})

fat_pct_binary_scores %>%
  mutate(sig=ifelse(abs(statistic) > 2, "*", "")) %>%
  mutate(snpset=factor(snpset, levels=ordered_snpsets, 
                       labels=names(ordered_snpsets))) %>%
  ggplot(aes(x=snpset, y=rf, fill=statistic)) +
  geom_tile() +
  scale_fill_gradient2() +
  geom_text(aes(label=sig)) +
  labs(title="Fat-RF interaction-based scores (fat as binary % of calories)") +
  theme(axis.text.x=element_text(angle=40, hjust=1))

fat_prod_scores <- expand.grid(
  rf=c("ldl", "hdl"), 
  snpset=c("full_cge", "nominalME", "suggestiveME", "gwME", "all"), 
  stringsAsFactors=F)

fat_prod_scores$statistic <- map2_dbl(fat_prod_scores$rf, 
                                      fat_prod_scores$snpset, function(r, s) {
  filename <- paste0("../data/processed/gen4/scores/fat_", r, "_prod_", s, ".profile")
  if(!file.exists(filename)) return (NA)
  df_with_scores <- read_profile(filename)
  delta_var <- paste0("delta_", tolower(gsub("log", "", r)))
  form <- as.formula(paste(delta_var, "~ SCORE"))
  # form <- as.formula(paste(delta_var, "~ SCORE + ", gsub("delta", "baseline", delta_var)))
  lm_fit <- lm(form, filter(df_with_scores, dm_intervention == T)) %>%
    tidy() %>%
    filter(term == "SCORE")
  lm_fit$statistic
})

fat_prod_scores %>%
  mutate(sig=ifelse(abs(statistic) > 2, "*", "")) %>%
  mutate(snpset=factor(snpset, levels=ordered_snpsets, 
                       labels=names(ordered_snpsets))) %>%
  ggplot(aes(x=snpset, y=rf, fill=statistic)) +
  geom_tile() +
  scale_fill_gradient2() +
  geom_text(aes(label=sig)) +
  labs(title="Fat-RF 'interaction product method'-based scores (fat as binary % of calories)") +
  theme(axis.text.x=element_text(angle=40, hjust=1))

# map_dfr(a, function(df) {
#   lm(delta_tg ~ SCORE, filter(df, dm_intervention == T, delta_sfa<-5)) %>%
#     tidy() %>%
#     filter(term == "SCORE") %>%
#     select(estimate, p.value)
# }, .id="score_type")
```

# Sanity check

```{r whi-sanity-check, message=F}
apoa5_snps <- unique(filter(cardio_gxe, Gene == "APOA5")$SNP)
apoa5_snps <- apoa5_snps[apoa5_snps %in% names(whi_rs_to_id)]
whi_apoa5_tg <- foreach(snp=whi_rs_to_id[apoa5_snps],
             .combine=rbind) %do%
  run_test(whi_genoData, "logTG", NULL, NULL, robust=F, start=snp, end=snp)
# whi_apoa5_tg %>%
#   inner_join(getAnnotation(whi_snpAnnot), by="snpID") %>%
#   mutate(p.value=format(Wald.pval, scientific=T, digits=3)) %>%
#   select(rsID, p.value) %>%
#   kable(caption="APOA5 SNPs -> TG")

fto_snps <- unique(filter(cardio_gxe, Gene == "FTO")$SNP)
fto_snps <- fto_snps[fto_snps %in% names(whi_rs_to_id)]
whi_fto_bmi <- foreach(snp=whi_rs_to_id[fto_snps],
             .combine=rbind) %do%
  run_test(whi_genoData, "logBMI", NULL, NULL, robust=F, start=snp, end=snp)
# whi_fto_bmi %>%
#   inner_join(getAnnotation(whi_snpAnnot), by="snpID") %>%
#   mutate(p.value=format(Wald.pval, scientific=T, digits=3)) %>%
#   select(rsID, p.value) %>%
#   kable(caption="FTO SNPs -> BMI")

bind_rows(`APOA5 -> TG`=whi_apoa5_tg, `FTO -> BMI`=whi_fto_bmi, 
          .id="SNP_pheno_combo") %>%
  group_by(SNP_pheno_combo) %>%
  summarise(median_pval=median(Wald.pval)) %>%
  kable(caption="Sanity check using known SNP-phenotype pairs",
        longtable=T)

lm(ldl ~ logBMI + logTG + hdl + logGLU + lipid_med, data=whi_phenos) %>%
  tidy() %>%
  select(term, estimate, p.value) %>%
  mutate_at(vars(estimate, p.value), function(x) round(x, 3)) %>%
  kable(caption="Check for internal consistency within biomarkers: LDL associations",
        longtable=T)

lm(ldl ~ sfa_pct + mufa_pct + pufa_pct + tot_cal + age, 
   data=filter(whi_phenos, lipid_med==F)) %>%
  tidy() %>%
  select(term, estimate, p.value) %>%
  mutate_at(vars(estimate, p.value), function(x) round(x, 3)) %>%
  kable(caption="Check for FFQ-biomarker consistency: LDL associations",
        longtable=T)

lm(logTG ~ n3 + fat2carb + tot_cal + age, data=whi_phenos) %>%
  tidy() %>%
  select(term, estimate, p.value) %>%
  mutate_at(vars(estimate, p.value), function(x) round(x, 3)) %>%
  kable(caption="Check for FFQ-biomarker consistency: log(TG) associations",
        longtable=T)

whi_apoa2_snpID <- whi_rs_to_id["rs5082"]
whi_apoa2 <- assocRegression(whi_genoData, "logBMI", covar=c("sfa"), ivar="sfa",
                             snpStart=whi_apoa2_snpID, snpEnd=whi_apoa2_snpID)
whi_apoa2 %>%
  select(GxE.Est, GxE.pval) %>%
  mutate_all(function(x) round(x, 3)) %>%
  kable(caption="SFA * APOA2 -> BMI interaction",
        longtable=T)
```

Everything seems fine here, including expected genotype-biomarker associations.

# Model selection: diet transformations

Use WHI as a "tuning set" to determine the best model/diet transformation to use in the main UK Biobank GWIS analysis. Regressions were run for SFA -> BMI using sets of known gene-diet SNPs for BMI (from CardioGxE) to determine which has the highest mean t-statistic. Covariates are age + total calories.

Conclusion: A binary split (daily SFA above or below the median) appears to result in a modest, but meaningful, increase in mean z-score across the set of SNPs and phenotypes compared to other potential SFA transformations. However, it seems that this conclusion is not consistent across phenotypes (e.g. GxEs for TG appear more strongly using SFA as % kcals, while GxEs for SBP appear more strongly using "raw" SFA).

# Model selection: covariate selection

Having determined a tentative SFA transformation to use (a simple binary split), a similar set of tests were run to determine whether any specific set of dietary covariates produce more GxE replications for the same set of SNPs. "carb_ex" and "pufa_ex" here refer to exchange/substitution models in which all calorie-providing nutrients are adjusted for other than the nutrient to be "substituted".

```{r compare-covars-2, message=F, cache=1, eval=F}
run_whi_test <- function(covars, snpset, outcome, robust=T) {
  res <- foreach(snp=whi_rs_to_id[snpset],
          .combine=rbind) %do%
    run_test(whi_genoData, outcome, 
             covars,
             "sfa_binary", robust=robust, start=snp, end=snp)
  res
}

covar_sets <- list(nodiet = c("age"),
                   tot_cal = c("age", "tot_cal"),
                   carb_ex = c("age", "tot_cal", "mufa", "pufa", "pro", "alc"),
                   pufa_ex = c("age", "tot_cal", "mufa", "cho", "pro", "alc"))

rfs <- c("logBMI", "ldl", "hdl", "logTG", "logSBP", "logGLU")
covar_set_snp_res <- map_dfr(setNames(rfs, rfs), function(rf) {
  map_dfr(covar_sets, function(cs) {
    run_whi_test(c("sfa_binary", cs), sfa_snps, rf)
  }, .id="covar_set")
}, .id="rf")
```

```{r show-compare-covars, eval=F}
covar_set_snp_res_summary <- covar_set_snp_res %>%
  group_by(covar_set, rf) %>%
  summarise(mean_abs_t_stat = mean(abs(GxE.Stat)),
            best_p = min(GxE.pval),
            num_nominal_p = sum(GxE.pval < 0.05)) %>%
  ungroup()
covar_set_snp_res_order <- covar_set_snp_res_summary %>%
  group_by(covar_set) %>%
  summarise(mean_mean = mean(mean_abs_t_stat)) %>%
  arrange(desc(mean_mean))
covar_set_snp_res_summary %>%
  mutate(covar_set = factor(covar_set,
                            levels=covar_set_snp_res_order$covar_set)) %>%
  ggplot(aes(x=rf, y=fct_rev(covar_set), fill=mean_abs_t_stat)) +
  geom_tile() +
  labs(title="Comparison of dietary covariate adjustments using CardioGxE SNPs")
```

```{r compare-covars, message=F, eval=F}
mytest <- function(covars, snpset, outcome, ivar, robust=T) {
  res <- foreach(snp=whi_rs_to_id[snpset],
          .combine=rbind) %do%
    run_test(whi_genoData, outcome, 
             covar=covars,
             ivar=ivar, robust=robust, start=snp, end=snp)
  res
}

covar_sets <- list(tot_cal = c("age", "tot_cal"))

myrfs <- c("delta_bmi", "delta_ldl", "delta_hdl", "delta_tg", "delta_sbp", "delta_glu")
a <- map_dfr(setNames(myrfs, myrfs), function(rf) {
  map_dfr(covar_sets, function(cs) {
    mytest(c("sfa_binary", cs), sfa_snps, rf, "sfa_binary")
  }, .id="covar_set")
}, .id="rf")
asum <- a %>%
  group_by(covar_set, rf) %>%
  summarise(mean_abs_t_stat = mean(abs(GxE.Stat), na.rm=T),
            best_p = min(GxE.pval),
            num_nominal_p = sum(GxE.pval < 0.05)) %>%
  ungroup()
```

```{r load-fhs, message=F, warning=F, eval=F}
fhs_bfile <- "../data/processed/fhs/fhs_hypothesis"
fhs_gds_name <- paste0(fhs_bfile, ".gds")
make_gds(fhs_bfile, fhs_gds_name)
fhs_gds <- GdsGenotypeReader(openfn.gds(fhs_gds_name, allow.fork=T))

fhs_snpAnnot <- bim_to_SnADF(fhs_bfile)
fhs_rs_to_id <- with(getAnnotation(fhs_snpAnnot), setNames(snpID, rsID))

fhs_phenos <- read_delim("../data/processed/gen4/fhs_gwas_phenos.txt",
                         delim=" ") %>%
  mutate(myvar=sfa > 22,
         myvar2=ifelse(sfa < 17, 0, ifelse(sfa > 30, 2, NA)),
         sfa_pct=sfa / tot_cal * 100,
         mufa_pct = mufa / tot_cal * 100,
         pufa_pct = pufa / tot_cal * 100,
         fat2carb = fat / cho,
         # myvar3=sfa_pct > quantile(sfa_pct, 0.5),
         logsfa=log(sfa),
         logsfa_pct=log(sfa_pct))
# mutate(myhei=-scale(palm) + scale(pufa) + scale(n3))
fhs_scanAnnot <- fam_to_ScADF(fhs_bfile, fhs_phenos)
fhs_genoData <- GenotypeData(fhs_gds, snpAnnot=fhs_snpAnnot, 
                             scanAnnot=fhs_scanAnnot)
```

```{r fhs-sanity-check, message=F, eval=F}
apoa5_snps <- unique(filter(cardio_gxe, Gene == "APOA5")$SNP)
apoa5_snps <- apoa5_snps[apoa5_snps %in% names(fhs_rs_to_id)]
fhs_apoa5_tg <- foreach(snp=fhs_rs_to_id[apoa5_snps],
             .combine=rbind) %do%
  run_test(fhs_genoData, "logTG", NULL, NULL, robust=F, start=snp, end=snp)
# kable(select(fhs_apoa5_tg, snpID, Wald.pval), caption="APOA5 -> TG")

fto_snps <- unique(filter(cardio_gxe, Gene == "FTO")$SNP)
fto_snps <- fto_snps[fto_snps %in% names(fhs_rs_to_id)]
fhs_fto_bmi <- foreach(snp=fhs_rs_to_id[fto_snps],
             .combine=rbind) %do%
  run_test(fhs_genoData, "logBMI", NULL, NULL, robust=F, start=snp, end=snp)
# fhs_fto_bmi %>%
#   inner_join(getAnnotation(fhs_snpAnnot), by="snpID") %>%
#   mutate(p.value=format(Wald.pval, scientific=T, digits=3)) %>%
#   select(rsID, p.value) %>%
#   kable(caption="FTO SNPs -> BMI")

bind_rows(`APOA5 -> TG`=fhs_apoa5_tg, `FTO -> BMI`=fhs_fto_bmi, 
          .id="SNP_pheno_combo") %>%
  group_by(SNP_pheno_combo) %>%
  summarise(median_pval=median(Wald.pval)) %>%
  kable(caption="Sanity check using known SNP-phenotype pairs",
        longtable=T)

lm(ldl ~ logBMI + logTG + hdl + logGLU + lipid_med, data=fhs_phenos) %>%
  tidy() %>%
  select(term, estimate, p.value) %>%
  mutate_at(vars(estimate, p.value), function(x) round(x, 3)) %>%
  kable(caption="Check for internal consistency within biomarkers: LDL associations",
        longtable=T)

lm(ldl ~ sfa_pct + mufa_pct + pufa_pct + tot_cal + age, 
   data=filter(fhs_phenos, lipid_med==F)) %>%
  tidy() %>%
  select(term, estimate, p.value) %>%
  mutate_at(vars(estimate, p.value), function(x) round(x, 3)) %>%
  kable(caption="Check for FFQ-biomarker consistency: LDL associations",
        longtable=T)

lm(logTG ~ n3 + fat2carb + tot_cal + age, data=fhs_phenos) %>%
  tidy() %>%
  select(term, estimate, p.value) %>%
  mutate_at(vars(estimate, p.value), function(x) round(x, 3)) %>%
  kable(caption="Check for FFQ-biomarker consistency: log(TG) associations",
        longtable=T)

fhs_apoa2_snpID <- fhs_rs_to_id["rs5082"]
fhs_apoa2 <- assocRegression(fhs_genoData, "logBMI", covar=c("sfa"), ivar="sfa",
                             snpStart=fhs_apoa2_snpID, snpEnd=fhs_apoa2_snpID)
fhs_apoa2 %>%
  select(GxE.Est, GxE.pval) %>%
  mutate_all(function(x) round(x, 3)) %>%
kable(caption="SFA * APOA2 -> BMI interaction", 
      longtable=T)
```