---
output: 
  pdf_document:
    # citation_package: natbib
    keep_tex: true
    fig_caption: true
    latex_engine: pdflatex
  #   # template: ~/Dropbox/miscelanea/svm-r-markdown-templates/svm-latex-ms.tex
#   word_document:
#     fig_caption: true
# always_allow_html: true
title: "A gene-diet interaction-based score predicts response to dietary fat in the Women's Health Initiative"
# thanks: "Replication files are available on the author's Github account..."
# author:
# - name: Steven V. Miller
#   affiliation: Clemson University
# - name: Mary Margaret Albright
#   affiliation: Pendelton State University
# abstract: "This is the abstract..."
# keywords: "pandoc, r markdown, knitr"
# date: "`r format(Sys.time(), '%B %d, %Y')`"
# geometry: margin=1in
# fontfamily: mathpazo
# fontsize: 11pt
# # spacing: double
bibliography: whi_frp_ms.bib
# biblio-style: apsr
# see http://svmiller.com/blog/2016/02/svm-r-markdown-manuscript/ for details on this .Rmd template
---

```{r setup, include=F}
knitr::opts_chunk$set(echo=F, fig.path="figures/")
suppressMessages(silent <- lapply(
  c("knitr", "kableExtra", "tidyverse", "broom", "cowplot",
    "jtools", "ggstance"), 
  library, character.only=T))
library(jtools)
library(ggstance)
```

```{r prep}
bim_to_SnADF <- function(bfile) {
  bim_df <- read_tsv(paste0(bfile, ".bim"), 
                     col_names=c("chromosome", "rsID", "cm", 
                                 "position", "A1", "A2")) %>%
    mutate(snpID=1:nrow(.)) %>%
    mutate_at(vars(chromosome, position), as.integer)
  SnpAnnotationDataFrame(data.frame(bim_df, stringsAsFactors=F))
}

fam_to_ScADF <- function(bfile, phenos) {
  fam_df <- read_delim(paste0(bfile, ".fam"), 
                       delim=" ", col_names=c("FID", "IID", "father", "mother", 
                                               "sex", "pheno")) %>%
    select(IID) %>%
    left_join(phenos, by=c("IID")) %>%
    mutate(scanID=IID)
  ScanAnnotationDataFrame(data.frame(fam_df, stringsAsFactors=F))
}

make_gds <- function(bfile, gds_name, summary=F) {
  snpgdsBED2GDS(paste0(bfile, ".bed"),
                paste0(bfile, ".fam"),
                paste0(bfile, ".bim"),
                gds_name,
                cvt.snpid="int",
                verbose=F)
  if (summary) snpgdsSummary(gds_name)
}

run_test <- function(genoData, outcome, covars, ivar, robust, start, end) {
  assocRegression(
    genoData,
    outcome=outcome,
    model.type="linear",
    gene.action="additive",
    covar=covars,
    ivar=ivar,
    robust=robust,
    snpStart=start, 
    snpEnd=end,
    verbose=F)
}

make_qqplot <- function(p_vec, plotTitle="Title") {
  p_vec <- p_vec[!is.na(p_vec)]
  qqplot(-log10(1:length(p_vec) / length(p_vec)), -log10(p_vec), pch=".", cex=5, 
         main=plotTitle, xlab="Expected (-logP)", ylab="Observed (-logP)")
  abline(0, 1, col="red")
}

gControl <- function(p_vals) {
  # See van Iterson 2017 methods and/or Lehne 2015 code for details on genomic control for EWAS
  # Below is modeled after Lehne 2015
  lambda <- median(qchisq(p_vals, df=1, lower.tail=F), 
                   na.rm=T) / qchisq(0.5, df=1)
  round(lambda, 2)
}

outcome_transforms <- list(
  bmi="logBMI",
  sbp="logSBP",
  hdl="logHDL",
  tg="logTG",
  glu="logGLU",
  ldl="ldl"
)

rfs <- c("bmi", "sbp", "ldl", "hdl", "tg", "glu")
pretty_rfs <- c(bmi="BMI", sbp="SBP", ldl="LDL-C", hdl="HDL-C", tg="TG", glu="FG")
# rf_units <- c(bmi="kg/m^2", sbp="mmHg", ldl="mg/dL", ldl="mg/dL", tg="mg/dL", glu="mg/dL")
```

```{r load-results}
# load("../../output/fat_response_objects.RData")
```

# Abstract

While diet response prediction for cardiometabolic risk factors (CRFs) has been demonstrated using single SNPs and main-effect genetic risk scores, little investigation has gone into the development of genome-wide diet response scores. Here, we leveraged the multi-study setup of the Women's Health Initiative cohort to generate and test genetic scores for response to dietary fat. A genome-wide interaction study was undertaken for each of six CRFs in women not participating in the fat reduction-focused Dietary Modification (DM) trial, and genetic scores based on these analyses were tested for the prediction of one-year CRF changes in DM trial participants. Only one of these genetic scores, for LDL-cholesterol (LDL-C), predicted changes in the associated CRF. This 1760-variant score explained 3.4% of the variance in one-year LDL-C changes in the intervention arm, but was unassociated with changes in the control arm. In contrast, a main-effect genetic risk score for LDL-C was not useful for predicting dietary fat response. Further investigation of this score with respect to downstream disease outcomes revealed suggestive differential associations with stroke subtypes across DM trial arms as well as a potential longevity benefit of an LDL-C-raising score in individuals not lowering their dietary fat. These results lay the foundation for the combination of many genome-wide gene-diet interactions for diet response prediction while highlighting the need for further research in order to achieve robust biomarkers for use in personalized nutrition.

# Introduction

<!--
* Para. 1: GxE central to the hype that has shown promise in particular applications....personalized nutrition!!
    - The emergence of gene-diet interactions (GDI) studies, along with specific well-replicated examples [@Corella___] have provided suggestions that genome-guided personalized nutrition can become a reality.
    - Many GDIs have focused in the cardiometabolic realm. Typically, these focus on cardiometabolic risk factors based on biology-based candidate genes/variants (pick a study from Jose, Lu Qi, and caffeine), but some have looked at actual outcomes (e.g. MI [@Cornelis2006]). Other approaches use main-effect genetic risk scores, such as that for obesity interacting with sugar-sweetened beverage intake to influence anthropometric traits [@Qi2012; @Olsen2016].

* Para. 2: However, to translate GxE into the most effective risk scores, 3 elements will be necessary:
    1. Replicable (find some review or similar that discusses lack of reproducibility and difficulty across populations)
    2. Genome-wide (mention that up until now most investigations are either few-gene hypothesis-based or genome-wide main effect scores)
    3. Causal/applicable to changes in RF based on change in dietary intake
-->

Nutrigenetics approaches (in which genetic information is used to predict response to dietary inputs) are a central component to the emerging promise of personalized nutrition for cardiometabolic risk reduction. Inter-individual differences in food preferences, metabolism, detoxification, excretion, etc. affect our responses to diet, in a similar way to the well-studied field of pharmacogenomics [@Ma2011]. Ideally, genotype-based nutrigenetic investigations would be conducted in large-scale dietary interventions. Two notable examples are the PREDIMED and POUNDS LOST trial, with significant findings including the interaction of a TCF7L2 variant with a Mediterranean diet pattern for glycemic traits [@Corella2013] and the interaction of a PCSK9 variant with dietary carbohydrate for insulin resistance [@Huang2015]. However, these are only able to examine one dietary change (whether food, nutrient, or pattern) at a time, and are often limited to lower sample sizes [@Ordovas2018].

To allow for more flexibility and greater statistical power, gene-diet interactions (GDIs) are more commonly investigated in observational datasets. There is a rich literature of GDI discovery in the cardiometabolic realm. Typically, these focus on cardiometabolic risk factors based on biology-based candidate genes/variants [@Corella2009; @Cuda2011], but some have looked at actual outcomes (e.g. MI [@Cornelis2006]). Other approaches use main-effect genetic risk scores, such as that for obesity interacting with sugar-sweetened beverage intake to influence anthropometric traits [@Qi2012; @Olsen2016].

Characterization of individuals based on single or small groups of single nucleotide polymorphisms (SNPs) may neglect important signal elsewhere in the genome, especially when dealing with highly polygenic cardiometabolic traits. Thus, for effective personalized nutrition approaches to be realized, it will be necessary to integrate signals across the genome. A few investigations have explored GDIs genome-wide, such as for dairy and BMI [@Smith2018] and for various dietary components and colorectal cancer [@Figueiredo2014]. However, genome-wide interaction studies (GWIS) can be problematic due to the lower statistical power inherent in gene-environment interaction analyses [@Dempfle2008]. Furthermore, the potential for confounding and reverse causation (i.e. cardiometabolic risk impacting dietary behavior) in statistical interactions from observational data means that GDIs may not always predict modification of the risk factor in question after a dietary intervention.

In order to provide proof-of-concept for use of GDIs in developing comprehensive diet response genetic scores, we sought to develop a genome-wide, GDI-based dietary fat response score for each of a series of cardiometabolic risk factors (CRFs). We performed genome-wide interaction study in sites showing nominal main effects in prior genome-wide association studies, using these intermediate results to derive fat response scores (FRS) for each CRF. We tested the performance of these scores in the fat reduction-focused Women's Health Initiative Dietary Modification trial, finding that a FRS for LDL-cholesterol (LDL-C) predicts 1-year LDL-C changes selectively in the intervention arm. Furthermore, we found that despite trends in the expected direction, the LDL-C FRS did not predict incident coronary heart disease or other chronic disease outcomes over approximately 20 years of follow-up.

# Results
## Dietary fat responder score development

A series of genome-wide interaction studies (GWIS) were undertaken in women from the Women's Health Initiative studies who did not participate in the dietary modification (DM) trial, using imputed genotypes along with baseline self-reported diet intakes (from food frequency questionnaires) and fasting blood biomarkers. Baseline characteristics of these women, along with those participating in the DM trial, are shown in Table 1.

```{r pop-description, message=F, warning=F}
get_median_iqr <- function(x) {
  if (all(is.na(x))) return (as.character(NA))
  median_fmt <- round(median(x, na.rm=T), 1)
  iqr_borders <- round(quantile(x, c(0.25, 0.75), na.rm=T), 1)
  paste0(median_fmt, " (", iqr_borders[1], "-", iqr_borders[2], ")")
}

# all_whi_phenos <- read_csv("../data/processed/metadata_whi.csv", 
#                            col_types=cols_only(subjID="c", smk_now="d", visit_year="d")) %>%
#   filter(visit_year == 0)
white_nondm_phenos <- read_delim(
  "../data/processed/whi_prediction/whi_white_gwas_phenos.txt", delim=" ") %>%
  # left_join(all_whi_phenos, by="subjID")
  select(age, smk_now, lipid_med, ht_med, dm_med, 
         bmi, sbp, ldl, hdl, tg, glu) %>%
  replace_na(list(smk_now=F, lipid_med=F, ht_med=F, dm_med=F))

white_dm_phenos <- read_delim("../data/processed/whi_prediction/whi_white_DM_phenos_1.txt", 
                        delim=" ") %>%
  filter(dm_trial == T) %>%
  select(age, smk_now, lipid_med, ht_med, dm_med, 
         bmi=baseline_bmi, sbp=baseline_sbp,
         ldl=baseline_ldl, hdl=baseline_hdl, tg=baseline_tg, glu=baseline_glu) %>%
  replace_na(list(smk_now=F, lipid_med=F, ht_med=F, dm_med=F))

pop_tbl <- bind_rows(list(white_dm_phenos, nondm=white_nondm_phenos), 
                     .id="group") %>%
  group_by(group) %>%
  summarise(`Sample size`=n(),
            Age=get_median_iqr(age),
            `Current smoking`=paste0(round(sum(smk_now) / n() * 100), "%"),
            `Lipid-lowering medication`=paste0(round(sum(lipid_med) / n() * 100), "%"),
            `Hypertension medication`=paste0(round(sum(ht_med) / n() * 100), "%"),
            `Diabetes medication`=paste0(round(sum(dm_med) / n() * 100), "%"),
            `Body mass index (BMI; kg/m^2)`=get_median_iqr(bmi),
            `Systolic blood pressure (SBP)`=get_median_iqr(sbp),
            `LDL cholesterol (LDL-C; mg/dL)`=get_median_iqr(ldl),
            `HDL cholesterol (HDL-C; mg/dL)`=get_median_iqr(hdl),
            `Triglycerides (TG; mg/dL)`=get_median_iqr(tg),
            `Fasting glucose (FG; mg/dL)`=get_median_iqr(glu)) %>%
  select(-group) %>%
  t()
colnames(pop_tbl) <- c("WHI DM trial", "WHI non-DM trial")
kable(pop_tbl, booktabs=T, longtable=T, 
      caption="WHI Population Description (European ancestry only)") %>%
  footnote(general=c("* Continuous values shown as: median (interquartile range)"),
           general_title="")
```

```{r power-calcs}
quanto_res <- data.frame(var_expl_GE=c(0.05, 0.1, 0.5, 1),
                         nominal=c(14046, 7021, 1401, 699),
                         suggestive=c(49488, 24737, 4936, 2461),
                         genome_wide=c(70866, 35423, 7069, 3524))
```

Preliminary power calculations were undertaken, based on parameter assumptions including a modest SNP main effect (0.5%) under an additive model and a binary environment with 50% prevalence explaining 10% of the outcome phenotypic variance. The results showed that, at the sample sizes available for European ancestry non-DM participants (7000-10000 individuals for each cardiometabolic risk factor (CRF)), this analysis was powered to detect only moderately large interaction effects (GxE variance explained of greater than approximately 0.5%) at genome-wide significance (Supp. Table S1). Based on these results, a filter for nominal main-effect SNPs (p < 0.05 in existing large-scale GWAS for each CRF) was applied before score development in order to balance the contribution of variants with smaller interaction effect sizes and the inclusion of spurious genetic relationships. 

Dietary fat response scores were generated for each CRF using results from the corresponding GWIS analysis. Linear regression models were fit for log-transformed CRF values (other than LDL-C), incorporating gene-dietary fat interactions (dietary fat represented as a binary % of total calories above or below the median) while adjusting for age, total calories, and five genetic principal components. Q-Q plots of the GWIS results showed that genomic inflation was fairly well-controlled (Supp. Fig. S1). For each CRF, the associated summary statistics (corresponding to the fat-genotype interaction term estimates) were filtered to include only those with nominal main-effect associations in large-scale published GWAS. A pruning-and-thresholding method was used to generate six FRS from these individual sets of summary statistics along with genotypes from the WHI DM participants as a linkage disequilibrium (LD) reference. Using parameters of seed p-value=0.05 and LD r^2^<0.5, six sets of score weights were generated, with relevant SNP set sizes ranging from 1536 (SBP) to 6042 (BMI). Scores were then calculated as the weighted sum of allele counts across SNPs, normalized by the number of non-missing SNPs per individual.

## Dietary fat responder score assessment

```{r test-scores, message=F}
res_dir <- "../data/processed/whi_prediction/"

white_dm_phenos <- read_delim(paste0(res_dir, "whi_white_DM_phenos_1.txt"), delim=" ") %>%
  filter(dm_trial == T)
black_dm_phenos <- read_delim(paste0(res_dir, "whi_black_DM_phenos_1.txt"), delim=" ") %>%
  filter(dm_trial == T)
hispanic_dm_phenos <- read_delim(paste0(res_dir, "whi_hispanic_DM_phenos_1.txt"), delim=" ") %>%
  filter(dm_trial == T)
dm_phenos <- white_dm_phenos

read_profile <- function(filename, df) {
  read_table2(filename) %>%
    select(IID, SCORE, CNT) %>%
    inner_join(dm_phenos, by=c("IID"="SampleID"))
}

get_model_results <- function(form, df, score_term, res_type="statistic") {
  lm_fit <- lm(as.formula(form), data=df) %>%
    tidy() %>%
    filter(term == score_term)
  lm_fit
}

pheno_transforms <- c(bmi="logBMI", sbp="logSBP", ldl="ldl", 
                      hdl="logHDL", tg="logTG", glu="logGLU")
score_snp_sizes <- c(bmi=6042, sbp=1536, ldl=1760, hdl=1731, tg=1774, glu=1924)

test_score <- function(score_rf, test_rf=score_rf, res_type="statistic") {
  scorefile <- paste0(res_dir, "fat_pct_binary_",
                      pheno_transforms[score_rf], "_whi_white_nominalME.profile")
  if(!file.exists(scorefile)) return (NA)
  score_df <- read_profile(scorefile)
  delta_var <- paste0("delta_", test_rf)
  direct_formula <- paste0(delta_var, " ~ SCORE")
  bladj_formula <- paste0(delta_var, " ~ SCORE + baseline_", test_rf)
  interaction_formula <- paste0(delta_var, " ~ SCORE * dm_intervention")
  intervention_df <- filter(score_df, dm_intervention == T)
  get_model_results(direct_formula, intervention_df, "SCORE") %>%
    mutate(n=sum(!is.na(intervention_df[[delta_var]])),
           score_sd=sd(intervention_df$SCORE, na.rm=T),
           rf_sd=sd(intervention_df[[delta_var]], na.rm=T),
           effect_per_sd=estimate * score_sd,
           effect_std=estimate / rf_sd * score_sd,
           n_snps=max(intervention_df$CNT, na.rm=T) / 2)
}

fat_score_res <- tibble(rf=rfs) %>%
  mutate(res=map(rf, test_score)) %>%
  unnest(res)

# fat_score_res %>%
#   ggplot(aes(x=rf, y=effect_std)) +
#   geom_bar(stat="identity")
# 
# fat_score_res %>%
#   ggplot(aes(x=1, y=rf, fill=n)) +
#   geom_tile() +
#   scale_fill_gradient2(name="Sample size with follow-up CRF values (DM trial)", high=scales::muted("red")) +
#   labs(subtitle="Estimated responder score effect (SD CRF change / SD score)") +
#   theme(axis.title=element_blank(), axis.line=element_blank(),
#         axis.ticks=element_blank(), axis.text.x=element_blank(),
#         panel.background=element_blank())

test_score_bladj <- function(score_rf, test_rf=score_rf, res_type="statistic") {
  scorefile <- paste0(res_dir, "fat_pct_binary_",
                      pheno_transforms[score_rf], "_whi_white_nominalME.profile")
  if(!file.exists(scorefile)) return (NA)
  score_df <- read_profile(scorefile)
  delta_var <- paste0("delta_", test_rf)
  bladj_formula <- paste0(delta_var, " ~ SCORE + baseline_", test_rf)
  intervention_df <- filter(score_df, dm_intervention == T)
  get_model_results(bladj_formula, intervention_df, "SCORE") %>%
    mutate(n=sum(!is.na(intervention_df[[delta_var]])),
           score_sd=sd(intervention_df$SCORE, na.rm=T),
           rf_sd=sd(intervention_df[[delta_var]], na.rm=T),
           effect_per_sd=estimate * score_sd,
           effect_std=estimate / rf_sd * score_sd,
           n_snps=max(intervention_df$CNT, na.rm=T) / 2)
}

fat_score_res_bladj <- tibble(rf=rfs) %>%
  mutate(res=map(rf, test_score_bladj)) %>%
  unnest(res) %>%
  mutate(effect_std=round(effect_std, 2),
         p.value=round(p.value, 3),
         n_snps=score_snp_sizes[rf],
         rf=pretty_rfs[rf])


fat_score_res %>%
  select(rf, n_snps, n, effect_std, p.value) %>%
  mutate(effect_std=round(effect_std, 2),
         p.value=round(p.value, 3),
         n_snps=score_snp_sizes[rf],
         rf=pretty_rfs[rf]) %>%
  inner_join(select(fat_score_res_bladj, rf, effect_std, p.value), 
             by="rf", suffix=c(".basic", ".bladj")) %>%
  setNames(c("Risk factor", "# SNPs in risk score", "N", 
             rep(c("Std. effect size", "P-value"), 2))) %>%
  kable(booktabs=T, longtable=T,
        caption="Responder score effects on CRF changes in DM trial participants") %>%
  add_header_above(c(" "=3, "Univariate"=2, "Baseline-adjusted"=2)) %>%
  footnote(general=c("* N = sample size available with 1-year follow-up measurements for the CRF in question",
                     "* Std. effect size represents the regression coefficient estimate in terms of CRF standard deviation per responder score standard deviation"),
           general_title="",
           threeparttable=T)
```

```{r test-scores-cross-ancestry, message=F}
dm_phenos <- bind_rows(white_dm_phenos, black_dm_phenos, hispanic_dm_phenos)

fat_score_res_ca <- tibble(rf=rfs) %>%
  mutate(res=map(rf, test_score)) %>%
  unnest(res)

dm_phenos <- black_dm_phenos

fat_score_res_black <- tibble(rf=rfs) %>%
  mutate(res=map(rf, test_score)) %>%
  unnest(res)

dm_phenos <- hispanic_dm_phenos

fat_score_res_his <- tibble(rf=rfs) %>%
  mutate(res=map(rf, test_score)) %>%
  unnest(res)

dm_phenos <- white_dm_phenos
```

As the scores were developed to predict a positive interaction with dietary fat intake, the expected direction of the FRS effect on risk factors in the present fat-reduction trial would be negative. Of the fat response scores examined, only the LDL-C fat response score (LDL-FRS) was predictive at p<0.05 of the associated CRF change in DM trial participants in the fat-reduction arm (passing a Bonferroni correction for the 6 CRFs tested in baseline-adjusted sensitivity models). For this score, the standardized effect size was `r round(fat_score_res$effect_std[fat_score_res$rf == "LDL-C"], 2)` (corresponding to a `r round(-fat_score_res$effect_per_sd[fat_score_res$rf == "LDL-C"], 2)` mg/dL greater decrease in LDL-C per score standard deviation; p=`r round(fat_score_res$p.value[fat_score_res$rf == "LDL-C"], 3)`). We note that the sample size of European-ancestry DM trial participants with follow-up measurements was much smaller for biochemical variables (n~150) compared to BMI and SBP (n~2000). Using the score developed in European-ancestry individuals, score performance was then tested in a combined-ancestry group including Black and Hispanic individuals, which almost doubled the sample size (Supp. Table S2). While some biomarkers showed strong relationships (e.g. SBP), the signs of many were in a counter-intuitive direction, including a flip in sign for the previously-strong LDL-C relationship, suggesting that these results reflect primarily differences in ancestry rather than the intended biological differences. This observation was reinforced by the lack of association of the score with CRF changes in either Blacks or Hispanics alone.

```{r test-ldl-scores-other-rfs, message=F}
ldl_score_res <- tibble(test_rf=rfs) %>%
  mutate(res=map(test_rf, function(trf) test_score("ldl", test_rf=trf))) %>%
  unnest(res)
```

```{r ldl-score-investigation, message=F}
ldl_score_df <- read_profile(paste0(res_dir, "fat_pct_binary_ldl_whi_white_nominalME.profile")) %>%
  mutate(fat_down=delta_fat < 0)

ldl_basic_lm_fit <- lm(delta_ldl ~ SCORE, data=ldl_score_df, 
                       subset=ldl_score_df$dm_intervention == T)
ldl_blonly_lm_fit <- lm(delta_ldl ~ baseline_ldl, data=ldl_score_df, 
                       subset=ldl_score_df$dm_intervention == T)
ldl_bladj_lm_fit <- lm(delta_ldl ~ SCORE + baseline_ldl, data=ldl_score_df, 
                       subset=ldl_score_df$dm_intervention == T)
ldl_interaction_lm_fit <- lm(delta_ldl ~ SCORE * dm_intervention, 
                             data=ldl_score_df)

ldl_score_hist <- ggplot(ldl_score_df, aes(x=SCORE)) +
  geom_histogram() +
  labs(x="LDL-FRS genetic score (arbitrary units)",
       y="Participant count")
ldl_score_weights <- read_tsv(paste0(res_dir, "fat_pct_binary_ldl_whi_white_nominalME.weights")) %>%
  arrange(desc(abs(BETA)))

annovar_output <- read_tsv(paste0(res_dir, "fat_pct_binary_ldl_whi_white_nominalME.avinput.variant_function"),
                           col_names=c("Type", "Gene", "CHR", "POS", "POS2", "REF", "ALT", "SNP", "BETA")) %>%
  arrange(desc(abs(BETA))) %>%
  mutate(Gene=gsub("\\(dist=[0-9]*\\)|\\(dist=NONE\\)", "", Gene),
         Gene=gsub(",", "/", Gene))

ldl_score_weights_hist <- annovar_output %>%
  ggplot(aes(x=log(abs(BETA)))) +
  geom_histogram(bins=50) +
  labs(x="Log-transformed abs. value of SNP weight",
       y="SNP count") +
  coord_cartesian(xlim=c(1, 6))
for (i in 1:4) {
  ldl_score_weights_hist <- ldl_score_weights_hist +
    annotate(geom="text", 
             x=log(abs(annovar_output$BETA[i])), y=20, size=3,
             label=paste0(annovar_output$SNP[i], " (", 
                          annovar_output$Gene[i], ")"),
             angle=75, hjust=0.1)
}

annovar_type_summary <- annovar_output %>%
  group_by(Type) %>%
  summarise(n=n()) %>% 
  arrange(desc(n)) %>%
  mutate(Type=ifelse(grepl("^nc", Type), Type, str_to_title(Type)),
         Type=factor(Type, levels=Type))
annovar_type_plt <- annovar_type_summary %>%
  ggplot(aes(x=fct_rev(Type), y=n)) +
  geom_bar(stat="identity") +
  labs(x="",
       y="SNP count") +
  coord_flip()

annovar_gene_summary <- annovar_output %>%
  separate_rows(Gene, sep="/") %>%
  group_by(Gene) %>%
  summarise(n=n(),
            total_weight=sum(abs(BETA))) %>%
  # arrange(desc(total_weight)) %>%
  arrange(desc(n)) %>%
  filter(Gene != "NONE") %>%
  mutate(Gene=factor(Gene, levels=Gene))
annovar_gene_plt <- annovar_gene_summary %>%
  # slice(1:20) %>%
  filter(n >= 5) %>%
  ggplot(aes(x=fct_rev(Gene), y=n)) +
  geom_bar(stat="identity") +
  scale_y_continuous(breaks=c(0, 5, 10)) +
  labs(x="Annotated gene",
       y="# of annotated SNPs in LDL-FRS") +
  coord_flip()
```

Based on its observed association in European-ancestry participants, the LDL-FRS was investigated further. Linear models showed that the LDL-FRS accounted for `r round(summary(ldl_basic_lm_fit)$r.squared * 100, 1)`% of the variance in 1-year LDL-C changes in the DM intervention arm. In baseline-adjusted models, this figure rose slightly to `r round((summary(ldl_bladj_lm_fit)$r.squared - summary(ldl_blonly_lm_fit)$r.squared) * 100, 1)`, based on the R^2^ change from a baseline-only model. Additional models confirmed an interaction between DM trial arm and the LDL-FRS (p=0.011), supporting the specificity of this score for the fat-reduction arm. The LDL-FRS also showed specificity for LDL-C in that it did not predict changes in any other CRF (Supp. Table S3). The 1760 component SNPs were annotated using Annovar [@Wang2010], revealing a predominance of intergenic and intronic variants and a set of genes with high numbers of independent SNPs contributing to the score (Fig. 1a-d). Top genes by number of contributing SNPs included CSMD1, PTPRD, and RGS12. <!--Genes at these loci included HSF2/SERINC1, TSN/LINC01826, BBS9, and DARS-AS1/CXCR4.-->

```{r ldl-me-score, message=F}
ldl_ME_score_df <- read_profile(paste0(res_dir, "ldl_ME_nominalME.profile")) %>%
  mutate(fat_down=delta_fat < 0)

ldl_ME_BL_fit <- lm(baseline_ldl ~ SCORE, data=ldl_ME_score_df)
ldl_ME_fit <- lm(delta_ldl ~ SCORE, data=ldl_ME_score_df, subset=ldl_ME_score_df$dm_intervention == T)
```

```{r ldl-score-strata, message=F}
ldl_by_randomization <- ldl_score_df %>%
  filter(!is.na(delta_ldl)) %>%
  mutate(score_quantile=cut(SCORE, quantile(SCORE, seq(0, 1, length.out=4)),
                            include.lowest=T, labels=paste0("T", 1:3))) %>%
         # dm_intervention=factor(dm_intervention, labels=c("Control Arm", "Fat-reduction Arm"))) %>%
  group_by(dm_intervention, score_quantile) %>%
  summarise(mean_change=mean(delta_ldl),
            n=n(),
            sem=sd(delta_ldl) / sqrt(n))
lbr_labels <- with(ldl_by_randomization,
                   c("TRUE"=paste0("DM Intervention (n=", sum(n[dm_intervention]), ")"),
                     "FALSE"=paste0("DM Control (n=", sum(n[!dm_intervention]), ")")))
ldl_by_randomization_plt <- ldl_by_randomization %>%
  ggplot(aes(x=score_quantile, y=mean_change)) +
  geom_point() +
  geom_errorbar(aes(ymin=mean_change - sem, ymax=mean_change + sem), width=0) +
  coord_cartesian(ylim=c(-30, -10)) + 
  labs(x="LDL-FRS tertile",
       y=expression(Delta*"LDL (baseline to year 1)")) +
  facet_wrap(vars(dm_intervention), labeller=labeller(dm_intervention=lbr_labels))

ldl_ME_by_randomization <- ldl_ME_score_df %>%
  filter(!is.na(delta_ldl)) %>%
  mutate(score_quantile=cut(SCORE, quantile(SCORE, seq(0, 1, length.out=4)),
                            include.lowest=T, labels=paste0("T", 1:3))) %>%
         # dm_intervention=factor(dm_intervention, labels=c("Control Arm", "Fat-reduction Arm"))) %>%
  group_by(dm_intervention, score_quantile) %>%
  summarise(mean_change=mean(delta_ldl),
            n=n(),
            sem=sd(delta_ldl) / sqrt(n))
lbr_ME_labels <- with(ldl_ME_by_randomization,
                   c("TRUE"=paste0("DM Intervention (n=", sum(n[dm_intervention]), ")"),
                     "FALSE"=paste0("DM Control (n=", sum(n[!dm_intervention]), ")")))
ldl_ME_by_randomization_plt <- ldl_ME_by_randomization %>%
  ggplot(aes(x=score_quantile, y=mean_change)) +
  geom_point() +
  geom_errorbar(aes(ymin=mean_change - sem, ymax=mean_change + sem), width=0) +
  coord_cartesian(ylim=c(-30, -10)) + 
  labs(x="LDL-C main-effect GRS tertile",
       y=expression(Delta*"LDL (baseline to year 1)")) +
  facet_wrap(vars(dm_intervention), labeller=labeller(dm_intervention=lbr_ME_labels))

ldl_by_diet <- ldl_score_df %>%
  filter(!is.na(delta_ldl)) %>%
  mutate(score_quantile=cut(SCORE, quantile(SCORE, seq(0, 1, length.out=4)),
                            include.lowest=T, labels=paste0("T", 1:3)),
         fat_down=cut(delta_fat, breaks=c(-1000, -20, 0, 1000))) %>%
         # dm_intervention=factor(dm_intervention, labels=c("Control Arm", "Fat-reduction Arm"))) %>%
  group_by(fat_down, score_quantile) %>%
  summarise(mean_change=mean(delta_ldl),
            n=n(),
            sem=sd(delta_ldl) / sqrt(n))
ldl_by_diet_plt <- ldl_by_diet %>%
  ggplot(aes(x=score_quantile, y=mean_change)) +
  geom_point() +
  geom_errorbar(aes(ymin=mean_change - sem, ymax=mean_change + sem), width=0) +
  # scale_x_discrete(labels=paste0("Tertile", 1:3)) + 
  labs(x="",
       y=expression(Delta*"LDL (baseline to year 1)")) +
  facet_wrap(vars(fat_down))

ldl_per_protocol <- ldl_score_df %>%
  filter(!is.na(delta_ldl)) %>%
  mutate(score_quantile=cut(SCORE, quantile(SCORE, seq(0, 1, length.out=4)),
                            include.lowest=T, labels=paste0("T", 1:3))) %>%
  # group_by(fat_down, score_quantile) %>%
  group_by(dm_intervention, fat_down, score_quantile) %>%
  summarise(mean_change=mean(delta_ldl),
            n=n(),
            sem=sd(delta_ldl) / sqrt(n)) %>%
  filter(dm_intervention == fat_down) %>%
  ungroup()
  # mutate(fat_down=factor(fat_down, labels=c("Fat Increased", "Fat Decreased")))
lpp_labels <- with(ldl_per_protocol,
                   c("TRUE"=paste0("DM Intervention (n=", sum(n[fat_down]), ")"),
                     "FALSE"=paste0("DM Control (n=", sum(n[!fat_down]), ")")))
ldl_per_protocol_plt <- ldl_per_protocol %>%
  ggplot(aes(x=score_quantile, y=mean_change)) +
  geom_point() +
  geom_errorbar(aes(ymin=mean_change - sem, ymax=mean_change + sem), width=0) +
  # scale_x_discrete(labels=paste0("Tertile", 1:3)) + 
  labs(x="LDL responder score tertile",
       y=expression(Delta*"LDL (baseline to year 1)")) +
  facet_wrap(vars(fat_down), labeller=labeller(fat_down=lpp_labels))
```

```{r ldl-score-characterization, message=F, fig.asp=1.1, fig.cap="LDL-FRS characterization. a) Distribution of LDL-FRS in WHI DM trial participants. b) Distribution of SNP weights constituting the LDL-FRS (shown as the natural log-transformed absolute values of the true weights). c) SNP counts in different loci types for LDL-FRS consituent SNPs. d) Genes are summarized by the number of annotated SNPs in the LDL-FRS (genes with at least 5 component SNPs are shown). e-f) 1-year changes in LDL-C in DM trial participants as a function of genetic scores. Mean changes in LDL-C (y-axis) are shown as a function of either LDL-FRS (e) or LDL-GRS (f) tertile (x-axis). Error bars represent standard errors. LDL-FRS: LDL-C fat response genetic score, LDL-GRS: LDL-C main-effect genetic score."}
upper_panel <- plot_grid(ldl_score_hist, ldl_score_weights_hist, 
                         annovar_type_plt, nrow=1, labels=c("a", "b", "c"))
lower_right_panel <- plot_grid(ldl_by_randomization_plt, ldl_ME_by_randomization_plt, 
                               nrow=2, align="v", labels=c("e", "f"))
lower_panel <- plot_grid(annovar_gene_plt, lower_right_panel,
                         nrow=1, rel_widths=c(1, 2), labels=c("d", ""))
plot_grid(upper_panel, lower_panel, nrow=2, rel_heights=c(1, 2))
```

Differences in mean LDL-C changes during the DM trial across genetic score strata are shown in Fig. 1e,f. As suggested by the regression results, those in the control arm trended towards less strong LDL-C reductions in higher LDL-FRS strata, while those in the fat-reduction arm showed the opposite trend. For comparison to the FRS, a main-effect genetic risk score (GRS) for LDL-C was developed using summary statistics from the Global Lipid Genetics Consortium meta-analysis [@Willer2013] and an identical pruning-and-thresholding procedure to that used for the GDI-based scores. As expected, this score was strongly predictive of baseline LDL-C concentrations (p=`r format(tidy(ldl_ME_BL_fit)$p.value[2], digits=3, scientific=T)`). However, unlike the GDI-based score, the GRS did not predict LDL-C changes in the DM intervention group (p=`r round(tidy(ldl_ME_fit)$p.value[2], 2)`; stratum-specific mean changes in Fig. 1f).

## LDL-FRS association with chronic disease outcomes

```{r cvd-outcomes, message=F, fig.asp=0.9, fig.cap="FRS prediction of chronic disease development. X-axis shows log(hazard ratio) estimates from Cox proportional hazards regression for a) coronary heart disease, b) ischemic stroke, c) hemorrhagic stroke, d) cancer (any type), and e) death from all causes. Separate estimates are shown for DM trial intervention arm, control arm, and the same strata filtered for FFQ-reported fat reduction or increase, respectively. Error bars represent 95% confidence interval estimates for the regression coefficients."}
ldl_score_df <- read_profile(paste0(res_dir, "fat_pct_binary_ldl_whi_white_nominalME.profile")) %>%
  mutate(fat_down=delta_fat < 0)
# dm_phenos <- read_delim(paste0(res_dir, "whi_white_DM_phenos_6.txt"), delim=" ") %>%
#   filter(dm_trial == T)
# ldl_score_df <- read_profile(paste0(res_dir, "fat_pct_binary_ldl_whi_white_nominalME.profile")) %>%
#   mutate(fat_down=delta_fat < 0)
# dm_phenos <- read_delim(paste0(res_dir, "whi_white_DM_phenos_1.txt"), delim=" ") %>%
#   filter(dm_trial == T)

outcomes <- read_csv("../data/processed/outcomes_whi.csv")

ldl_score_df_with_outcomes <- inner_join(ldl_score_df, outcomes, by="subjID")

ldl_score_df_subsets <- list(dm_int=filter(ldl_score_df_with_outcomes, dm_intervention == T),
                             dm_ctrl=filter(ldl_score_df_with_outcomes, dm_intervention == F),
                             fat_down=filter(ldl_score_df_with_outcomes, fat_down == T, dm_intervention == T),
                             fat_up=filter(ldl_score_df_with_outcomes, fat_down == F, dm_intervention == F)
                             # fat_down=filter(ldl_score_df_with_outcomes, fat_down == T),
                             # fat_up=filter(ldl_score_df_with_outcomes, fat_down == F)
                             )
subset_sample_sizes <- lapply(ldl_score_df_subsets, nrow)

library(survival)
outcome_names <- c(chd="Coronary Heart Disease", 
                   #mi="Myocardial Infarction", 
                   isch_stroke="Ischemic Stroke", hemo_stroke="Hemorrhagic Stroke", 
                   cancer="All Cancers", death="Death")
disease_fits <- lapply(names(outcome_names),
                       function(disease) {
  fits <- lapply(ldl_score_df_subsets, function(df) {
    # glm(formula=as.formula(paste(disease, "~ SCORE + age + smk_now + baseline_bmi + baseline_ldl + baseline_glu + baseline_tg + baseline_hdl + baseline_sbp")),
    #     data=df, family="binomial")
    surv_obj <- Surv(df[[paste0("time_to_", disease)]], df[[disease]])
    coxph(surv_obj ~ SCORE + age, data=df)
  })
  fits
})
names(disease_fits) <- names(outcome_names)

disease_plots <- lapply(names(outcome_names), function(disease) {
  fits <- disease_fits[[disease]]
  plot_summs(fits[[1]], fits[[2]], fits[[3]], fits[[4]],
             model.names=paste0(c("DM Intervention", "DM Control", 
                                  "DM Intervention & Fat Reduced", 
                                  "DM Control & Fat Increased"),
                                "\n(n=", subset_sample_sizes, ")"),
             coefs=c("SCORE"),
             scale=T) +
    labs(x="log(HR) of LDL-FRS",
         # x=paste("log(HR) of LDL-FRS for", outcome_names[disease]),
         y="",
         title=outcome_names[disease]) +
    theme(axis.text.y=element_blank(),
          legend.title=element_blank(),
          plot.title=element_text(hjust=0.5, size=12))
})
names(disease_plots) <- names(outcome_names)

legend <- get_legend(disease_plots[[1]] +
                       theme(legend.text=element_text(size=11),
                             legend.key.size=unit(2, "line")))
disease_plots <- lapply(disease_plots, 
                        function(p) p + 
                          # coord_cartesian(xlim=c(-6, 30)) +
                          theme(legend.position="none"))
with(disease_plots, {
  plot_grid(chd, isch_stroke, hemo_stroke, 
            cancer, death, legend,
            ncol=2, labels=c("a", "b", "c", "d", "e")) 
})

# mi_strat_fit <- coxph(Surv(time_to_hemo_stroke, hemo_stroke) ~ SCORE, 
#                       # data=ldl_score_df_with_outcomes)
#                       data=filter(ldl_score_df_with_outcomes, dm_intervention == T, dm_intervention == fat_down))
#                       # subset=ldl_score_df_with_outcomes$dm)
# tidy(mi_strat_fit)
```

```{r old-outcomes-stuff, eval=F}
# chd_fits <- lapply(ldl_score_df_subsets, function(df) {
#     glm(formula=chd ~ SCORE + age,
#         data=df, family="binomial")
#   })
# plot_summs(chd_fits[[1]], chd_fits[[2]], chd_fits[[3]], chd_fits[[4]],
#            model.names=c("DM Intervention", "DM Control", "DM Intervention &\nFat reduced", 
#                          "DM Intervention &\nFat increased"),
#            coefs=c("SCORE"),
#            scale=T) +
#   labs(x="Logistic regression: log(OR) of HDL responder score for CHD",
#        y="",
#        title="Logistic regression for incident CHD in WHI DM participants") +
#   theme(legend.title=element_blank())
# stroke_fits <- lapply(ldl_score_df_subsets, function(df) {
#     glm(formula=stroke ~ SCORE + age,
#         data=df, family="binomial")
#   })
# plot_summs(stroke_fits[[1]], stroke_fits[[2]], stroke_fits[[3]], stroke_fits[[4]],
#            model.names=c("DM Intervention", "DM Control", "DM Intervention &\nFat reduced", 
#                          "DM Intervention &\nFat increased"),
#            coefs=c("SCORE"),
#            scale=T) +
#   labs(x="Logistic regression: log(OR) of HDL responder score for CHD",
#        y="",
#        title="Logistic regression for incident CHD in WHI DM participants") +
#   theme(legend.title=element_blank())
# death_fits <- lapply(ldl_score_df_subsets, function(df) {
#     glm(formula=death ~ SCORE + age,
#         data=df, family="binomial")
#   })
# plot_summs(death_fits[[1]], death_fits[[2]], death_fits[[3]], death_fits[[4]],
#            model.names=c("DM Intervention", "DM Control", "DM Intervention &\nFat reduced", 
#                          "DM Intervention &\nFat increased"),
#            coefs=c("SCORE"),
#            scale=T) +
#   labs(x="Logistic regression: log(OR) of HDL responder score for CHD",
#        y="",
#        title="Logistic regression for incident CHD in WHI DM participants") +
#   theme(legend.title=element_blank())
```

Next, the LDL-FRS was tested for relationships with incident disease outcomes using Cox proportional hazards models adjusted for age at baseline (Fig. 2). In addition to intervention versus control arm, another set of "per protocol-like" strata were produced by additionally filtering for FFQ-based self-reported fat reduction (in the intervention group) or fat increase (in the control group). CHD and ischemic stroke appeared visually to demonstrate expected interaction patterns (an inverse association between LDL-FRS and disease risk only in the fat reduction group), but these estimates contained a high level of variability. Results for cancer were equivocal and did not vary across groups. Hemorrhagic stroke showed significant positive associations with the LDL-FRS only in the fat reduction group. Additionally, the score associated with a decrease in time-to-death in fat-increasing strata, those most likely to have undergone a sustained increase in dietary fat over time.
<!--While most relationships were equivocal (stratum-specific and comparisons across strata), a few stood out. 
Using the second ("per protocol-like") stratification, the LDL-FRS associated with a reduction in MI risk in those reducing fat versus an increase in MI risk in those increasing fat (p for interaction = 0.078).--> 

```{r cvd-outcomes-experimental, include=F, message=F, eval=F}
# sample_ids <- read_tsv("../data/raw/whi/gen/imputedSampleInfo.txt", skip=15) %>%
#   rename(subjID=SubjectID) %>%
#   select(SampleID, subjID)

outcomes <- read_csv("../data/processed/outcomes_whi.csv")

hdl_score_df_with_outcomes <- inner_join(hdl_score_df, outcomes, by="subjID")
bmi_score_df_with_outcomes <- inner_join(bmi_score_df, outcomes, by="subjID")
sbp_score_df_with_outcomes <- inner_join(read_profile("../data/processed/whi_predictions_hardcalls/fat_pct_binary_sbp.profile"),
                                         outcomes, by="subjID")
glu_score_df_with_outcomes <- inner_join(read_profile("../data/processed/whi_predictions_hardcalls/fat_pct_binary_glu.profile"),
                                         outcomes, by="subjID")
tg_score_df_with_outcomes <- inner_join(read_profile("../data/processed/whi_predictions_hardcalls/fat_pct_binary_tg.profile"),
                                         outcomes, by="subjID")
ldl_score_df_with_outcomes <- inner_join(read_profile("../data/processed/whi_predictions_hardcalls/fat_pct_binary_ldl.profile"),
                                         outcomes, by="subjID")

# bmi_score_df_with_outcomes %>%
#   mutate(fat_down=delta_fat < 0) %>%
#   select(dm_intervention, fat_down, chd, SCORE) %>%
#   na.omit() %>%
#   mutate(score_quartile=cut(SCORE, breaks=quantile(SCORE, seq(0, 1, length.out=4), na.rm=T),
#                             labels=paste0("Q", 1:3), include.lowest=T)) %>%
#   group_by(score_quartile, dm_intervention) %>%
#   summarise(mean=sum(chd) / n(),
#             # sem=sd(delta_hdl) / sqrt(n()),
#             n=n()) %>%
#   ggplot(aes(x=score_quartile, y=mean)) +
#   geom_point() +
#   # geom_errorbar(aes(ymin=mean - sem, ymax=mean + sem), width=0) +
#   labs(x="", y="delta HDL") +
#   facet_wrap(vars(dm_intervention), labeller=labeller(.cols=label_both))

chd_glm <- bmi_score_df_with_outcomes %>%
  filter(dm_intervention == T) %>%
  glm(formula=chd ~ SCORE + age + baseline_bmi + baseline_hdl + baseline_ldl + baseline_sbp, family="binomial")
chd_glm_control <- bmi_score_df_with_outcomes %>%
  filter(dm_intervention == F) %>%
  glm(formula=chd ~ SCORE + age + baseline_bmi + baseline_hdl + baseline_ldl + baseline_sbp, family="binomial")

plot_summs(chd_glm, chd_glm_control,
           model.names=c("DM Intervention", "DM Control"),
           scale=T) +
  geom_text(x=0, y=6.4, label="Predicts lower bmi w/ fat reduction") +
  labs(x="Logistic regression: effect on CHD",
       title="Same, using fat-BMI responder score...")
```

# Discussion

Diet response scores have shown some success at predicting the response of cardiovascular risk factors (CRFs) to nutritional interventions, but they are often based solely on main effects or single GDI SNPs. Here, we explored the potential for gene diet interaction (GDI)-based diet responder score development, leveraging the multi-trial setup of the Women's Health Initiative. We developed what to our knowledge is the first example of a diet response score based on a hypothesis-free genome scan for each of six risk factors, and showed preliminary evidence for the viability of a LDL-C fat response score.

Though FRS for six CRFs were developed and tested, only that for LDL-C showed nominal significance in predicting 1-year changes in the corresponding CRF. Multiple factors could explain this lack of predictive performance in general. First, analysis in observational datasets carries with it the potential for confounding of the observed relationships. Second, FFQs are imprecise instruments for measuring dietary intake, and though the FFQs used in the present study were optimized for detection of dietary fat, there was potential for substantial misclassification of this environmental exposure. Third, power calculations shown here and elsewhere suggest that a cohort of this size may not be powered to detect many very small gene-environment interactions such as are sought in genome-wide approaches like the one used here.

CSMD1, PTPRD, and RGS12 stood out as genes containing the highest number of SNPs in the LDL-FRS (11, 9, and 9, respectively). CSMD1 variants are notably associated with LDL-C response to statin treatment [@Thompson2009] as well as SBP response to a high-salt diet [Newton-Cheh2009]. CSMD1 has also shown epigenetic associations with LDL-C [@Bell2012] as well as response to modification of dietary fat composition [@Perfilyev2017]. PTPRD variants modulate the response of T2D patients to pioglitazone therapy [@Pei2013] and show suggestive associations with eating behaviors (caloric intake at dinner) [@Comuzzie2012]. RGS12 has been linked to LDL-C in GWAS [@Spracklen2017]. Altogether, these genes have literature evidence for relationships to dietary intake, response to cardiometabolic therapies, and LDL-C, but have not until now been shown to directly modify the LDL-C response to dietary fat proportions. We note that there is a bias towards identifying LDL-C-related variants in the LDL-FRS, as only nominally-associated main-effect SNPs were used as input to the score development algorithm.

A reasonable body of literature exists establishing GDIs for both dietary fat on CRFs [@Cuda2012; @Zheng2015] and general dietary exposures on LDL-C [@Ordovas2011]. Multiple studies have looked specifically at genetic variants modulating the LDL-C response to dietary fat. For example, a caloric restriction intervention in type 2 diabetics was more effective in reducing LDL-C in APOE4 carriers (-15.6% versus -0.7%) [@Saito2004]. Two studies using the POUNDS LOST trial found relationships of specific polymorphisms with 2-year changes in response to a dietary fat intervention. Carriers of a risk allele at the APOA5 variant rs964184 show a decrease of 7.5 mg/dL in LDL-C in low-fat but not high-fat interventions [@Zhang2012], and carriers of the minor allele at CETP rs3764261 show an 8.9 mg/dL greater decrease in LDL-C on a low-fat diet [@Xu2015]. Our observed effect size of a 5.4 mg/dL decrease in LDL-C is of a similar magnitude to these findings, and emerged despite the multi-factorial nature of the WHI DM trial (which incorporated additional nutrition recommendations). The observed variance explained of 3.4% for the LDL-FRS means that the score, while contributing meaningfully to the prediction, does not capture most of the interindividual variability in LDL-C response to the WHI DM trial intervention.

There has been interest in the past in using main-effect genetic risk scores (GRS) as genetic variables in order to improve statistical power to detect gene-environment interactions [@Aschard2016]. Such interactions may be viewed from a lens in which genetic risk corresponds to a predisposition that is only triggered in certain environments (e.g. dietary behaviors). Here, we observed only a minor association of a main-effect GRS for LDL-C with greater LDL-C reductions in the DM trial (p=`r round(tidy(ldl_ME_BL_fit)$p.value[2], 2)`). This trend runs counter to a prior observation of greater lifestyle intervention effectiveness for LDL-C reduction in those with low genetic risk of hyperlipidemia [@Zubair2019]. This discrepancy may be due to differences between the DM trial and the personalized diet and lifestyle changes recommended in the intervention in question. Regardless, the meaningful increase in predictive power of the GDI-based FRS compared to the main-effect GRS for LDL-C indicates the value in using interaction-based scores rather than simple genetic predispositions for the development of personalized dietary recommendations.

A diet response score such as that developed here is most useful if its value extends beyond just risk factor changes and predicts downstream changes in chronic disease and mortality risk. A modest interaction for CHD and ischemic stroke was visually apparent across strata (Fig. 2a,b), corresponding to a decreased CHD risk in fat-reduction participants whose LDL-C would be expected to drop more prominently according to the score. In contrast, hemorrhagic stroke showed the opposite trend, with a positive score-disease relationship only in the fat reduction group. This result is in line with existing evidence for the detrimental effects of low LDL-C on hemorrhagic stroke risk [@Sun2019]. Cancer showed no major patterns, which could be expected due to the equivocal associations of cancer with lipids [@Koene2016]. Somewhat surprisingly, we observed that a fat-response score predisposing individuals to a positive fat-LDL-C correlation is inversely associated with risk of death most strongly control individuals with self-reported fat increases, whose LDL-C would be expected to rise. However, it is well-appreciated that LDL-C associates inversely with survival in older populations [@Ravnskov2016] and this inverse relationship in the control group was not notably different than that in the intervention group. We note that all disease outcome relationships assessed here are subject to the major caveat that dietary evolution and decreased adherence likely developed over time in many subjects, diluting the utility of the randomization and 1-year changes used for stratification in Fig. 2.

The present study had the advantage of developing a GDI-based dietary fat score in almost 10,000 women and testing in a dietary intervention trial from the same population. However, the sample size did not afford statistical power to scan genome-wide for GDIs, and the smaller fractions of alternate ancestries in this population made development of ancestry-specific response scores unrealistic. Additionally, the DM trial intervention in which the scores were tested may not exactly match the intervention relevant to the purely fat reduction-focused score developed here; it included additional non-fat-related dietary recommendations that may have effected the interactions examined here, and did not ultimately achieve its intended 20% fat reduction. Finally, this study only examined women, despite the fact that CRF profiles and their genetic trait architectures are known to vary across sexes.

In summary, we present a method for the development of diet response scores based on genome-wide gene-diet interaction study summary statistics. While the resulting dietary fat response scores are not all informative, a score focused on LDL-C is predictive of 1-year LDL-C changes during a fat reduction trial. Its performance is risk factor-specific, and is superior to that of a main-effect genetic risk score for LDL-C. Furthermore, it displays suggestive relationships with chronic disease outcomes, adding a new perspective to known discrepancies between associations of LDL-C with cardiovascular disease, stroke subtypes, and all-cause mortality. These findings support the utility of gene-diet interactions for personalized nutrition while highlighting the need for increased sample sizes and improved diet measures for the discovery of robust genetic predictors of diet response.


<!--
Old:
Recently, a 14-SNP score designed based on prior GDI literature was shown to predict differences in HDL-C response to a saturated fat-reduction trial in overweight adults (-6 mg/dL average change in those with high versus low response scores) [@Guevara-Cruz2019]. ____ link to our results 

A reasonable body of literature exists establishing GDIs for both dietary fat on CRFs [@Cuda2012; @Zheng2015] and general dietary exposures on HDL-C [@Ordovas2011]. A few studies have looked specifically at genetic variants modulating the HDL-C response to dietary fat. Cross-sectionally in the Framingham Heart Study, the -514(C/T) LIPC polymorphism associates with a 16% greater HDL-C in those with two minor alleles and high dietary fat intake, versus a 10% lower HDL-C in those with two minor alleles and low dietary fat intake [@Ordovas2002]. In addition, two studies using the POUNDS LOST trial find relationships of specific polymorphisms with 2-year changes in response to a dietary fat intervention. Carriers of a risk allele at the APOA5 variant rs964184 show an approximately 3.5 mg/dL greater high-fat diet-related increase in HDL-C [@Zhang2012], and carriers of the minor allele at CETP rs3764261 show a 4.4% lower high-fat diet-related increase in HDL-C [@Qi2015]. Recently, a 14-SNP score designed based on prior GDI literature was shown to predict differences in HDL-C response to a saturated fat-reduction trial in overweight adults (-6 mg/dL average change in those with high versus low response scores) [@Guevara-Cruz2019]. ____ link to our results

* Tested a whole series and only HDL came up as interesting for prediction. 
    - Made scores based on suggestive main-effect SNPs (reference the (Winkler?) paper from Danielle)
    - May be a continuing power issue -- reference the supp table with power calcs
    - However, power has to be balanced with population specificity
    - HDL scores developed in white individuals did not successfully predict in the full combined-ancestry DM trial. However, the failure of genetic risk scores to generalize across populations has been been consistently demonstrated [@Martin2017].
-->

# Methods

## Women's Health Initiative Dataset

The Women's Health Initiative study consists of a series of substudies: three clinical trials (related to cancer, cardiovascular disease, and osteoporosis) and an observational study [@Anderson1998]. Over 160,000 participants were enrolled between 1993-1998, with the ability to enroll in up to 3 of the clinical trials simultaneously. For the purposes of this analysis, participants were categorized based only on whether or not they were enrolled in the dietary modification (DM) trial, which randomized almost 50,000 women to a low-fat diet or a control diet with no recommended dietary changes, with primary outcomes being incidence of breast and colorectal cancers and heart disease [@Ritenbaugh2003. Participants were comprehensively screened at baseline, including physical measurements, blood sample collection, and questionnaire administration, while only a subset of participants provided blood samples or returned questionnaires during later visits. The food frequency questionnaire (FFQ) was designed specifically for the WHI study, emphasizing specific foods and preparation methods to maximize its sensitivity to changes in fat intake [@Patterson1999].

Phenotype data were accessed from dbGaP (accession: phs000746.v2.p3). Values shown in Table 1 only pertain to women whose genotypes were measured in one of a series of follow-up studies. For gene-diet interaction analyses, SBP, LDL-C, and GLU were adjusted for medication use: LDL-C and GLU values were divided by 0.75 for those on lipid-lowering and anti-diabetic medication, respectively, and SBP values were increased by 15 mmHg for those on anti-hypertensive medication. Cardiovascular risk factors (CRFs) were winsorized at 5 standard deviations from the mean and those other than LDL-C (BMI, SBP, HDL-C, TG, and GLU) were log-transformed prior to analysis. Longitudinal risk factor changes were calculated in DM trial participants as the difference between baseline and year 1. Phenotype data processing was performed using R version 3.4.3 [@RCoreTeam2017] and Python version 3.6.0.

## Genotype data and preprocessing

Imputed genotype data were retrieved from dbGaP (accession: phs000746.v2.p3) as a harmonized set of imputation outputs from a series of genotyping studies involving WHI participants. Prior to imputation, study-specific quality control steps had been undertaken on directly-genotyped SNPs, with filters based on sample and call rate, Hardy-Weinberg equilibrium, and minor allele frequency. Phasing had been performed for autosomes using BEAGLE, followed by imputation using minimac (MACH for the SHARe study subset). After download from dbGaP, variants were filtered for imputation R-squared > 0.3 and minor allele frequency (MAF) >0.001 and annotated with rsIDs, loci, and allelic information using the 1000 Genomes Phase 3 download from dbSNP (download date: April 13, 2018). Only variants passing the imputation quality threshold in all genotyping sub-studies were included in the final dosage dataset. For score development and imputed variant dosages were converted to hard-calls and set to missing if the estimated dosage was not within 0.1 of an integer allele count. Post-imputation genotype data processing was performed using PLINK 2.0, while clumping, and score calculation were performed using PLINK 1.9 [@Chang2015].

## Genome-wide interaction study

A genome-wide interaction study was performed for each of the six cardiometabolic risk factors. The genome-wide scan used an additive genotype model, adjusted for dietary fat (binary: % of kcals above or below the median), age, sex, and five ancestry principal components. <!--Robust standard errors were used to provide tolerance to model mis-specification as can arise in studies of gene-environment interaction [@Voorman2011].--> The primary estimand of interest was the interaction term between dietary fat and minor allele count at the SNP of interest. Interaction analyses were carried out using PLINK 2.0 [@Chang2015]. Variants of interest were annotated to genes using the Annovar tool [@Wang2010].

Gene-environment interaction power calculations for single SNPs were performed using the Quanto tool [@Gauderman2002]. The following assumptions were made: additive model; variance explained by genotype alone = 0.5%; and binary environment with 50% prevalence and explaining 10% of variance. (Note: there is no effect of minor allele frequency in this case given that variances explained are directly specified.)

## Genetic responder score construction and evaluation

To prioritize variants for inclusion in genetic responder scores, nominal (p<0.05) variants for main-effect on each risk factor were retrieved from large-scale meta-analyses: GIANT for BMI [@Yengo2018]; International Consortium for Blood Pressure for SBP [@Ehret2011]; Global Lipid Genetics Consortium (GLGC) for LDL-C, HDL-C, and TG [@Willer2013]; and MAGIC for fasting glucose [@Dupuis2010]. Each FRS was constructed using summary statistics for the diet-SNP interaction terms from the associated GWIS. SNPs were filtered for nominal main-effect relationships using the above meta-analyses, and interaction summary statistics were used as input to a pruning-and-thresholding (P&T) procedure (using the "--clump" function in PLINK 1.9), with a seed threshold of p=0.05 and an LD threshold of r^2^=0.5. The linkage disequilibrium references for the procedure was calculated from genotypes of the white DM trial participants. Genetic fat response scores (FRS) for each individual were then calculated as the weighted sum of allelic dosages for variants selected by the P&T procedure, with weights corresponding to the GWIS interaction term estimates. A genetic risk score for LDL-C was created using the GLGC LDL-C meta-analysis summary statistics and the same P&T method and parameters as was used for the interaction analyses, resulting in a 26467-SNP score.

FRS were used to test for discrimination of changes in CRFs over the first year of the DM trial. Risk factor changes were assessed using linear models in participants in the intervention arm, with and without adjustment for baseline CRF levels. As a sensitivity analysis, p-values were calculated in separate models for interaction of the genetic score with 1) trial arm (control vs. dietary modification), and 2) observed fat reduction (negative vs. positive 3-year change in FFQ-estimated dietary fat). GRS were further tested for prediction of chronic disease development during follow-up across DM trial strata. Time-to-event for each of coronary heart disease, myocardial infarction, stroke, cancer, and death were collected and used to fit age-adjusted Cox proportional hazards models. Estimated log-hazard ratios were extracted from regressions conducted in the following strata: 1) DM trial intervention arm, 2) DM trial control arm, 3) DM trial intervention arm filtered for participants with 1-year fat reduction based on FFQ, and 4) DM trial control arm filtered for participants with 1-year fat increase based on FFQ.

# Supplementary Data

\newcommand{\beginsupplement}{%
        \setcounter{table}{0}
        \renewcommand{\thetable}{S\arabic{table}}%
        \setcounter{figure}{0}
        \renewcommand{\thefigure}{S\arabic{figure}}%
     }
\beginsupplement

```{r show-power-calcs}
quanto_res %>%
  setNames(c("GxE variance explained (%)", "N (nominal)", "N (suggestive)", 
             "N (genome-wide)")) %>%
  kable(booktabs=T, longtable=T,
        caption="Sample size necessary to achieve power of 0.8")
```

```{r show-qq-plots, fig.asp=1, fig.cap="Q-Q plots from individual CRF GWIS. The distribution of p-values from each GWIS is plotted against the expected uniform p-value distribution. Plots correspond to: a) BMI, b) SBP, c) LDL-C, d) HDL-C, e) TG, and f) FG."}
a <- lapply(rfs, function(rf) {
  qq_path <- paste0(res_dir, "fat_pct_binary_", outcome_transforms[rf], 
                    "_whi_white_qq.png")
  grid::rasterGrob(png::readPNG(qq_path), interpolate=T)
})

plot_grid(a[[1]], a[[2]], a[[3]], a[[4]], a[[5]], a[[6]],
          nrow=3, labels=c("a", "b", "c", "d", "e", "f"))
```


```{r show-test-scores-cross-ancestry, message=F}
bind_rows(ca=fat_score_res_ca,
          black=fat_score_res_black,
          hispanic=fat_score_res_his,
          .id="ancestry") %>%
  select(rf, ancestry, n, effect_std, p.value) %>%
  mutate(effect_std=round(effect_std, 3),
         p.value=ifelse(p.value < 1e-3, 
                        format(p.value, digits=3, scientific=T),
                        round(p.value, 3)),
         rf=pretty_rfs[rf]) %>%
  mutate_all(as.character) %>%
  gather(key=trait, value=value, n:p.value) %>%
  mutate(anc_trait=paste(ancestry, trait, sep="_")) %>%
  select(-ancestry, -trait) %>%
  spread(key=anc_trait, value=value) %>%
  select(rf,
         ca_n, ca_effect_std, ca_p.value,
         black_n, black_effect_std, black_p.value,
         hispanic_n, hispanic_effect_std, hispanic_p.value) %>%
  setNames(c("CRF", rep(c("N", "Std. effect size", "P-value"), 3))) %>%
  kable(booktabs=T, longtable=T,
        caption="Responder score effects on CRF changes in DM trial participants across ancestries") %>%
  add_header_above(c("", "All combined"=3, "Black"=3, "Hispanic"=3))

# fat_score_res_ca %>%
#   select(rf, n_snps, n, effect_std, p.value) %>%
#   mutate(effect_std=round(effect_std, 3),
#          p.value=round(p.value, 3)) %>%
#   setNames(c("Risk factor", "# SNPs in risk score", "Sample size", 
#              "Std. effect size", "P-value")) %>%
#   kable(booktabs=T, longtable=T,
#         caption="Responder score effects on CRF changes in DM trial participants (all ancestries combined)") %>%
#   footnote(general=c("* Sample size is restricted to those with 3-year follow-up measurements for the CRF in question",
#                      "* Std. effect size represents the regression coefficient estimate in terms of CRF standard deviation per responder score standard deviation"),
#            general_title="",
#            threeparttable=T)
```

```{r show-test-ldl-scores-other-rfs, message=F}
ldl_score_res %>%
  filter(test_rf != "ldl") %>%
  select(test_rf, n_snps, n, effect_std, p.value) %>%
  mutate(effect_std=round(effect_std, 2),
         p.value=round(p.value, 2),
         test_rf=pretty_rfs[test_rf]) %>%
  setNames(c("Outcome risk factor", "# SNPs in risk score", "Sample size",
             "Std. effect size", "P-value")) %>%
  kable(booktabs=T, longtable=T,
        caption="LDL-FRS effects on alternate CRF changes in DM trial participants") %>%
  footnote(general=c(
    "* Std. effect size represents the regression coefficient estimate in terms of CRF standard deviation per responder score standard deviation"),
    general_title="",
    threeparttable=T)
```

# References





