---
output: 
  pdf_document:
    # citation_package: natbib
    keep_tex: true
    fig_caption: true
    latex_engine: pdflatex
    # template: ~/Dropbox/miscelanea/svm-r-markdown-templates/svm-latex-ms.tex
title: "Development of dietary fat responder scores using gene-diet interactions in the Women's Health Initiative"
# thanks: "Replication files are available on the author's Github account..."
# author:
# - name: Steven V. Miller
#   affiliation: Clemson University
# - name: Mary Margaret Albright
#   affiliation: Pendelton State University
# abstract: "This is the abstract..."
# keywords: "pandoc, r markdown, knitr"
# date: "`r format(Sys.time(), '%B %d, %Y')`"
# geometry: margin=1in
# fontfamily: mathpazo
# fontsize: 11pt
# # spacing: double
bibliography: whi_frp_ms.bib
# biblio-style: apsr
# see http://svmiller.com/blog/2016/02/svm-r-markdown-manuscript/ for details on this .Rmd template
---

```{r setup, include=F}
knitr::opts_chunk$set(echo=F, fig.path="figures/", 
                      cache.path="../../cache/whi_frp_ms/")
suppressMessages(silent <- lapply(
  c("knitr", "kableExtra", "tidyverse", "broom", "cowplot",
    "jtools", "ggstance"), 
  library, character.only=T))
library(jtools)
library(ggstance)
```

```{r prep}
bim_to_SnADF <- function(bfile) {
  bim_df <- read_tsv(paste0(bfile, ".bim"), 
                     col_names=c("chromosome", "rsID", "cm", 
                                 "position", "A1", "A2")) %>%
    mutate(snpID=1:nrow(.)) %>%
    mutate_at(vars(chromosome, position), as.integer)
  SnpAnnotationDataFrame(data.frame(bim_df, stringsAsFactors=F))
}

fam_to_ScADF <- function(bfile, phenos) {
  fam_df <- read_delim(paste0(bfile, ".fam"), 
                       delim=" ", col_names=c("FID", "IID", "father", "mother", 
                                               "sex", "pheno")) %>%
    select(IID) %>%
    left_join(phenos, by=c("IID")) %>%
    mutate(scanID=IID)
  ScanAnnotationDataFrame(data.frame(fam_df, stringsAsFactors=F))
}

make_gds <- function(bfile, gds_name, summary=F) {
  snpgdsBED2GDS(paste0(bfile, ".bed"),
                paste0(bfile, ".fam"),
                paste0(bfile, ".bim"),
                gds_name,
                cvt.snpid="int",
                verbose=F)
  if (summary) snpgdsSummary(gds_name)
}

run_test <- function(genoData, outcome, covars, ivar, robust, start, end) {
  assocRegression(
    genoData,
    outcome=outcome,
    model.type="linear",
    gene.action="additive",
    covar=covars,
    ivar=ivar,
    robust=robust,
    snpStart=start, 
    snpEnd=end,
    verbose=F)
}

make_qqplot <- function(p_vec, plotTitle="Title") {
  p_vec <- p_vec[!is.na(p_vec)]
  qqplot(-log10(1:length(p_vec) / length(p_vec)), -log10(p_vec), pch=".", cex=5, 
         main=plotTitle, xlab="Expected (-logP)", ylab="Observed (-logP)")
  abline(0, 1, col="red")
}

gControl <- function(p_vals) {
  # See van Iterson 2017 methods and/or Lehne 2015 code for details on genomic control for EWAS
  # Below is modeled after Lehne 2015
  lambda <- median(qchisq(p_vals, df=1, lower.tail=F), 
                   na.rm=T) / qchisq(0.5, df=1)
  round(lambda, 2)
}

outcome_transforms <- list(
  bmi="logBMI",
  sbp="logSBP",
  hdl="logHDL",
  tg="logTG",
  glu="logGLU",
  ldl="ldl"
)

rfs <- c("bmi", "sbp", "ldl", "hdl", "tg", "glu")
pretty_rfs <- c(bmi="BMI", sbp="SBP", ldl="LDL-C", hdl="HDL-C", tg="TG", glu="Glucose")
```

```{r load-results}
# load("../../output/fat_response_objects.RData")
```

# Abstract

# Introduction

* Para. 1: GxE central to the hype that has shown promise in particular applications....personalized nutrition!!
    - The emergence of gene-diet interactions (GDI) studies, along with specific well-replicated examples [@Corella___] have provided suggestions that genome-guided personalized nutrition can become a reality.
    - Many GDIs have focused in the cardiometabolic realm. Typically, these focus on cardiometabolic risk factors based on biology-based candidate genes/variants (pick a study from Jose, Lu Qi, and caffeine), but some have looked at actual outcomes (e.g. MI [@Cornelis2006]). Other approaches use main-effect genetic risk scores, such as that for obesity interacting with sugar-sweetened beverage intake to influence anthropometric traits [@Qi2012; @Olsen2016].

* Para. 2: However, to translate GxE into the most effective risk scores, 3 elements will be necessary:
    1. Replicable (find some review or similar that discusses lack of reproducibility and difficulty across populations)
    2. Genome-wide (mention that up until now most investigations are either few-gene hypothesis-based or genome-wide main effect scores)
    3. Causal/applicable to changes in RF based on change in dietary intake

Nutrigenetics approaches (in which genetic information is used to predict response to dietary inputs) are a central component to the emerging promise of personalized nutrition for cardiometabolic risk reduction. Inter-individual differences in preferences, metabolism, detoxification, excretion, etc. affect our responses to food, in a similar way to the well-studied field of pharmacogenomics [@Ma2011]. Ideally, genotype-based nutrigenetic investigations would be conducted in large-scale dietary interventions. Two notable examples are the PREDIMED and POUNDS LOST trial, with notable findings including the interaction of a TCF7L2 variant with a Mediterranean diet pattern for glycemic traits [@Corella2013] and the interaction of a PCSK9 variant with dietary carbohydrate for insulin resistance [@Huang2015]. However, these are only able to examine one dietary change (whether food, nutrient, or pattern) at a time, and are often limited to lower sample sizes [@Ordovas2018].

To allow for more flexibility and greater statistical power, gene-diet interactions (GDIs) are more commonly investigated in observational datasets. There is a rich literature of GDI discovery in the cardiometabolic realm. Typically, these focus on cardiometabolic risk factors based on biology-based candidate genes/variants [@Corella2009; @Cuda2011], but some have looked at actual outcomes (e.g. MI [@Cornelis2006]). Other approaches use main-effect genetic risk scores, such as that for obesity interacting with sugar-sweetened beverage intake to influence anthropometric traits [@Qi2012; @Olsen2016].

Characterization of individuals based on single or small groups of SNPs may neglect important signal elsewhere in the genome, especially when dealing with highly polygenic cardiometabolic traits. Thus, for effective personalized nutrition approaches to be realized, it will be necessary to integrate signals across the genome. A few investigations have explored GDIs genome-wide, such as for dairy and BMI [@Smith2018] and for various dietary components and colorectal cancer [@Figueiredo2014]. However, genome-wide interaction studies (GWIS) can be problematic due to the lower statistical power inherent in gene-environment interaction analyses [@Dempfle2008]. Furthermore, the potential for confounding and reverse causation (i.e. cardiometabolic risk impacting dietary behavior) in statistical interactions from observational data means that GDIs may not always predict modification of the risk factor in question due to dietary modification.

In order to provide proof-of-concept for use of GDIs in developing comprehensive diet response genetic scores, we sought to develop a genome-wide, GDI-based dietary fat response score for each of a series of cardiometabolic risk factors. We began by performing hypothesis-free**. We tested the performance of these dietary fat response genetic scores in the fat reduction-focused Women's Health Initiative Dietary Modification trial, observing ... (CRFs) and .... (incident CHD). We then characterized...

HYPOTHESIS-FREE VS GENOME-WIDE VS OTHER

# Results

## Dietary fat responder score development

A series of genome-wide interaction studies (GWIS) were undertaken in women from the Women's Health Initiative studies who did not participate in the dietary modification (DM) trial, using imputed genotypes along with self-reported diet intakes (from food frequency questionnaires) and fasting blood biomarkers. Baseline characteristics of these women, along with those participating in the DM trial, are shown in Table 1.

```{r pop-description, message=F}
get_median_iqr <- function(x) {
  if (all(is.na(x))) return (as.character(NA))
  median_fmt <- round(median(x, na.rm=T), 1)
  iqr_borders <- round(quantile(x, c(0.25, 0.75), na.rm=T), 1)
  paste0(median_fmt, " (", iqr_borders[1], "-", iqr_borders[2], ")")
}

# all_whi_phenos <- read_csv("../data/processed/metadata_whi.csv", 
#                            col_types=cols_only(subjID="c", smk_now="d", visit_year="d")) %>%
#   filter(visit_year == 0)
white_nondm_phenos <- read_delim(
  "../data/processed/whi_predictions_hardcalls/whi_white_gwas_phenos.txt", delim=" ") %>%
  # left_join(all_whi_phenos, by="subjID")
  select(age, smk_now, lipid_med, ht_med, dm_med, 
         bmi, sbp, ldl, hdl, tg, glu) %>%
  replace_na(list(smk_now=F, lipid_med=F, ht_med=F, dm_med=F))

white_dm_phenos <- read_delim("../data/processed/whi_predictions_hardcalls/whi_white_DM_phenos.txt", 
                        delim=" ") %>%
  filter(dm_trial == T) %>%
  select(age, smk_now, lipid_med, ht_med, dm_med, 
         bmi=baseline_bmi, sbp=baseline_sbp,
         ldl=baseline_ldl, hdl=baseline_hdl, tg=baseline_tg, glu=baseline_glu) %>%
  replace_na(list(smk_now=F, lipid_med=F, ht_med=F, dm_med=F))

pop_tbl <- bind_rows(list(nondm=white_nondm_phenos, dm=white_dm_phenos), 
                     .id="group") %>%
  group_by(group) %>%
  summarise(`Sample size`=n(),
            Age=get_median_iqr(age),
            `Current smoking`=paste0(round(sum(smk_now) / n() * 100), "%"),
            `Lipid-lowering medication`=paste0(round(sum(lipid_med) / n() * 100), "%"),
            `Hypertension medication`=paste0(round(sum(ht_med) / n() * 100), "%"),
            `Diabetes medication`=paste0(round(sum(dm_med) / n() * 100), "%"),
            `Body mass index`=get_median_iqr(bmi),
            `Systolic blood pressure`=get_median_iqr(sbp),
            `LDL cholesterol (LDL-C; mg/dL)`=get_median_iqr(ldl),
            `HDL cholesterol (HDL-C; mg/dL)`=get_median_iqr(hdl),
            `Triglycerides (Tg; mgdL)`=get_median_iqr(tg),
            `Fasting glucose (FG; mg/dL)`=get_median_iqr(glu)) %>%
  t()
colnames(pop_tbl) <- c("WHI non-DM trial", "WHI DM trial")
kable(pop_tbl, booktabs=T, longtable=T, 
      caption="WHI Population Description (European ancestry only)") %>%
  footnote(general=c("* Continuous values shown as: median (interquartile range)",
                     "* Note: Only women of European ancestry are displayed"),
           general_title="")
```

## Dietary fat responder score performance

```{r test-scores, message=F}
white_dm_phenos <- read_delim("../data/processed/whi_predictions_hardcalls/whi_white_DM_phenos_6.txt", 
                        delim=" ") %>%
  filter(dm_trial == T)
black_dm_phenos <- read_delim("../data/processed/whi_predictions_hardcalls/whi_black_DM_phenos_6.txt", 
                        delim=" ") %>%
  filter(dm_trial == T)
hispanic_dm_phenos <- read_delim("../data/processed/whi_predictions_hardcalls/whi_hispanic_DM_phenos_6.txt", 
                        delim=" ") %>%
  filter(dm_trial == T)

# dm_phenos <- bind_rows(white_dm_phenos, black_dm_phenos, hispanic_dm_phenos)
dm_phenos <- white_dm_phenos

read_profile <- function(filename, df) {
  read_table2(filename) %>%
    select(IID, SCORE, CNT) %>%
    inner_join(dm_phenos, by=c("IID"="SampleID"))
}

get_model_results <- function(form, df, score_term, res_type="statistic") {
  lm_fit <- lm(as.formula(form), data=df) %>%
    tidy() %>%
    filter(term == score_term)
  lm_fit
  # lm_fit[[res_type]]
}

test_score <- function(score_rf, test_rf=score_rf, res_type="statistic") {
  scorefile <- paste0("../data/processed/whi_predictions_hardcalls/fat_pct_binary_",
                      score_rf, ".profile")
  # scorefile <- paste0("../data/processed/ukbb_res/", score_rf, "/fat_",
  #                     score_rf, ".profile")
  if(!file.exists(scorefile)) return (NA)
  score_df <- read_profile(scorefile)
  delta_var <- paste0("delta_", test_rf)
  direct_formula <- paste0(delta_var, " ~ SCORE")
  bladj_formula <- paste0(delta_var, " ~ SCORE + baseline_", test_rf)
  interaction_formula <- paste0(delta_var, " ~ SCORE * dm_intervention")
  # data.frame(direct=get_model_results(direct_formula, filter(score_df, dm_intervention == T), "SCORE", res_type),
  #   direct_low_fat=get_model_results(
  #     direct_formula, filter(score_df, dm_intervention == T, delta_fat < 0), "SCORE", res_type),
  #   direct_bladj=get_model_results(bladj_formula, filter(score_df, dm_intervention == T), "SCORE", res_type),
  #   interaction=get_model_results(interaction_formula, score_df, "SCORE:dm_interventionTRUE", res_type),
  #   score_sd=sd(score_df$SCORE, na.rm=T))
  intervention_df <- filter(score_df, dm_intervention == T)
  get_model_results(direct_formula, intervention_df,
                               "SCORE") %>%
      # get_model_results(interaction_formula, score_df,
      #                          "SCORE") %>%
    mutate(n=sum(!is.na(intervention_df[[delta_var]])),
           score_sd=sd(intervention_df$SCORE, na.rm=T),
           rf_sd=sd(intervention_df[[delta_var]], na.rm=T),
           effect_std=estimate / rf_sd * score_sd,
           n_snps=max(intervention_df$CNT, na.rm=T) / 2)
}

fat_score_res <- tibble(rf=rfs) %>%
  mutate(res=map(rf, test_score)) %>%
  unnest(res) %>%
  mutate(rf=pretty_rfs[rf])

# fat_score_res %>%
#   ggplot(aes(x=rf, y=effect_std)) +
#   geom_bar(stat="identity")
# 
# fat_score_res %>%
#   ggplot(aes(x=1, y=rf, fill=n)) +
#   geom_tile() +
#   scale_fill_gradient2(name="Sample size with follow-up CRF values (DM trial)", high=scales::muted("red")) +
#   labs(subtitle="Estimated responder score effect (SD CRF change / SD score)") +
#   theme(axis.title=element_blank(), axis.line=element_blank(),
#         axis.ticks=element_blank(), axis.text.x=element_blank(),
#         panel.background=element_blank())

fat_score_res %>%
  select(rf, n_snps, n, effect_std, p.value) %>%
  mutate(effect_std=round(effect_std, 2),
         p.value=round(p.value, 2)) %>%
  setNames(c("Risk factor", "# SNPs in risk score", "Sample size", 
             "Std. effect size", "P-value")) %>% View()
  kable(booktabs=T, longtable=T,
        caption="Responder score effects on CRF changes in DM trial participants") %>%
  footnote(general=c("* Sample size is restricted to those with 3-year follow-up measurements for the CRF in question",
                     "* Std. effect size represents the regression coefficient estimate in terms of CRF standard deviation per responder score standard deviation"),
           general_title="",
           threeparttable=T)

# rf_units <- c(bmi="kg/m^2", sbp="mmHg", ldl="mg/dL", hdl="mg/dL", tg="mg/dL", glu="mg/dL")
```

Of the fat response scores examined, only the HDL score was predictive of the associated CRF change at nominal statistical significance (p<0.05) in DM trial participants in the fat-reduction arm. For this HDL-FRS, the standardized effect size was _____ (p=_____), translating to a lower HDL change (reduced positive change or increased negative change) over the 3-year period of _____ mg/dL per standard deviation increase in HDL-FRS. We note that the number of European-ancestry WHI participants in the intervention arm with follow-up biochemical measurements (all CRFs other than BMI and SBP) was low.

```{r test-scores-cross-ancestry, message=F}
dm_phenos <- bind_rows(white_dm_phenos, black_dm_phenos, hispanic_dm_phenos)

fat_score_res_ca <- tibble(rf=rfs) %>%
  mutate(res=map(rf, test_score)) %>%
  unnest(res) %>%
  mutate(rf=pretty_rfs[rf])

fat_score_res_ca %>%
  select(rf, n_snps, n, effect_std, p.value) %>%
  mutate(effect_std=round(effect_std, 2),
         p.value=round(p.value, 2)) %>%
  setNames(c("Risk factor", "# SNPs in risk score", "Sample size", 
             "Std. effect size", "P-value")) %>%
  kable(booktabs=T, longtable=T,
        caption="Responder score effects on CRF changes in DM trial participants (all ancestries combined)") %>%
  footnote(general=c("* Sample size is restricted to those with 3-year follow-up measurements for the CRF in question",
                     "* Std. effect size represents the regression coefficient estimate in terms of CRF standard deviation per responder score standard deviation"),
           general_title="",
           threeparttable=T)

dm_phenos <- white_dm_phenos
```

```{r test-hdl-scores-other-rfs, message=F}
hdl_score_res <- tibble(test_rf=rfs) %>%
  mutate(res=map(test_rf, function(trf) test_score("hdl", test_rf=trf))) %>%
  unnest(res) %>%
  mutate(test_rf=pretty_rfs[test_rf])

hdl_score_res %>%
  select(test_rf, n_snps, n, effect_std, p.value) %>%
  mutate(effect_std=round(effect_std, 2),
         p.value=round(p.value, 2)) %>%
  setNames(c("Outcome risk factor", "Num. SNPs in risk score", "Sample size w/ follow-up CRF measurements",
             "Std. effect size", "P-value")) %>%
  kable(booktabs=T, longtable=T,
        caption="HDL responder score effects on CRF changes in DM trial participants")
```

```{r hdl-score-strata, message=F, fig.asp=1.1, fig.cap="3-year changes in HDL-C in DM trial participants as a function of HDL responder score. Mean changes in HDL-C (y-axis) are shown as a function of responder score tertile (x-axis) in strata defined by either DM trial randomization (a) or positive vs. negative change in FFQ-based dietary fat percentage (b). Error bars represent standard errors."}
hdl_score_df <- read_profile("../data/processed/whi_predictions_hardcalls/fat_pct_binary_hdl.profile") %>%
    mutate(fat_down=delta_fat < 0)

hdl_by_randomization <- hdl_score_df %>%
  filter(!is.na(delta_hdl)) %>%
  mutate(score_quantile=cut(SCORE, quantile(SCORE, seq(0, 1, length.out=4)),
                            include.lowest=T, labels=paste0("T", 1:3)),
         dm_intervention=factor(dm_intervention, labels=c("Control Arm", "Fat-reduction Arm"))) %>%
  group_by(dm_intervention, score_quantile) %>%
  summarise(mean_change=mean(delta_hdl),
            n=n(),
            sem=sd(delta_hdl) / sqrt(n))
hdl_by_randomization_plt <- hdl_by_randomization %>%
  ggplot(aes(x=score_quantile, y=mean_change)) +
  geom_point() +
  geom_errorbar(aes(ymin=mean_change - sem, ymax=mean_change + sem), width=0) +
  # scale_x_discrete(labels=paste0("Tertile", 1:3)) + 
  labs(x="",
       y=expression(Delta*"HDL (baseline to year 3)")) +
  facet_wrap(vars(dm_intervention))

hdl_per_protocol <- hdl_score_df %>%
  filter(!is.na(delta_hdl)) %>%
  mutate(score_quantile=cut(SCORE, quantile(SCORE, seq(0, 1, length.out=4)),
                            include.lowest=T, labels=paste0("T", 1:3)),
         fat_down=factor(fat_down, labels=c("Fat Increased", "Fat Decreased"))) %>%
  group_by(fat_down, score_quantile) %>%
  summarise(mean_change=mean(delta_hdl),
            n=n(),
            sem=sd(delta_hdl) / sqrt(n))
hdl_per_protocol_plt <- hdl_per_protocol %>%
  ggplot(aes(x=score_quantile, y=mean_change)) +
  geom_point() +
  geom_errorbar(aes(ymin=mean_change - sem, ymax=mean_change + sem), width=0) +
  # scale_x_discrete(labels=paste0("Tertile", 1:3)) + 
  labs(x="HDL responder score tertile",
       y=expression(Delta*"HDL (baseline to year 3)")) +
  facet_wrap(vars(fat_down))

plot_grid(hdl_by_randomization_plt, hdl_per_protocol_plt, nrow=2,
          align="v", labels=c("a", "b"))
```

```{r test-bmi-scores, message=F, eval=F}
bmi_score_df <- read_profile("../data/processed/whi_predictions_hardcalls/fat_pct_binary_bmi.profile")

bmi_score_df %>%
  mutate(fat_down=delta_fat < 0) %>%
  select(dm_intervention, fat_down, delta_bmi, SCORE) %>%
  na.omit() %>%
  mutate(score_quartile=cut(SCORE, breaks=quantile(SCORE, seq(0, 1, length.out=4), na.rm=T),
                            labels=paste0("Q", 1:3), include.lowest=T)) %>%
  group_by(score_quartile, dm_intervention) %>%
  summarise(mean=mean(delta_bmi),
            sem=sd(delta_bmi) / sqrt(n()),
            n=n()) %>%
  ggplot(aes(x=score_quartile, y=mean)) +
  geom_point() + 
  geom_errorbar(aes(ymin=mean - sem, ymax=mean + sem), width=0) +
  labs(x="", y="delta bmi") +
  facet_wrap(vars(dm_intervention), labeller=labeller(.cols=label_both))

bmi_score_df %>%
  mutate(fat_down=delta_fat < 0) %>%
  select(dm_intervention, fat_down, delta_bmi, SCORE) %>%
  na.omit() %>%
  mutate(score_quartile=cut(SCORE, breaks=quantile(SCORE, seq(0, 1, length.out=4), na.rm=T),
                            labels=paste0("Q", 1:3), include.lowest=T)) %>%
  group_by(score_quartile, fat_down) %>%
  summarise(mean=mean(delta_bmi),
            sem=sd(delta_bmi) / sqrt(n()),
            n=n()) %>%
  ggplot(aes(x=score_quartile, y=mean)) +
  geom_point() + 
  geom_errorbar(aes(ymin=mean - sem, ymax=mean + sem), width=0) +
  labs(x="", y="delta bmi") +
  facet_wrap(vars(fat_down), labeller=labeller(.cols=label_both))

test_bmi_score <- function(rf) {
  scorefile <- "../data/processed/whi_predictions_hardcalls/fat_pct_binary_bmi.profile"
  if(!file.exists(scorefile)) return (NA)
  score_df <- read_profile(scorefile)
  delta_var <- paste0("delta_", rf)
  direct_formula <- paste0(delta_var, " ~ SCORE")
  bladj_formula <- paste0(delta_var, " ~ SCORE + baseline_", rf)
  interaction_formula <- paste0(delta_var, " ~ SCORE * dm_intervention")
  data.frame(direct=get_model_results(direct_formula, filter(score_df, dm_intervention == T), "SCORE"),
    direct_low_fat=get_model_results(
      direct_formula, filter(score_df, dm_intervention == T, delta_fat < 0), "SCORE"),
    direct_bladj=get_model_results(bladj_formula, filter(score_df, dm_intervention == T), "SCORE"),
    interaction=get_model_results(interaction_formula, score_df, "SCORE:dm_interventionTRUE"))
}

bmi_scores <- tibble(rf=rfs) %>%
  mutate(tstats=map(rf, test_bmi_score)) %>%
  unnest(tstats) %>% 
  gather(key=model, value=tstat, -rf)

ggplot(bmi_scores, aes(x=model, y=rf, fill=tstat)) +
  geom_tile() +
  scale_fill_gradient2() +
  labs("")
```

To further investigate the performance of the HDL-C FRS, mean changes in HDL-C wre calculated across responder score strata (Fig. 2). While there appeared to be a monotonic decrease in the magnitude of HDL-C increase across FRS tertiles in the fat-reduction arm, this pattern was not apparent in the control arm. As a sensitivity analysis, participants were stratified by self-reported changes in dietary fat (positive vs. negative; calculated from baseline and 3-year FFQs). Here, those increasing dietary fat showed an increase in HDL change across tertiles, versus a general decrease in those decreasing dietary fat.


Does the HDL responder score predict incident CHD down the line in DM participants?
Hypothesis: Higher (fat -> HDL) responder score will lead a lower HDL in response to fat reduction, and thus increase risk of CHD.
Observation: The opposite seems to be the case, in which higher responder score links to relatively less CHD in DM trial participants compared to controls. Maybe this is due to a more important link between this score and other RFs? For example, this HDL score corresponded positively (though NS) with TG reduction in DM intervention participants.

```{r cvd-outcomes, message=F}
outcomes <- read_csv("../data/processed/outcomes_whi.csv")

hdl_score_df_with_outcomes <- inner_join(hdl_score_df, outcomes, by="subjID")
hdl_score_df_with_outcomes <- inner_join(hdl_score_df, outcomes, by="subjID")

hdl_score_df_subsets <- list(dm_int=filter(hdl_score_df_with_outcomes, dm_intervention == T),
                             dm_ctrl=filter(hdl_score_df_with_outcomes, dm_intervention == F),
                             fat_down=filter(hdl_score_df_with_outcomes, fat_down == T),
                             fat_up=filter(hdl_score_df_with_outcomes, fat_down == F))

glm_fits <- lapply(hdl_score_df_subsets, function(df) {
    glm(formula=chd ~ SCORE * baseline_fat,
        data=df, family="binomial")
  })

plot_summs(glm_fits[[1]], glm_fits[[2]], glm_fits[[3]], glm_fits[[4]],
           model.names=c("DM Intervention", "DM Control", "Fat reduced", "Fat increased"),
           coefs=c("SCORE", "baseline_fat", "SCORE:baseline_fat"),
           scale=T) +
  labs(x="Logistic regression: log(OR) of HDL responder score for CHD",
       y="",
       title="Logistic regression for incident CHD in WHI DM participants") +
  theme(legend.title=element_blank())
```

```{r cvd-outcomes-experimental, include=F, message=F, eval=F}
# sample_ids <- read_tsv("../data/raw/whi/gen/imputedSampleInfo.txt", skip=15) %>%
#   rename(subjID=SubjectID) %>%
#   select(SampleID, subjID)

outcomes <- read_csv("../data/processed/outcomes_whi.csv")

hdl_score_df_with_outcomes <- inner_join(hdl_score_df, outcomes, by="subjID")
bmi_score_df_with_outcomes <- inner_join(bmi_score_df, outcomes, by="subjID")
sbp_score_df_with_outcomes <- inner_join(read_profile("../data/processed/whi_predictions_hardcalls/fat_pct_binary_sbp.profile"),
                                         outcomes, by="subjID")
glu_score_df_with_outcomes <- inner_join(read_profile("../data/processed/whi_predictions_hardcalls/fat_pct_binary_glu.profile"),
                                         outcomes, by="subjID")
tg_score_df_with_outcomes <- inner_join(read_profile("../data/processed/whi_predictions_hardcalls/fat_pct_binary_tg.profile"),
                                         outcomes, by="subjID")
ldl_score_df_with_outcomes <- inner_join(read_profile("../data/processed/whi_predictions_hardcalls/fat_pct_binary_ldl.profile"),
                                         outcomes, by="subjID")

# bmi_score_df_with_outcomes %>%
#   mutate(fat_down=delta_fat < 0) %>%
#   select(dm_intervention, fat_down, chd, SCORE) %>%
#   na.omit() %>%
#   mutate(score_quartile=cut(SCORE, breaks=quantile(SCORE, seq(0, 1, length.out=4), na.rm=T),
#                             labels=paste0("Q", 1:3), include.lowest=T)) %>%
#   group_by(score_quartile, dm_intervention) %>%
#   summarise(mean=sum(chd) / n(),
#             # sem=sd(delta_hdl) / sqrt(n()),
#             n=n()) %>%
#   ggplot(aes(x=score_quartile, y=mean)) +
#   geom_point() +
#   # geom_errorbar(aes(ymin=mean - sem, ymax=mean + sem), width=0) +
#   labs(x="", y="delta HDL") +
#   facet_wrap(vars(dm_intervention), labeller=labeller(.cols=label_both))

chd_glm <- bmi_score_df_with_outcomes %>%
  filter(dm_intervention == T) %>%
  glm(formula=chd ~ SCORE + age + baseline_bmi + baseline_hdl + baseline_ldl + baseline_sbp, family="binomial")
chd_glm_control <- bmi_score_df_with_outcomes %>%
  filter(dm_intervention == F) %>%
  glm(formula=chd ~ SCORE + age + baseline_bmi + baseline_hdl + baseline_ldl + baseline_sbp, family="binomial")

plot_summs(chd_glm, chd_glm_control,
           model.names=c("DM Intervention", "DM Control"),
           scale=T) +
  geom_text(x=0, y=6.4, label="Predicts lower bmi w/ fat reduction") +
  labs(x="Logistic regression: effect on CHD",
       title="Same, using fat-BMI responder score...")
```


# Discussion

* Diet response scores have been shown to predict, but are often based solely on main effects or single GDI SNPs. Here, we explored the potential for gene diet interaction (GDI)-based diet responder score development, leveraging the multi-trail setup of the Women's Health Initiative. ###We showed X and did Y.###

* Tested a whole series and only HDL came up as interesting for prediction. 
    - Made scores based on suggestive main-effect SNPs (reference the (Winkler?) paper from Danielle)
    - May be a continuing power issue -- reference the supp table with power calcs
    - However, power has to be balanced with population specificity
    - HDL sores developed in white individuals did not successfully predict in the full combined-ancestry DM trial. However, the failure of genetic risk scores to generalize across populations has been been consistently demonstrated [@Martin2017].

* What is the existing literature on prediction of HDL response?
    - First, what are expected effect sizes for macronutrient-switch CRF changes due to intervention?
        + POUNDS LOST -- find most relevant effect size here
        + Differences of ([@Qi2014] CETP rs3764261 (10.1-4.5% in non-carriers vs. 7.4-6.2% in carriers))
        + Differences 0f ([@Zhang2012] APOA5 rs964184 (approximately 3.5 mg/dL greater increase in HDL-C in high-fat group in carriers)) 
    - Specifically HDL response -- what is known and what GDIs are know for it? Overlap with specific genes?
    - Any discussion of the components of the HDL GRS?
    

    
A reasonable body of literature exists establishing GDIs for both dietary fat on CRFs [@Cuda2012; @Zheng2015] and general dietary exposures on HDL-C [@Ordovas2011]. A few studies have looked specifically at genetic variants modulating the HDL-C response to dietary fat. Cross-sectionally in the Framingham Heart Study, the -514(C/T) LIPC polymorphism associates with a 16% greater HDL-C in those with two minor alleles and high dietary fat intake, versus a 10% lower HDL-C in those with two minor alleles and low dietary fat intake [@Ordovas2002]. In addition, two studies using the POUNDS LOST trial find relationships of specific polymorphisms with 2-year changes in response to a dietary fat intervention. Carriers of a risk allele at the APOA5 variant rs964184 show an approximately 3.5 mg/dL greater high-fat diet-related increase in HDL-C [@Zhang2012], and carriers of the minor allele at CETP rs3764261 show a 4.4% lower high-fat diet-related increase in HDL-C [@Qi2015]. Recently, a 14-SNP score designed based on prior GDI literature was shown to predict differences in HDL-C response to a saturated fat-reduction trial in overweight adults (-6 mg/dL average change in those with high versus low response scores) [@Guevara-Cruz2019]. ____ link to our results
    
* Move into CRF
    - Didn't find that HDL score predicts
    - Major limitation -- how well did the trial reduce fat, and most importantly don't know how well these changes were maintained

The present study had the advantage of developing a GDI-based dietary fat score in almost 10,000 women and testing in a dietary intervention trial from the same population. However, the sample size did not afford statistical power to scan genome-wide for GDIs, and the much smaller fractions of alternate ancestries in this population made development of ancestry-specific response scores unrealistic. Additionally, the DM trial intervention in which the scores were tested did not achieve its intended fat reduction and included additional non-fat-related dietary recommendations that may have effected the interactions examined here.

# Methods

## Women's Health Initiative Dataset

The Women's Health Initiative study consists of a series of substudies: three clinical trials (related to cancer, cardiovascular disease, and osteoporosis) and an observational study [@Anderson1998]. Over 160,000 participants were enrolled between 1993-1998, with the ability to enroll in up to 3 of the clinical trials simultaneously. For the purposes of this analysis, participants were categorized based only on whether or not they were enrolled in the dietary modification (DM) trial, which randomized almost 50,000 women to a low-fat diet or a control diet with no recommended dietary changes, with primary outcomes being incidence of breast and colorectal cancers and heart disease [@Ritenbaugh2003. Participants were comprehensively screened at baseline, including physical measurements, blood sample collection, and questionnaire administration, while only a subset of participants provided blood samples or returned questionnaires during later visits. The food frequency questionnaire (FFQ) was designed specifically for the WHI study, emphasizing specific foods and preparation methods to maximize its sensitivity to changes in fat intake [@Patterson1999].

Phenotype data were accessed from dbGaP (accession: phs000746.v2.p3). Values shown in Table 1 only pertain to women whose genotypes were measured in one of a series of follow-up studies. For gene-diet interaction analyses, SBP, LDL-C, and GLU were adjusted for medication use: LDL-C and GLU values were divided by 0.75 for those on lipid-lowering and anti-diabetic medication, respectively, and SBP values were increased by 15 mmHg for those on anti-hypertensive medication. Cardiovascular risk factors (CRFs) were winsorized at 5 standard deviations from the mean and those other than LDL-C (BMI, SBP, HDL-C, TG, and GLU) were log-transformed prior to analysis.

Longitudinal risk factor changes between baseline and most recent follow-up (primarily year 3) were calculated for DM trial participants.

Phenotype data processing was performed using Python version 3.6.0 [@cite#####] (#####R version 3.4.3 [@cite#####] and).


## Genotype data and preprocessing

Imputed genotype data were retrieved from dbGaP (accession: phs000746.v2.p3) as a harmonized set of imputation outputs from a series of genotyping studies involving WHI participants. Prior to imputation, study-specific quality control steps had been undertaken on directly-genotyped SNPs, with filters based on sample and call rate, Hardy-Weinberg equilibrium, and minor allele frequency. Phasing had been performed for autosomes using BEAGLE, followed by imputation using minimac (MACH for the SHARe study subset). After download from dbGaP, variants were filtered for imputation R-squared > 0.8 and annotated with rsIDs, loci, and allelic information using the 1000 Genomes Phase 3 download from dbSNP (download date: April 13, 2018). Imputed variants were converted to hard-calls and set to missing if the estimated dosage was not within 0.1 of an integer allele count. Post-imputation genotype data processing, clumping, and score calculation were performed using PLINK 1.9 [@Chang2015].

## Genome-wide interaction study

A genome-wide interaction study was performed for each of the six (###) cardiometabolic risk factors. The genome-wide scan used an additive genotype model, adjusted for dietary fat (binary: % of kcals above or below the median), age, sex, and five ancestry principal components. Robust standard errors were used to provide tolerance to model mis-specification as can arise in studies of gene-environment interaction [@Voorman2011]. The primary estimand of interest was the interaction term between dietary fat and minor allele count at the SNP of interest. Interaction analyses were carried out using the GWASTools package for R [@cite#####].

To prioritize variants for inclusion in genetic responder scores, genome-wide "suggestive" variants for main-effect on each risk factor were retrieved from large-scale meta-analyses: GIANT for BMI [@cite GIANT]; International Consortium for Blood Pressure for SBP [@cite ICBP]; Global Lipid Genetics Consortium for LDL-C, HDL-C, and TG [@cite GLGC]; and MAGIC for fasting glucose [@cite MAGIC].

## Genetic responder score construction and evaluation

Each GRS was constructed based on summary statistics for the diet-SNP interaction terms from the associated GWIS. Summary statistics were used as input to a pruning-and-thresholding procedure (using the "--clump" function in Plink 1.9), with a seed threshold of p=#### and an LD threshold of r^2^=0.5. Genetic scores for each individual were then calculated as a weighted sum of allelic dosages (####) for variants selected by the pruning-and-thresholding procedure, with weights corresponding to the GWIS interaction term estimates.

Genetic responder scores were calculated for WHI-DM participants (as above), and used to test for discrimination of changes in CRFs over the first three years of the dietary modification trial. Risk factor changes were assessed using linear models in participants in the intervention arm, with and without adjustment for baseline CRF levels. As a sensitivity analysis, p-values were calculated in separate models for interaction of the genetic score with 1) trial arm (control vs. dietary modification), and 2) observed fat reduction (negative vs. positive 3-year change in FFQ-estimated dietary fat). GRS were further tested for prediction of 

# References