---
title: Genetic scores for saturated fat response prediction
output: pdf_document
---

```{r setup, include=F}
knitr::opts_chunk$set(echo=F, cache.path="../cache/sfa_ldl/",
                      fig.path="../output/figures/sfa_ldl/")
suppressMessages(silent <- lapply(
  c("knitr", "tidyverse", "cowplot", "parallel", "kableExtra"), library, character.only=T))
```

```{r load-metadata, message=F}
# raw_phenos_fhs <- read_csv("../data/processed/metaData_fhs.csv")
raw_phenos_fhs <- read_csv("../data/processed/metadata_fhs.csv")

raw_phenos_whi <- read_csv("../data/processed/metadata_whi.csv", 
                           col_types=cols(sex="c", lipid_med="l", dm_trial="l",
                                          dm_intervention="l"))

raw_phenos_mesa <- read_csv("../data/processed/metadata_mesa.csv")

raw_phenos_bprhs <- read_csv("../data/processed/metadata_bprhs.csv")
bprhs_id_link <- read_csv("../data/raw/bprhs/admin/BPRHS_id_link.csv")
```

```{r load-scores, message=F}
parse_scorefile <- function(filename) {
  read_tsv(filename) %>%
    dplyr::select(subjID=IID, score=SCORE1_AVG) %>%
    select(subjID, score)
}

whi_sample_to_subject <- read_tsv("../data/raw/whi/sample_info.txt", skip=15) %>%
  dplyr::select(sampleID=SampleID, subjID=SubjectID) %>%
  select(sampleID, subjID)

# whi_scores <- parse_scorefile("../data/processed/f2c_ldl_WIN/whi_scores.sscore") %>%
whi_scores <- parse_scorefile("../data/processed/gen2/sfa_ldl/whi_scores_white.sscore") %>%
  select(sampleID=subjID) %>%
  inner_join(whi_sample_to_subject, by="sampleID") %>%
  mutate(subjID=paste0("whi_", subjID))
```

```{r clean-data, message=F}
# phenos_fhs <- raw_phenos_fhs %>%
#   filter(lipid_med_5 == F) %>%
#   mutate(ldl=ldl_5, bmi=bmi_5, sfa_pct=sfa_pct_5, pufa=pufa_pct_5, age=age_5,
#          subjID=paste0("fhs_", subjID))

phenos_whi <- raw_phenos_whi %>%
  mutate(ht_med=as.logical(ht_med)) %>%
  filter(!is.na(sbp)
         # ht_med == F | is.na(ht_med)
         ) %>%
  mutate(subjID=paste0("whi_", subjID),
         f2c=(sfa + mufa + pufa) / carb)
phenos_whi_white <- phenos_whi %>%
  filter(race == "white")
phenos_nondm <- phenos_whi %>%
  filter(dm_trial == F,
         visit_year == 0)
phenos_dm <- phenos_whi %>%
  filter(dm_trial == T,
         visit_year == 0)
phenos_dm_long <- phenos_whi %>%  
  filter(
    race == "white",  ## CURRENTLY WHITES ONLY!!!
    dm_trial == T) %>%
  mutate(arm=ifelse(dm_intervention, "intervention", "control")) %>%
  group_by(subjID, age, race, sex, arm) %>%
  filter(n() > 1) %>%
  arrange(visit_year) %>%
  summarise(baseline_bmi=bmi[1],
            baseline_ldl=ldl[1],
            baseline_tg=tg[1],
            baseline_glu=glu[1],
            baseline_sbp=sbp[1],
            baseline_sfa=sfa[1],
            baseline_f2c=f2c[1],
            delta_bmi=bmi[n()] - bmi[1],
            delta_ldl=ldl[n()] - ldl[1],
            delta_tg=tg[n()] - tg[1],
            delta_glu=glu[n()] - glu[1],
            delta_sbp=sbp[n()] - sbp[1],
            delta_hsCRP=hsCRP[n()] - hsCRP[1],
            delta_sfa=sfa[n()] - sfa[1],
            delta_pufa=pufa[n()] - pufa[1],
            delta_palmitic=palmitic[n()] - palmitic[1],
            delta_f2c=f2c[n()] - f2c[1])
# phenos_whi_intervention <- phenos_whi %>%
#   filter(dm_trial == T,
#          dm_intervention == T) %>%
#   group_by(subjID, age, race, sex) %>%
#   filter(!is.na(ldl),
#          n() > 1) %>%
#   arrange(visit_year) %>%
#   summarise(baseline_ldl=ldl[1],
#             baseline_sfa_pct=sfa_pct[1],
#             baseline_f2c=f2c[1],
#             delta_ldl=ldl[n()] - ldl[1],
#             delta_sfa_pct=sfa_pct[n()] - sfa_pct[1],
#             delta_f2c=f2c[n()] - f2c[1])
# phenos_whi_control <- phenos_whi %>%
#   filter(dm_trial == T,
#          dm_intervention == F) %>%
#   group_by(subjID, age, race, sex) %>%
#   filter(!is.na(ldl),
#          n() > 1) %>%
#   arrange(visit_year) %>%
#   summarise(baseline_ldl=ldl[1],
#             baseline_sfa_pct=sfa_pct[1],
#             baseline_f2c=f2c[1],
#             delta_ldl=ldl[n()] - ldl[1],
#             delta_sfa_pct=sfa_pct[n()] - sfa_pct[1],
#             delta_f2c=f2c[n()] - f2c[1])

# phenos_mesa <- raw_phenos_mesa %>%
#   mutate(race=c("white", "asian", "black", "hispanic")[race],
#          sfa=sfa_pct,
#          pufa=pufa_pct,
#          subjID=paste0("mesa_", subjID))
# phenos_mesa_white <- phenos_mesa %>%
#   filter(race == "white")
# 
# phenos_bprhs <- raw_phenos_bprhs %>%
#   mutate(lipid_med=as.logical(lipid_med))
# scores_bprhs <- inner_join(phenos_bprhs, filter(all_scores, study == "bprhs"))
```

```{r load-scores-old, message=F, eval=F}
models <- c("fhs", "whi_white", "mesa_cau")
studies <- c("fhs", "whi_white", "mesa_cau")

parse_scorefile <- function(filename) {
  read_table(filename) %>%
    rename(subjID=IID, score=SCORE) %>%
    select(subjID, score)
}

md_combos <- expand.grid(model=models, study=studies, stringsAsFactors=F)
all_scores_long <- map2_dfr(md_combos$model, md_combos$study, function(m, s) {
  parse_scorefile(paste0("../data/processed/sfa/", s, "_scores_", m, "_model.profile")) %>%
    mutate(model=m,
           subset=s)
})
all_scores <- all_scores_long %>%
  spread(key=model, value=score) %>%
  mutate(study=case_when(subset == "fhs" ~ "fhs",
                         subset == "whi_white" ~ "whi",
                         subset == "mesa_cau" ~ "mesa")) %>%
  mutate(own_score=case_when(subset == "fhs" ~ fhs,
                             subset == "whi_white" ~ whi_white,
                             subset == "mesa_cau" ~ mesa_cau)) %>%
  mutate(subjID=ifelse(study != "bprhs", paste(study, subjID, sep="_"),
                       bprhs_id_link$studyid[match(.$subjID, bprhs_id_link$nelid_b)]))
```

# Model intuition

To develop a genetic risk score for "saturated fat responsiveness", the ideal dataset would be a large-scale dietary intervention either increasing of decreasing saturated fat, enabling a model to directly predict changes in a cardiovascular risk factor over the course of the intervention. However, we can create a noisy approximation of this model using cross-sectional data by instead using a product of SFA intake (centered) and some risk factor (centered) as the outcome. The intuition for this approach is based on the fact that we are ultimately trying to model the correlation between SFA and a risk factor, and correlations are defined mathematically as the expected value of a product of two variables.

So, the model being used in each GWAS is the following: 
$Y=\alpha g+X\beta+\epsilon$,
where Y is the product of two standardized variables (SFA and LDL-C), g is a vector of genotype dosages at the SNP in question, and X is a matrix of covariates.

# Simulations

Strategy: for each simulation...

1. Draw N genotypes from a binomial distribution (binom(2, maf)).
2. Draw SFA intakes with a potential genotype effect ($N(\beta_{SFA}*g, 1)$).
3. Draw LDL values with a direct genotype effect, a direct SFA effect, and a genotype-SFA interaction ($N(\beta_{LDL}*g+\beta_{SFA}*SFA+\beta_{int}*SFA*g, 1)$)).

Repeated simulations (100) are then used to calculate power for each set of marameters using a linear model predicting $SFA_{scaled}*LDL_{scaled}$ from genotype.

Default parameter values:

* N = 5000
* $\beta_{g\_SFA}$ = 0.1
* $\beta_{g\_LDL}$ = 0.2
* $\beta_{SFA\_LDL}$ = 0.1
* $\beta_{int}$ = 1 * $\beta_{SFA\_LDL}$ (interaction effect magnitude is similar to the genotype main effect)

```{r run-simulations, cache=1}
base_N <- 5000
base_maf <- 0.2
base_beta_g_sfa <- 0.1  # relationship w/ genotype contributes ~1% of variance in SFA
base_beta_g_ldl <- 0.1  # relationship w/ genotype contributes ~1% of variance in LDL
base_beta_sfa_ldl <- sqrt(0.05)  # relationship w/ SFA contributes ~5% of variance in LDL
base_beta_int_mult <- 0.5  # SFA change contributes 50% as much per minor allele as SFA main effect 

# g <- rbinom(N, 2, maf)
# sfa <- rnorm(N, g * beta_g_sfa, 1)  # G contrib. to sfa variance is beta_g_sfa^2 * var(g) / var(sfa)
# ldl <- rnorm(N, g * beta_g_ldl + sfa * beta_sfa_ldl +
#                (g - 2 * maf) * sfa * beta_int_mult * beta_sfa_ldl, 1)

simulate_geno_pheno <- function(N=base_N, 
                                maf=base_maf,
                                beta_g_sfa=base_beta_g_sfa, 
                                beta_g_ldl=base_beta_g_ldl,
                                beta_sfa_ldl=base_beta_sfa_ldl,
                                beta_int_mult=base_beta_int_mult,
                                test_type="product") {
  # Run a single simulation, drawing N genotypes and phenotype pairs
  g <- rbinom(N, 2, maf)
  sfa <- rnorm(N, g * beta_g_sfa, 1)
  ldl <- rnorm(N, g * beta_g_ldl + sfa * beta_sfa_ldl + 
                 (g - 2 * maf) * sfa * beta_int_mult * beta_sfa_ldl, 1)
  if (test_type == "product") {
    lm_fit <- lm(scale(sfa) * scale(ldl) ~ g)
    summary(lm_fit)$coef["g", c("Estimate", "Pr(>|t|)")]
  } else {
    lm_fit <- lm(ldl ~ sfa * g)
    summary(lm_fit)$coef["sfa:g", c("Estimate", "Pr(>|t|)")]
  }
}

calc_power <- function(arglist, n_sims=250, alpha=5e-8) {
  # tryCatch(sim_res <- sapply(1:n_sims, function(i) do.call(simulate_geno_pheno, arglist)),
  #          error=function(e) print("AAAAAAA"))
  sim_res <- sapply(1:n_sims, function(i) do.call(simulate_geno_pheno, arglist))
  hits <- (sim_res["Estimate", ] > 0) & (sim_res["Pr(>|t|)", ] < alpha)
  power <- sum(hits) / length(hits)
  power
}

calc_power_interactions <- function(arglist, n_sims=250, alpha=5e-8) {
  # tryCatch(sim_res <- sapply(1:n_sims, function(i) do.call(simulate_geno_pheno, arglist)),
  #          error=function(e) print("AAAAAAA"))
  sim_res <- sapply(1:n_sims, function(i) do.call(simulate_geno_pheno, c(arglist, list(test_type="int"))))
  hits <- (sim_res["Estimate", ] > 0) & (sim_res["Pr(>|t|)", ] < alpha)
  power <- sum(hits) / length(hits)
  power
}

vary_N <- data.frame(N=c(1e3, 2.5e3, 5e3, 1e4, 5e4),
                     maf=0.2, beta_g_ldl=0.1, beta_int_mult=0.5)
vary_maf <- data.frame(maf=c(0.05, 0.1, 0.2, 0.3, 0.4),
                     N=5e3, beta_g_ldl=0.1, beta_int_mult=0.5)
vary_beta_g_ldl <- data.frame(beta_g_ldl=c(0.01, 0.05, 0.1, 0.2, 0.3),
                     N=5e3, maf=0.2, beta_int_mult=0.5)
vary_beta_int_mult <- data.frame(beta_int_mult=c(0.1, 0.25, 0.5, 1, 2),
                     N=5e3, maf=0.2, beta_g_ldl=0.1)
param_variations <- bind_rows(list(N=vary_N, maf=vary_maf,
                                   beta_g_ldl=vary_beta_g_ldl,
                                   beta_int_mult=vary_beta_int_mult), .id="varied")

powers <- param_variations %>%
  rowwise() %>%
  mutate(
    power_nominal=calc_power(
      list(N=N, maf=maf, beta_g_ldl=beta_g_ldl, beta_int_mult=beta_int_mult), alpha=0.05),
    power_suggestive=calc_power(
      list(N=N, maf=maf, beta_g_ldl=beta_g_ldl, beta_int_mult=beta_int_mult), alpha=1e-5),
    power_gw=calc_power(
      list(N=N, maf=maf, beta_g_ldl=beta_g_ldl, beta_int_mult=beta_int_mult)),
    # power_nominal_int=calc_power_interactions(
    #   list(N=N, maf=maf, beta_g_ldl=beta_g_ldl, beta_int_mult=beta_int_mult), alpha=0.05),
    # power_suggestive_int=calc_power_interactions(
    #   list(N=N, maf=maf, beta_g_ldl=beta_g_ldl, beta_int_mult=beta_int_mult), alpha=1e-5),
    power_gw_int=calc_power_interactions(
      list(N=N, maf=maf, beta_g_ldl=beta_g_ldl, beta_int_mult=beta_int_mult))
  ) %>%
  ungroup() %>%
  mutate(var=case_when(varied == "N" ~ N,
                       varied == "maf" ~ maf,
                       varied == "beta_g_ldl" ~ beta_g_ldl,
                       varied == "beta_int_mult" ~ beta_int_mult),
         var=factor(var),
         varied=factor(varied, levels=c("N", "maf", "beta_g_ldl", "beta_int_mult")))
```

```{r negative-simulations}
neg_sims_gw <- c(calc_power(list(beta_g_ldl=0.1, beta_g_sfa=0.1, beta_int_mult=0)),
                 calc_power(list(beta_g_ldl=0.3, beta_g_sfa=0.3, beta_int_mult=0)),
                 calc_power(list(beta_g_ldl=0.5, beta_g_sfa=0.5, beta_int_mult=0)),
                 calc_power(list(beta_g_ldl=0.5, beta_g_sfa=0.1, beta_int_mult=0)))

neg_sim_tbl <- data.frame(beta_g_ldl=c(0.1, 0.3, 0.5, 0.5),
                          beta_g_sfa=c(0.1, 0.3, 0.5, 0.1),
                          power_GW=neg_sims_gw)
kable(neg_sim_tbl, format="latex", booktabs=T,
      caption="Negative simulations: fraction of false positives when interaction beta is 0")
```

```{r run-simulations-old, cache=1, eval=F}
simulate_geno_pheno <- function(N=1000, 
                                mean_corr=0.1, sd_corr=0.1,
                                mean_sfa=20, sd_sfa=5, 
                                mean_ldl=120, sd_ldl=20, 
                                maf=0.2, delta_p_per_sd=0.1) {
  # Run a single simulation, drawing N phenotype pairs and N genotypes
  corrs <- rnorm(N, mean_corr, sd_corr)
  phenos <- sapply(corrs, function(corr) {
    cov_i <- corr * sd_sfa * sd_ldl
    V <- matrix(c(sd_sfa ** 2, cov_i, cov_i, sd_ldl ** 2), 2, 2)
    tryCatch(mvnfast::rmvn(1, c(mean_sfa, mean_ldl), V),
             error=function(e) c(NA, NA))
  })
  if (sum(is.na(phenos)) / length(phenos) > 0.05) {
    print(">5% rmvn failed")
    return(NA)}
  phenos_df <- setNames(as.tibble(t(phenos)), c("sfa", "ldl"))
  p_alt_vec <- maf + (delta_p_per_sd * scale(corrs))
  p_alt_vec <- na.omit(p_alt_vec)
  p_alt_vec[p_alt_vec < 0] <- 0
  p_alt_vec[p_alt_vec > 1] <- 1
  genos <- rbinom(length(p_alt_vec), 1, p_alt_vec) +
    rbinom(length(p_alt_vec), 1, p_alt_vec)
  summary(lm(scale(sfa) * scale(ldl) ~ genos,
             data=phenos_df))$coef["genos", "Pr(>|t|)"]
}

get_pvals <- function(arglist, n_sims, alpha) {
  tryCatch(pvec <- sapply(1:n_sims, function(i) do.call(simulate_geno_pheno, arglist)),
           error=function(e) print("AAAAAAA"))
  pvec
}

vary_N <- tibble(N=c(1e3, 5e3, 1e4, 5e4), sd_corr=0.1, maf=0.2, delta_p_per_sd=0.1,
                     varied="N")
vary_sd_corr <- tibble(N=5000, sd_corr=c(0.05, 0.1, 0.2, 0.3),
                       maf=0.2, delta_p_per_sd=0.1, 
                       varied="sd_corr")
vary_maf <- tibble(N=5000, sd_corr=0.1, maf=c(0.1, 0.2, 0.3, 0.4), 
                           delta_p_per_sd=0.1, varied="maf")
vary_delta_p_per_sd <- tibble(N=5000, sd_corr=0.1, maf=0.2, 
                           delta_p_per_sd=c(0.01, 0.05, 0.1, 0.5), 
                           varied="delta_p_per_sd")
params_explore <- bind_rows(vary_N, vary_sd_corr, vary_maf, vary_delta_p_per_sd)

p_vecs <- with(params_explore, mcmapply(function (a,b,c,d) get_pvals(
  list(N=a, sd_corr=b, maf=c, delta_p_per_sd=d), n_sims=100, alpha=0.05),
  N, sd_corr, maf, delta_p_per_sd, mc.cores=4))
```

```{r show-simulations}
ggplot(powers, aes(x=var, y=power_nominal)) +
  geom_bar(stat="identity") +
  facet_wrap(~varied, scales="free_x") +
  labs(title="Power at nominal significance (alpha = 0.05)",
       x=NULL, y="Power")
ggplot(powers, aes(x=var, y=power_suggestive)) +
  geom_bar(stat="identity") +
  facet_wrap(~varied, scales="free_x") +
  labs(title="Power at suggestive significance (alpha = 1e-5)",
       x=NULL, y="Power")
ggplot(powers, aes(x=var, y=power_gw)) +
  geom_bar(stat="identity") +
  facet_wrap(~varied, scales="free_x") +
  labs(title="Power at genome-wide significance (alpha = 5e-8)", 
       x=NULL, y="Power")
ggplot(powers, aes(x=var, y=power_gw_int)) +
  geom_bar(stat="identity") +
  facet_wrap(~varied, scales="free_x") +
  labs(title="Power of standard interaction method at genome-wide significance (alpha = 5e-8)", 
       x=NULL, y="Power")
```


# Individual GWAS results

GWAS model (as above): $Y=\alpha g+X\beta+\epsilon$

* Y is the product of two standardized variables (SFA and LDL-C)
    - SFA in units of g/day, normalized by total calories
* g is a vector of genotype dosages at the SNP in question
* X is a matrix of covariates

Inclusion criteria:

* White
* No lipid medication use

## FHS

* N ~ 750 unrelated individuals
* Covariates: age, sex, BMI, PUFA

```{r fhs-plots, eval=F}
fhs_qq <- ggdraw() + draw_image("../data/processed/sfa_ldl/fhs_qq.png")
fhs_man <- ggdraw() + draw_image("../data/processed/sfa_ldl/fhs_manhattan.png")
plot_grid(fhs_qq, fhs_man, ncol=2)
```

## WHI

* N ~ 5500
* Covariates: age, BMI, PUFA, 5 ancestry principal components

```{r whi-manhattan, eval=F}
whi_white_qq <- ggdraw() + 
  draw_image("../data/processed/sfa_ldl/whi_white_qq.png")
whi_white_man <- ggdraw() + 
  draw_image("../data/processed/sfa_ldl/whi_white_manhattan.png")
whi_black_qq <- ggdraw() + 
  draw_image("../data/processed/sfa_ldl/whi_black_qq.png")
whi_black_man <- ggdraw() + 
  draw_image("../data/processed/sfa_ldl/whi_black_manhattan.png")
whi_hispanic_qq <- ggdraw() + 
  draw_image("../data/processed/sfa_ldl/whi_hispanic_qq.png")
whi_hispanic_man <- ggdraw() + 
  draw_image("../data/processed/sfa_ldl/whi_hispanic_manhattan.png")
plot_grid(whi_white_qq, whi_white_man, 
          whi_black_qq, whi_black_man,
          whi_hispanic_qq, whi_hispanic_man,
          nrow=3, ncol=2)
```

## MESA

* N ~ 1300
* Covariates: age, BMI, PUFA

```{r mesa-manhattan, eval=F}
mesa_white_qq <- ggdraw() + 
  draw_image("../data/processed/sfa_ldl/mesa_white_qq.png")
mesa_white_man <- ggdraw() + 
  draw_image("../data/processed/sfa_ldl/mesa_white_manhattan.png")
mesa_black_qq <- ggdraw() + 
  draw_image("../data/processed/sfa_ldl/mesa_black_qq.png")
mesa_black_man <- ggdraw() + 
  draw_image("../data/processed/sfa_ldl/mesa_black_manhattan.png")
# mesa_hispanic_qq <- ggdraw() + 
#   draw_image("../data/processed/sfa_ldl/mesa_hispanic_qq.png")
# mesa_hispanic_man <- ggdraw() + 
#   draw_image("../data/processed/sfa_ldl/mesa_hispanic_manhattan.png")
# mesa_asian_qq <- ggdraw() +
#   draw_image("../data/processed/sfa_ldl/mesa_asian_qq.png")
# mesa_asian_man <- ggdraw() +
#   draw_image("../data/processed/sfa_ldl/mesa_asian_manhattan.png")
plot_grid(mesa_white_qq, mesa_white_man, 
          mesa_black_qq, mesa_black_man,
          # mesa_hispanic_qq, mesa_hispanic_man,
          # mesa_asian_qq, mesa_asian_man,
          nrow=2, ncol=2)
```


# Meta-analysis results

Fixed-effects meta-analysis

## Whites only

WHI (not in diet mod. arm, white) + FHS + MESA (white)

```{r meta-white-plots, eval=F}
white_meta_qq <- ggdraw() + draw_image("../data/processed/f2c_ldl/white_meta_qq.png")
# white_meta_man <- ggdraw() + draw_image("../data/processed/f2c_ldl/white_meta_manhattan.png")
# plot_grid(white_meta_qq, white_meta_man, ncol=2)
white_meta_qq
```

## Non-whites

WHI (not in diet mod. arm, black + hispanic) + MESA (black + hispanic + asian)

```{r meta-nonwhite-plots, eval=F}
nonwhite_meta_qq <- ggdraw() + draw_image("../data/processed/f2c_ldl/nonwhite_meta_qq.png")
nonwhite_meta_man <- ggdraw() + draw_image("../data/processed/f2c_ldl/nonwhite_meta_manhattan.png")
plot_grid(nonwhite_meta_qq, nonwhite_meta_man, ncol=2)
```

## All races

WHI (not in diet mod. arm, all races) + FHS + MESA (all) + BPRHS

```{r meta-all-plots, eval=F}
all_meta_qq <- ggdraw() + draw_image("../data/processed/f2c_ldl/all_meta_qq.png")
all_meta_man <- ggdraw() + draw_image("../data/processed/f2c_ldl/all_meta_manhattan.png")
plot_grid(all_meta_qq, all_meta_man, ncol=2)
```

# Comparison to existing literature
## Sanity check: specific SNPs of interest

```{r load-gwas-results, eval=F}
system.time(meta_res <- read_tsv("../data/processed/f2c_ldl_WIN/all_meta.res", 
                                 col_types = cols_only(SNP="c", BETA="d", P="d")))
```

SNPs from CardioGxE (Parnell et al., 2014) showing GxE for SFA --> LDL-C should in theory be found by this analysis.

```{r cardiogxe-snps, message=F, eval=F}
cardio_gxe <- read_tsv("../data/raw/literature/cardio_gxe_catalog.txt", skip=2)
f2c_gxe <- cardio_gxe %>% 
  filter(grepl("SFA|fat", `Environmental factor`),
         !grepl("ratio", `Environmental factor`),
         # grepl("crp", Phenotype, ignore.case=T),
         `Interaction significance` == "significant interaction") %>%
  select(SNP, Gene, `Environmental factor`, Phenotype) %>%
  distinct(SNP, .keep_all=T)

lit_comparison <- left_join(sfa_gxe, meta_res, by="SNP") %>%
  arrange(P)

kable(lit_comparison, booktabs=T) %>%
  kable_styling(latex_options="scale_down")
```

```{r tmp, eval=F}
ldl <- read_tsv("../data/processed/adj_set/sfa_ldl_R/whi_meta.res",
                col_types=cols_only(SNP="c", P="d"))
ldl <- setNames(ldl, c("SNP", "P_LDL"))
tg <- read_tsv("../data/processed/adj_set/sfa_tg_R/whi_meta.res",
                col_types=cols_only(SNP="c", P="d"))
tg <- setNames(tg, c("SNP", "P_TG"))
glu <- read_tsv("../data/processed/adj_set/sfa_glu_R/whi_meta.res",
                col_types=cols_only(SNP="c", P="d"))
glu <- setNames(glu, c("SNP", "P_GLU"))
crp <- read_tsv("../data/processed/adj_set/sfa_hsCRP_R/whi_meta.res",
                col_types=cols_only(SNP="c", P="d"))
crp <- setNames(crp, c("SNP", "P_CRP"))

a <- Reduce(left_join, list(f2c_gxe, ldl, tg, glu, crp))
b <- a %>% 
  select(SNP, contains("P_")) %>% 
  mutate_at(vars(contains("P_")), function(x) x < 0.05)
```

Known LDL-C drug response modifier SNPs:

```{r drug-response-snps, eval=F}
drug_response_snps <- c("rs3841260", "rs140229040",
                        "rs17111584", "rs2199936", "rs10455872", "rs71352238")
drug_response_comparison <- filter(meta_res, SNP %in% drug_response_snps)
```

## Replication of top hits

```{r replication, eval=F}
white_res <- read_tsv("../data/processed/sfa_ldl/white_meta.res")
nonwhite_res <- read_tsv("../data/processed/sfa_ldl/nonwhite_meta.res")

a <- filter(white_res, P < 1e-8)
b <- filter(nonwhite_res, SNP %in% a$SNP) %>%
  arrange(P)
```


<!--
# Initial results

How to combine these cohorts?

* Meta-analysis is an option, but didn't perform it here because of the plans to incorporate other ethnicities and the questionable transference of genetic scores across ancestries.
* Approach based on Patil et al. 2018 -- combine individual predictions from each cohort rather than first meta-analyzing the GWAS results
* They are interested in collaborating on this, but here I have implemented a basic version of one of their strategies.

```{r merged-predictions, eval=F}
dummy_merge_func <- function(stacked_matrix, train_subsets, weights=NULL) {
  # Takes in a "stacked" matrix of predictions (N_subjects x n_models),
  # calculates appropriate model weights (now just dummy weights),
  # and outputs final scores from the ensemble model
  score_matrix <- as.matrix(stacked_matrix[, train_subsets])
  n_models <- ncol(score_matrix)
  weights <- if (is.null(weights)) rep(1 / n_models, n_models) else weights
  as.vector(score_matrix %*% weights)
}

better_merge_func <- function(stacked_matrix) {
  models <- c("fhs", "whi_white", "mesa_cau")
  studies <- c("fhs", "whi_white", "mesa_cau")
  lm_weights <- 1 / summary(factor(stacked_matrix$subset))[studies]
  lm_fit <- lm(as.formula(paste("own_score ~ ", paste(studies, collapse="+"))), 
               data=stacked_matrix, 
               weights=lm_weights[stacked_matrix$subset])
  pred_weights <- summary(lm_fit)$coef[studies, "Estimate"]
  pred_weights_norm <- pred_weights / sum(pred_weights)
  as.vector(as.matrix(stacked_matrix[, studies]) %*% pred_weights_norm)
}

all_scores <- all_scores %>%
  mutate(equal_score=dummy_merge_func(., c("fhs", "whi_white", "mesa_cau")),
    w_score=whi_white,
    fw_score=dummy_merge_func(., c("fhs", "whi_white"), weights=c(0.2, 0.8))) %>%
    # fwm_score=dummy_merge_func(
    #   as.matrix(.[, c("fhs", "whi_white", "mesa_cau")]), c(0.2, 0.6, 0.2))) %>%
  mutate(stack_score=better_merge_func(.)) %>%
  select(-study, -subset)

# all_scores$score <- dummy_merge_func(as.matrix(
#   all_scores[, c("fhs", "whi_white", "mesa_cau")]))
# all_scores$w_score <- all_scores$whi_white
# all_scores$fw_score <- dummy_merge_func(as.matrix(
#   all_scores[, c("fhs", "whi_white")]
# ), c(0.2, 0.8))
# all_scores$fwm_score <- dummy_merge_func(as.matrix(
#   all_scores[, c("fhs", "whi_white", "mesa_cau")]
# ), c(0.2, 0.6, 0.2))
```

```{r add-scores, eval=F}
all_phenos <- bind_rows(fhs=phenos_fhs,
                        whi_white=phenos_whi_white,
                        whi_nondm=phenos_nondm,
                        whi_dm=phenos_dm,
                        whi_intervention=phenos_whi_intervention,
                        whi_control=phenos_whi_control,
                        mesa_white=phenos_mesa_white,
                        .id="subset")

full_data <- inner_join(all_phenos, all_scores, by="subjID")

# scores_fhs <- inner_join(phenos_fhs, filter(all_scores, study == "fhs"))
# 
# scores_whi_white <- inner_join(phenos_whi, 
#                                filter(all_scores, study == "whi_white"))
# scores_whi_nondm <- filter(scores_whi_white,
#                            dm_trial == F,
#                            visit_year == 0)
# scores_whi_dm <- filter(scores_whi_white,
#                         dm_trial == T,
#                         visit_year == 0)
# scores_whi_intervention <- scores_whi_white %>%
#   filter(dm_trial == T,
#          dm_intervention == T) %>%
#   group_by(subjID, score, fh_score, age, race, sex) %>%
#   filter(sum(!is.na(ldl)) > 1) %>%
#   arrange(visit_year) %>%
#   summarise(delta_ldl=ldl[n()] - ldl[1])
# scores_whi_control <- scores_whi_white %>%
#   filter(dm_trial == T,
#          dm_intervention == F) %>%
#   group_by(subjID, score, fh_score, age, race, sex) %>%
#   filter(sum(!is.na(ldl)) > 1) %>%
#   arrange(visit_year) %>%
#   summarise(delta_ldl=ldl[n()] - ldl[1])
# 
# scores_mesa <- inner_join(phenos_mesa, filter(all_scores, study == "mesa_cau"),
#                           by="subjID")
```

```{r cs-eval, eval=F}
# phenos_all <- bind_rows(fhs=phenos_fhs,
#                         whi_dm=phenos_dm,
#                         whi_nondm=phenos_nondm,
#                         mesa_cau=phenos_mesa, .id="subset") %>%
#   inner_join(select(all_scores, -study), by="subjID")

# phenos_all <- bind_rows(fhs=scores_fhs, 
#                         whi_dm=scores_whi_dm,
#                         whi_nondm=scores_whi_nondm,
#                         mesa_cau=scores_mesa, .id="subset") %>%
#   inner_join(select(all_scores, -study), by="subjID")

find_slope <- function(df) {
  df$sfa <- scale(df$sfa)
  df$pufa <- scale(df$pufa)
  lm_fit <- lm(ldl ~ sfa + pufa + age, data=df)
  coefs <- summary(lm_fit)$coef
  unname(coefs["sfa", c("Estimate", "Std. Error")])
}

find_slopes_by_quantile <- function(df, score_col, n_groups) {
  df$quant <- cut(df[[score_col]], 
                  breaks=quantile(df[[score_col]], 
                                  seq(0, 1, length.out=n_groups + 1)),
                  labels=paste0("Q", seq(1, n_groups)),
                  include.lowest=T)
  df %>%
    nest(-quant) %>%
    mutate(n=map_int(data, nrow)) %>%
    filter(n > 50) %>%
    mutate(params=map(data, find_slope),
           slope=map_dbl(params, 1),
           se=map_dbl(params, 2)) %>%
    select(quant, slope, se, n)
}

train_subsets <- c("whi_nondm", "fhs", "mesa_white")
test_subsets <- c("whi_dm")
train_slopes_df <- map_dfr(setNames(train_subsets, train_subsets), function(s) {
  find_slopes_by_quantile(filter(full_data, subset == s), "stack_score", 4)
}, .id="subset")
test_slopes_df <- map_dfr(setNames(test_subsets, test_subsets), function(s) {
  find_slopes_by_quantile(filter(full_data, subset == s), "stack_score", 4)
}, .id="subset")
all_slopes_df <- bind_rows(trainsets=train_slopes_df, 
                           testset=test_slopes_df, .id="traintest")
  

ggplot(all_slopes_df, aes(x=quant, y=slope, color=subset, group=subset)) +
  geom_point(aes(size=n), position=position_dodge(width=0.5)) +
  geom_errorbar(aes(ymin=slope-se, ymax=slope+se), width=0,
                position=position_dodge(width=0.5)) +
  labs(x="Polygenic score quintile", 
       y="Slope of SFA -> LDL relationship") +
  facet_wrap(~traintest, scales="free")
```
-->


# Longitudinal results in WHI DM subjects

In WHI dietary modification trial subjects, does the polygenic score modify changes in LDL-C (and, specifically, differentially between control and intervention arms)?

```{r whi-longitudinal, fig.asp=1.2, eval=F}
# intervention_plt_data <- scores_whi_control %>%
#   mutate(score_quintile=cut(score,
#                             breaks=quantile(.$fh_score, seq(0, 1, 0.25)),
#                             # breaks=quantile(.$fh_score, c(0,0.3,0.6,1)),
#                             labels=paste0("Q", 1:4),
#                             include.lowest=T)) %>% 
#   group_by(score_quintile) %>%
#   summarise(delta=mean(delta_ldl),
#             se=sd(delta_ldl) / sqrt(n()))

whi_dm_cs <- inner_join(whi_scores, phenos_dm, by="subjID") %>%
  filter(visit_year == 0) %>%
  mutate(sfa_ldl_product=scale(sfa_pct) * scale(ldl),
         f2c_ldl_product=scale(f2c) * scale(ldl),
         sfa_sbp_product=scale(sfa_pct) * scale(sbp))

whi_dm_long <- inner_join(whi_scores, phenos_dm_long, by="subjID") %>%
  mutate(score_bucket=cut(score, 
                          # breaks=4,
                          breaks=quantile(score, seq(0, 1, length.out=5)),
                          labels=seq(1, 4), include.lowest=T),
         # score_bucket=cut(score, 3, labels=1:3),
         delta_sfa_bucket=case_when(delta_sfa_pct < median(delta_sfa_pct) ~ "decreased",
                                    delta_sfa_pct > 0 ~ "increased",
                                    TRUE ~ as.character(NA)))
whi_dm_long_plt_data <- whi_dm_long %>%
  group_by(arm, score_bucket) %>%
  summarise(mean_delta_ldl=mean(delta_ldl),
            SD=sd(delta_ldl),
            SE=SD / sqrt(n()))
# whi_dm_long_plt_data_sfabucket <- whi_dm_long %>%
#   group_by(delta_sfa_bucket, score_bucket) %>%
#   summarise(mean_delta_ldl=mean(delta_ldl),
#             SD=sd(delta_ldl),
#             SE=SD / sqrt(n()))
  
by_assignment_plt <- ggplot(data=whi_dm_long_plt_data, aes(x=score_bucket)) +
  geom_boxplot(data=whi_dm_long, aes(y=delta_ldl), outlier.shape=NA, alpha=0.25) +
  geom_point(data=whi_dm_long, aes(x=score_bucket, y=delta_ldl),
             alpha=0.3, position=position_jitter(width=0.3)) +
  coord_cartesian(ylim=c(-100, 60)) +
  facet_wrap(~arm)

means_plus_ses_plt <- ggplot(data=whi_dm_long_plt_data, aes(x=score_bucket)) +
  geom_point(data=whi_dm_long_plt_data, 
             aes(x=score_bucket, y=mean_delta_ldl),
             stat="identity") +
  geom_errorbar(data=whi_dm_long_plt_data, 
                aes(ymin=mean_delta_ldl - SE, ymax=mean_delta_ldl + SE), width=0) +
  facet_wrap(~arm)

# by_delta_plt <- ggplot(data=whi_dm_long_plt_data_sfabucket, aes(x=score_bucket)) +
#   geom_point(aes(x=score_bucket, y=mean_delta_ldl),
#              stat="identity") +
#   geom_errorbar(aes(ymin=mean_delta_ldl - SE, ymax=mean_delta_ldl + SE), width=0) +
#   # geom_boxplot(color="gray") +
#   geom_point(data=whi_dm_long, aes(x=score_bucket, y=delta_ldl),
#              alpha=0.2, position=position_jitter()) +
#   coord_cartesian(ylim=c(-100, 60)) +
#   facet_wrap(~delta_sfa_bucket)

plot_grid(by_assignment_plt, means_plus_ses_plt, nrow=2)

# need to make sure that there are >1 LDL measurement when taking differences
```

```{r TESTING-TESTING, fig.asp=1, eval=F}
testlong <- function(filepath, rf) {
  myscores <- parse_scorefile(filepath) %>%
    select(sampleID=subjID, score) %>%
    inner_join(whi_sample_to_subject, by="sampleID") %>%
    mutate(subjID=paste0("whi_", subjID))
  
  
  my_whi_dm_long <- inner_join(myscores, phenos_dm_long, by="subjID") %>%
    mutate(score_bucket=cut(score, 
                            # breaks=4,
                            breaks=quantile(score, seq(0, 1, length.out=5)),
                            labels=seq(1, 4), include.lowest=T),
           # score_bucket=cut(score, 3, labels=1:3),
           delta_sfa_bucket=case_when(delta_sfa < median(delta_sfa) ~ "decreased",
                                      delta_sfa > 0 ~ "increased",
                                      TRUE ~ as.character(NA)),
           delta_rf=.[[rf]])
  
  print(summary(lm(delta_rf ~ arm * score, data=my_whi_dm_long))$coef)
  # print(summary(lm(delta_rf ~ arm * race, data=my_whi_dm_long))$coef)
  
  # mod1 <- lm(delta_rf ~ arm * race, data=my_whi_dm_long)
  # mod2 <- lm(delta_rf ~ arm * score + arm * race, data=my_whi_dm_long)
  # print(anova(mod1, mod2))
  
  print(summary(lm(delta_rf ~ score, 
                   data=filter(my_whi_dm_long, arm == "intervention")))$coef)
  print(summary(lm(delta_rf ~ score + baseline_ldl, 
                   data=filter(my_whi_dm_long, arm == "intervention")))$coef)
  
  my_whi_dm_long_plt_data <- my_whi_dm_long %>%
    group_by(arm, score_bucket) %>%
    summarise(mean_delta_rf=mean(delta_rf),
              SD=sd(delta_rf),
              SE=SD / sqrt(n()))
  
  by_assignment_plt <- ggplot(data=my_whi_dm_long_plt_data, aes(x=score_bucket)) +
    geom_boxplot(data=my_whi_dm_long, aes(y=delta_rf), alpha=0.1) +
    geom_point(data=my_whi_dm_long, aes(x=score_bucket, y=delta_rf),
               alpha=0.3, position=position_jitter(width=0.3)) +
    coord_cartesian(ylim=quantile(my_whi_dm_long$delta_rf, c(0.25, 0.75)) + 
                      c(-1.5, 1.5) * IQR(my_whi_dm_long$delta_rf)) +
    facet_wrap(~arm)
  
  means_plus_ses_plt <- ggplot(data=my_whi_dm_long_plt_data, aes(x=score_bucket)) +
    geom_point(data=my_whi_dm_long_plt_data, 
               aes(x=score_bucket, y=mean_delta_rf), 
               stat="identity") +
    geom_errorbar(data=my_whi_dm_long_plt_data, 
                  aes(ymin=mean_delta_rf - SE, ymax=mean_delta_rf + SE), width=0) +
    facet_wrap(~arm)
  
  plot_grid(by_assignment_plt, means_plus_ses_plt, nrow=2)
}
# testlong("../data/processed/gen2/sfa_ldl/whi_scores_whi_white.sscore", "delta_ldl")
testlong("../data/processed/gen2/sfa_metabolic/whi_scores_metab.sscore", "delta_sbp")
```

```{r non-score-correlations, eval=F}
nonscore <- whi_dm_long %>%
  select(delta_ldl, delta_sfa_pct, delta_f2c, delta_bmi, arm, baseline_sfa_pct, baseline_f2c)

cor(model.matrix(~.-1, data=nonscore), method="spearman") %>%
  heatmap(symm=T)
```

```{r whi-longitudinal-measured, eval=F}
measured_plt_data <- full_data %>%
  filter(subset %in% c("whi_intervention", "whi_control")) %>%
  mutate(score_tertile=cut(stack_score,
                           # breaks=quantile(.$stack_score, c(0, 0.33, 0.66, 1)),
                           breaks=quantile(.$stack_score, c(0, 0.25, 0.5, 0.75, 1)),
                           labels=paste0("Q", 1:4),
                           include.lowest=T)) %>%
  mutate(subset=ifelse(delta_sfa < 0, "decreased SFA", "didn't decrease SFA")) %>%
  group_by(subset, score_tertile) %>%
  summarise(delta=mean(delta_ldl),
            se=sd(delta_ldl) / sqrt(n()))
  
ggplot(measured_plt_data, aes(x=score_tertile, y=delta)) +
  geom_point() +
  geom_errorbar(aes(ymin=delta - se, ymax=delta + se), width=0) +
  facet_wrap(~subset)
```

```{r fhs-longitudinal-measured, eval=F}
measured_plt_data <- full_data %>%
  filter(subset %in% c("whi_intervention", "whi_control")) %>%
  mutate(score_tertile=cut(stack_score,
                           # breaks=quantile(.$stack_score, c(0, 0.33, 0.66, 1)),
                           breaks=quantile(.$stack_score, c(0, 0.25, 0.5, 0.75, 1)),
                           labels=paste0("Q", 1:4),
                           include.lowest=T)) %>%
  mutate(subset=ifelse(delta_sfa < 0, "decreased SFA", "didn't decrease SFA")) %>%
  group_by(subset, score_tertile) %>%
  summarise(delta=mean(delta_ldl),
            se=sd(delta_ldl) / sqrt(n()))
  
ggplot(measured_plt_data, aes(x=score_tertile, y=delta)) +
  geom_point() +
  geom_errorbar(aes(ymin=delta - se, ymax=delta + se), width=0) +
  facet_wrap(~subset)
```

```{r interaction-tests, eval=F}
print("Test for arm-score interaction in predicting delta_LDL:")
summary(lm(delta_ldl ~ arm * score, data=whi_dm_long))$coef

print("Same test, additionally adjusting for baseline LDL:")
summary(lm(delta_ldl ~ arm * score + baseline_ldl, data=whi_dm_long))$coef
```

# Moving forward

* Application is in to UK Biobank to try to replicate any findings in the 500k subjects there
* How well do LDL-C or CardioGxE-constructed scores work in comparison to this one?

